# Comparing `tmp/rna_tools-3.8.5-py3-none-any.whl.zip` & `tmp/rna_tools-3.8.6-py3-none-any.whl.zip`

## zipinfo {}

```diff
@@ -1,511 +1,515 @@
-Zip file size: 1074112 bytes, number of entries: 509
--rw-r--r--  2.0 unx     1305 b- defN 22-Feb-21 22:11 rna_tools/BlastPDB.py
--rw-r--r--  2.0 unx     3815 b- defN 22-Feb-21 22:11 rna_tools/RfamSearch.py
--rw-r--r--  2.0 unx     3819 b- defN 22-Feb-21 22:11 rna_tools/SecondaryStructure.py
--rw-r--r--  2.0 unx    27464 b- defN 22-Feb-21 22:11 rna_tools/Seq.py
--rw-r--r--  2.0 unx       93 b- defN 22-Feb-21 22:11 rna_tools/__init__.py
--rw-r--r--  2.0 unx      496 b- defN 22-Feb-21 22:11 rna_tools/_version.py
--rw-r--r--  2.0 unx     2466 b- defN 22-Feb-21 22:11 rna_tools/rna_calc_fenergy.py
--rw-r--r--  2.0 unx     2448 b- defN 22-Feb-21 22:11 rna_tools/rna_dot2ct.py
--rw-r--r--  2.0 unx     1756 b- defN 22-Feb-21 22:11 rna_tools/rna_pdb_replace.py
--rw-r--r--  2.0 unx    41423 b- defN 22-Feb-21 22:11 rna_tools/rna_pdb_toolsx.py
--rw-r--r--  2.0 unx     2172 b- defN 22-Feb-21 22:11 rna_tools/rna_secondary_structure_prediction.py
--rw-r--r--  2.0 unx    12473 b- defN 22-Feb-21 22:11 rna_tools/rna_simrnaweb_download_job.py
--rw-r--r--  2.0 unx     3713 b- defN 22-Feb-21 22:11 rna_tools/rna_tools_config.py
--rw-r--r--  2.0 unx    82236 b- defN 22-Feb-21 22:11 rna_tools/rna_tools_lib.py
--rw-r--r--  2.0 unx     1342 b- defN 22-Feb-21 22:11 rna_tools/rna_tools_test_all.py
--rw-r--r--  2.0 unx      709 b- defN 22-Feb-21 22:11 rna_tools/update_readme.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/__init__.py
--rw-r--r--  2.0 unx     3080 b- defN 22-Feb-21 22:11 rna_tools/tools/ClashCalc/ClashCalc.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/ClashCalc/__init__.py
--rw-r--r--  2.0 unx    35375 b- defN 22-Feb-21 22:11 rna_tools/tools/PyMOL4RNA/PyMOL4RNA.py
--rw-r--r--  2.0 unx    13336 b- defN 22-Feb-21 22:11 rna_tools/tools/PyMOL4RNA/PyMOL4Spliceosome.py
--rw-r--r--  2.0 unx     6836 b- defN 22-Feb-21 22:11 rna_tools/tools/PyMOL4RNA/Spliceosome_PyMOL.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/PyMOL4RNA/__init__.py
--rw-r--r--  2.0 unx    15836 b- defN 22-Feb-21 22:11 rna_tools/tools/PyMOL4RNA/code_for_color_spl.py
--rw-r--r--  2.0 unx     7898 b- defN 22-Feb-21 22:11 rna_tools/tools/PyMOL4RNA/code_for_spl.py
--rw-r--r--  2.0 unx    11652 b- defN 22-Feb-21 22:11 rna_tools/tools/PyMOL4RNA/external_flatten_object.py
--rw-r--r--  2.0 unx     5431 b- defN 22-Feb-21 22:11 rna_tools/tools/PyMOL4RNA/rna_draw_edges.py
--rw-r--r--  2.0 unx     3208 b- defN 22-Feb-21 22:11 rna_tools/tools/PyMOL4RNA/rnapuzzles.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/PyMOL4RNA/libs/__init__.py
--rw-r--r--  2.0 unx     7612 b- defN 22-Feb-21 22:11 rna_tools/tools/PyMOL4RNA/libs/get_raw_distances.py
--rw-r--r--  2.0 unx    15707 b- defN 22-Feb-21 22:11 rna_tools/tools/PyMOL4RNA/libs/show_contacts.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/clanstix/__init__.py
--rw-r--r--  2.0 unx    21529 b- defN 22-Feb-21 22:11 rna_tools/tools/clanstix/rna_clanstix.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_app/__init__.py
--rw-r--r--  2.0 unx     5685 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_app/clarna_app.py
--rw-r--r--  2.0 unx      603 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_app/test.py
--rw-r--r--  2.0 unx    14950 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/PDBSeq.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/__init__.py
--rw-r--r--  2.0 unx    48062 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/rna_clarna_compare.py
--rw-r--r--  2.0 unx    45347 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/rna_clarna_run.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/__init__.py
--rw-r--r--  2.0 unx    13864 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/aggregate-doublets-dict.py
--rw-r--r--  2.0 unx     2945 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/align-structures.py
--rw-r--r--  2.0 unx   782660 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/bp_limits.py
--rw-r--r--  2.0 unx      803 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/check-descriptions-dict.py
--rw-r--r--  2.0 unx       33 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/cl_settings.py
--rw-r--r--  2.0 unx     1911 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/cl_settings_global.py
--rw-r--r--  2.0 unx    46848 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/clarna.py
--rw-r--r--  2.0 unx    10101 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/cluster-doublets.py
--rw-r--r--  2.0 unx    28593 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/combine-contact-graphs.py
--rw-r--r--  2.0 unx     4658 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/compute-bp-limits.py
--rw-r--r--  2.0 unx    63976 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/compute-classifier.py
--rw-r--r--  2.0 unx     2609 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/cross-validation-groups.py
--rw-r--r--  2.0 unx     6898 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/cross-validation-groups2.py
--rw-r--r--  2.0 unx    43738 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/distances.py
--rw-r--r--  2.0 unx    14757 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/doublet-params.py
--rw-r--r--  2.0 unx     5309 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/draw-contact-graph.py
--rw-r--r--  2.0 unx     5111 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/eval-graphs.py
--rw-r--r--  2.0 unx     4267 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/export-ref-classes.py
--rw-r--r--  2.0 unx    15896 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/extract-contacts.py
--rw-r--r--  2.0 unx     2215 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/extract-doublets-from-graph.py
--rw-r--r--  2.0 unx    98501 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/fabfile.py
--rw-r--r--  2.0 unx    23171 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/fix-reference-groups.py
--rw-r--r--  2.0 unx     9592 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/gen-other-groups.py
--rw-r--r--  2.0 unx    12132 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/gen-pdb.py
--rw-r--r--  2.0 unx     3245 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/graph-to-secondary-structure.py
--rw-r--r--  2.0 unx     3193 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/json-tool.py
--rw-r--r--  2.0 unx     4098 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/make-residue-dict.py
--rw-r--r--  2.0 unx     3380 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/make-reverse-descriptions-dict.py
--rw-r--r--  2.0 unx     2452 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/merge-dicts.py
--rw-r--r--  2.0 unx    11530 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/moderna-stackings.py
--rw-r--r--  2.0 unx     3656 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/morph-doublet.py
--rw-r--r--  2.0 unx     4083 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/normalize_pdb.py
--rw-r--r--  2.0 unx    12645 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/progressbar.py
--rw-r--r--  2.0 unx     2300 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/show-doublet-info.py
--rw-r--r--  2.0 unx     9025 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/structure_ciach.py
--rw-r--r--  2.0 unx    21974 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/supp-files.py
--rw-r--r--  2.0 unx     7801 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/testcase.py
--rw-r--r--  2.0 unx     1867 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/translate-doublets-ids.py
--rw-r--r--  2.0 unx    48150 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/utils.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/get_pairs/__init__.py
--rw-r--r--  2.0 unx     2616 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/get_pairs/get_pairs.py
--rw-r--r--  2.0 unx      157 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/ClaRNAlib/get_pairs/test.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/lib/__init__.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/lib/rna_pdb_tools/__init__.py
--rw-r--r--  2.0 unx    38027 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/lib/rna_pdb_tools/pdb_parser_lib.py
--rw-r--r--  2.0 unx     6354 b- defN 22-Feb-21 22:11 rna_tools/tools/clarna_play/lib/rna_pdb_tools/rna-pdb-tools.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/cluster_load/__init__.py
--rw-r--r--  2.0 unx     2943 b- defN 22-Feb-21 22:11 rna_tools/tools/cluster_load/cluster_load.py
--rw-r--r--  2.0 unx      319 b- defN 22-Feb-21 22:11 rna_tools/tools/cluster_load/cluster_mini_free.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/diffpdb/__init__.py
--rw-r--r--  2.0 unx     4645 b- defN 22-Feb-21 22:11 rna_tools/tools/diffpdb/diffpdb.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/extra_functions/__init__.py
--rw-r--r--  2.0 unx      506 b- defN 22-Feb-21 22:11 rna_tools/tools/extra_functions/gen_numbering.py
--rw-r--r--  2.0 unx     3744 b- defN 22-Feb-21 22:11 rna_tools/tools/extra_functions/select_fragment.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/md/__init__.py
--rw-r--r--  2.0 unx     1405 b- defN 22-Feb-21 22:11 rna_tools/tools/md/rna_get_energy.py
--rw-r--r--  2.0 unx      932 b- defN 22-Feb-21 22:11 rna_tools/tools/md/rna_gromacs_ready.py
--rw-r--r--  2.0 unx     5940 b- defN 22-Feb-21 22:11 rna_tools/tools/md/rna_md.py
--rw-r--r--  2.0 unx     3276 b- defN 22-Feb-21 22:11 rna_tools/tools/md/rna_minimize.py
--rw-r--r--  2.0 unx     1195 b- defN 22-Feb-21 22:11 rna_tools/tools/md/rna_traj_n.py
--rw-r--r--  2.0 unx     1164 b- defN 22-Feb-21 22:11 rna_tools/tools/md/rna_traj_nohetm.py
--rw-r--r--  2.0 unx     1811 b- defN 22-Feb-21 22:11 rna_tools/tools/md/rna_traj_range.py
--rw-r--r--  2.0 unx    20335 b- defN 22-Feb-21 22:11 rna_tools/tools/md/rnakb_utils.py
--rw-r--r--  2.0 unx     1717 b- defN 22-Feb-21 22:11 rna_tools/tools/md/test.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/__init__.py
--rw-r--r--  2.0 unx    19630 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/CheckPdb.py
--rw-r--r--  2.0 unx     4810 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/Constants.py
--rw-r--r--  2.0 unx     1089 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/FragmentInsertion.py
--rw-r--r--  2.0 unx     6839 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/Helix.py
--rw-r--r--  2.0 unx    10220 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/ModelingRecipe.py
--rw-r--r--  2.0 unx    25493 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/ModernaFragment.py
--rw-r--r--  2.0 unx    11822 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/ModernaStructure.py
--rw-r--r--  2.0 unx     3043 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/ModernaSuperimposer.py
--rw-r--r--  2.0 unx     9166 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/PdbConverter.py
--rw-r--r--  2.0 unx     9262 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/RNAChain.py
--rw-r--r--  2.0 unx    17244 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/RNAModel.py
--rw-r--r--  2.0 unx    12712 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/RNAResidue.py
--rw-r--r--  2.0 unx     5992 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/Renumerator.py
--rw-r--r--  2.0 unx     3052 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/Renumerator_new.py
--rw-r--r--  2.0 unx     1837 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/Template.py
--rw-r--r--  2.0 unx      564 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/__init__.py
--rw-r--r--  2.0 unx     7626 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/_version.py
--rw-r--r--  2.0 unx    44552 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/commands.py
--rw-r--r--  2.0 unx     4642 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/moderna.py
--rw-r--r--  2.0 unx     5696 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/BasePairCalculator.py
--rw-r--r--  2.0 unx     7527 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/BaseRecognizer.py
--rw-r--r--  2.0 unx     4459 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/ChainConnectivity.py
--rw-r--r--  2.0 unx     7680 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/ClashRecognizer.py
--rw-r--r--  2.0 unx     5299 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/Constants.py
--rw-r--r--  2.0 unx     3157 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/GeometryAnalyzer.py
--rw-r--r--  2.0 unx     9854 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/GeometryParameters.py
--rw-r--r--  2.0 unx    13260 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/GeometryStatistics.py
--rw-r--r--  2.0 unx     6691 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/HBondCalculator.py
--rw-r--r--  2.0 unx     8995 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/MolGraphParser.py
--rw-r--r--  2.0 unx     8941 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/MolParameters.py
--rw-r--r--  2.0 unx    28177 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/MolTopologies.py
--rw-r--r--  2.0 unx    23655 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/MolecularGraph.py
--rw-r--r--  2.0 unx     3526 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/PuckerCalculator.py
--rw-r--r--  2.0 unx     3178 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/RNASuites.py
--rw-r--r--  2.0 unx     8097 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/StackingCalculator.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/analyze/__init__.py
--rw-r--r--  2.0 unx     7625 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/builder/BackboneBuilder.py
--rw-r--r--  2.0 unx     1026 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/builder/ChiRotator.py
--rw-r--r--  2.0 unx      401 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/builder/Constants.py
--rw-r--r--  2.0 unx     1974 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/builder/CoordBuilder.py
--rw-r--r--  2.0 unx     5086 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/builder/FCCDLoopCloser.py
--rw-r--r--  2.0 unx    12133 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/builder/PhosphateBuilder.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/builder/__init__.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/examples/__init__.py
--rw-r--r--  2.0 unx     6477 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/examples/usage_examples.py
--rw-r--r--  2.0 unx     8489 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/fragment_library/LIR.py
--rw-r--r--  2.0 unx    11310 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/fragment_library/LIRdb.py
--rw-r--r--  2.0 unx    21344 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/fragment_library/SearchLIR.py
--rw-r--r--  2.0 unx     4552 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/fragment_library/StructureLibrary.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/fragment_library/__init__.py
--rw-r--r--  2.0 unx     3674 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/isosteric/IsostericModelling.py
--rw-r--r--  2.0 unx    10949 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/isosteric/Isostericity.py
--rw-r--r--  2.0 unx     3035 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/isosteric/IsostericityMatrices.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/isosteric/__init__.py
--rw-r--r--  2.0 unx     3089 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/modifications/BaseExchanger.py
--rw-r--r--  2.0 unx     4052 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/modifications/ModificationAdder.py
--rw-r--r--  2.0 unx     4170 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/modifications/ModificationRemover.py
--rw-r--r--  2.0 unx     1468 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/modifications/ModifyResidue.py
--rw-r--r--  2.0 unx     4826 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/modifications/ResidueEditor.py
--rw-r--r--  2.0 unx      494 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/modifications/__init__.py
--rw-r--r--  2.0 unx     7245 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/sequence/AlignmentMatcher.py
--rw-r--r--  2.0 unx     6090 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/sequence/ModernaAlphabet.py
--rw-r--r--  2.0 unx     6619 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/sequence/ModernaSequence.py
--rw-r--r--  2.0 unx    17350 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/sequence/RNAAlignment.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/sequence/__init__.py
--rw-r--r--  2.0 unx     3546 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/sequence/commands.py
--rw-r--r--  2.0 unx     2271 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/util/Errors.py
--rw-r--r--  2.0 unx     3075 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/util/LogFile.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/util/__init__.py
--rw-r--r--  2.0 unx     2210 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/util/decorators.py
--rw-r--r--  2.0 unx     5837 b- defN 22-Feb-21 22:11 rna_tools/tools/mini_moderna3/moderna/util/validators.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/misc/__init__.py
--rw-r--r--  2.0 unx      154 b- defN 22-Feb-21 22:11 rna_tools/tools/misc/reverse.py
--rw-r--r--  2.0 unx     2394 b- defN 22-Feb-21 22:11 rna_tools/tools/misc/rna_add_chain.py
--rw-r--r--  2.0 unx      916 b- defN 22-Feb-21 22:11 rna_tools/tools/misc/rna_csv_sort.py
--rw-r--r--  2.0 unx     1207 b- defN 22-Feb-21 22:11 rna_tools/tools/misc/rna_merge_dfs.py
--rw-r--r--  2.0 unx     8082 b- defN 22-Feb-21 22:11 rna_tools/tools/misc/rna_merge_two_dfs.py
--rwxr-xr-x  2.0 unx       64 b- defN 22-Feb-21 22:11 rna_tools/tools/misc/rna_ok
--rw-r--r--  2.0 unx     1378 b- defN 22-Feb-21 22:11 rna_tools/tools/misc/rna_ok_mail.py
--rw-r--r--  2.0 unx      691 b- defN 22-Feb-21 22:11 rna_tools/tools/misc/rna_tools_which.py
--rw-r--r--  2.0 unx      502 b- defN 22-Feb-21 22:11 rna_tools/tools/misc/translate.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/__init__.py
--rw-r--r--  2.0 unx    24593 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/rna_mq_collect.py
--rw-r--r--  2.0 unx     2441 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/rna_mq_exp.py
--rw-r--r--  2.0 unx     1441 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/rna_mq_plot.py
--rwxr-xr-x  2.0 unx      290 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/rna_mq_remote.sh
--rw-r--r--  2.0 unx     4648 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/rna_mq_score.py
--rw-r--r--  2.0 unx     2726 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/rna_mq_wx_collect.py
--rw-r--r--  2.0 unx     4458 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/AnalyzeGeometry/AnalyzeGeometry.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/AnalyzeGeometry/__init__.py
--rw-r--r--  2.0 unx     3050 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/ClashScore/ClashScore.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/ClashScore/__init__.py
--rw-r--r--  2.0 unx     1182 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/ClashScore/rna_mq_clashscore.py
--rw-r--r--  2.0 unx     2735 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/Dfire/Dfire.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/Dfire/__init__.py
--rw-r--r--  2.0 unx     1229 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/Dfire/rna_mq_dfire.py
--rw-r--r--  2.0 unx     9627 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/FARFAR2/FARFAR2.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/FARFAR2/__init__.py
--rw-r--r--  2.0 unx     2448 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/FARFAR2/rna_mq_farfar2.py
--rw-r--r--  2.0 unx       19 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/FARFAR2/test/__init__.py
--rw-r--r--  2.0 unx     8873 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/FARFAR2/test/test.py
--rw-r--r--  2.0 unx     7252 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/FARNA/FARNA.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/FARNA/__init__.py
--rw-r--r--  2.0 unx     5057 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/FARNA/rna_mq_farna.py
--rw-r--r--  2.0 unx       19 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/FARNA/test_data/__init__.py
--rw-r--r--  2.0 unx     8873 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/FARNA/test_data/test.py
--rw-r--r--  2.0 unx     5902 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/QRNAS/QRNAS.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/QRNAS/__init__.py
--rw-r--r--  2.0 unx       19 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/QRNAS/test/__init__.py
--rw-r--r--  2.0 unx     1699 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/QRNAS/test/test.py
--rw-r--r--  2.0 unx     5234 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RASP/RASP.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RASP/__init__.py
--rw-r--r--  2.0 unx     1384 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RASP/rna_mq_rasp.py
--rw-r--r--  2.0 unx      382 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RASP/test.py
--rw-r--r--  2.0 unx       19 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RASP/test/__init__.py
--rw-r--r--  2.0 unx     6408 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RASP/test/test.py
--rw-r--r--  2.0 unx     4639 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RNA3DCNN/RNA3DCNN.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RNA3DCNN/__init__.py
--rw-r--r--  2.0 unx      962 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RNA3DCNN/rna_mq_rna3dcnn.py
--rw-r--r--  2.0 unx    14264 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RNAkb/RNAkb.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RNAkb/__init__.py
--rw-r--r--  2.0 unx     1720 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RNAkb/rna_mq_rnakb.py
--rw-r--r--  2.0 unx     2783 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RNAscore/RNAscore.py
--rw-r--r--  2.0 unx        1 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/RNAscore/__init__.py
--rw-r--r--  2.0 unx     4953 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/SimRNA/SimRNA.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/SimRNA/__init__.py
--rw-r--r--  2.0 unx     1559 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/SimRNA/rna_mq_simrna.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/eSCORE/__init__.py
--rw-r--r--  2.0 unx     2752 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/eSCORE/eSCORE.py
--rw-r--r--  2.0 unx      921 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/eSCORE/rna_mq_eSCORE.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/lib/__init__.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/lib/shellgraphics/__init__.py
--rw-r--r--  2.0 unx    10890 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/lib/shellgraphics/shellgraphics.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/lib/timex/__init__.py
--rw-r--r--  2.0 unx     1341 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/lib/timex/timex.py
--rw-r--r--  2.0 unx     2582 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/lib/wrappers/SubprocessUtils.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/lib/wrappers/__init__.py
--rw-r--r--  2.0 unx     8823 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/lib/wrappers/base_wrappers.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/lib/wrappers/test/__init__.py
--rw-r--r--  2.0 unx     1226 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/lib/wrappers/test/test.py
--rw-r--r--  2.0 unx        1 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/rsRNASP/__init__.py
--rw-r--r--  2.0 unx      925 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/rsRNASP/rna_mq_rsRNASP.py
--rw-r--r--  2.0 unx     2718 b- defN 22-Feb-21 22:11 rna_tools/tools/mq/rsRNASP/rsRNASP.py
--rw-r--r--  2.0 unx    12536 b- defN 22-Feb-21 22:11 rna_tools/tools/pdb_formatix/PDBFile.py
--rw-r--r--  2.0 unx     1964 b- defN 22-Feb-21 22:11 rna_tools/tools/pdb_formatix/RosettaUtils.py
--rw-r--r--  2.0 unx     1492 b- defN 22-Feb-21 22:11 rna_tools/tools/pdb_formatix/SingleLineUtils.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/pdb_formatix/__init__.py
--rw-r--r--  2.0 unx     2612 b- defN 22-Feb-21 22:11 rna_tools/tools/pdb_formatix/rebuilder.py
--rw-r--r--  2.0 unx      623 b- defN 22-Feb-21 22:11 rna_tools/tools/pdb_formatix/test-quick-and-dirty.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/pdb_formatix/test/__init__.py
--rw-r--r--  2.0 unx     5106 b- defN 22-Feb-21 22:11 rna_tools/tools/pdb_formatix/test/test.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/pdbs_measure_atom_dists/__init__.py
--rw-r--r--  2.0 unx     5992 b- defN 22-Feb-21 22:11 rna_tools/tools/pdbs_measure_atom_dists/pdbs_measure_atom_dists.py
--rw-r--r--  2.0 unx        1 b- defN 22-Feb-21 22:11 rna_tools/tools/plotting/__init__.py
--rw-r--r--  2.0 unx     2276 b- defN 22-Feb-21 22:11 rna_tools/tools/plotting/rna_plot_boxplotlike.py
--rw-r--r--  2.0 unx     1623 b- defN 22-Feb-21 22:11 rna_tools/tools/plotting/rna_plot_dendogram.py
--rw-r--r--  2.0 unx     1793 b- defN 22-Feb-21 22:11 rna_tools/tools/plotting/rna_plot_density.py
--rw-r--r--  2.0 unx     2037 b- defN 22-Feb-21 22:11 rna_tools/tools/plotting/rna_plot_heatmap.py
--rw-r--r--  2.0 unx     1959 b- defN 22-Feb-21 22:11 rna_tools/tools/plotting/rna_plot_hist.py
--rwxr-xr-x  2.0 unx     2104 b- defN 22-Feb-21 22:11 rna_tools/tools/pymol_color_by_conserv/pymol_color_by_conserv.py
--rw-r--r--  2.0 unx        1 b- defN 22-Feb-21 22:11 rna_tools/tools/pymol_drawing/__init__.py
--rw-r--r--  2.0 unx     3539 b- defN 22-Feb-21 22:11 rna_tools/tools/pymol_drawing/pymol_drawing.py
--rwxr-xr-x  2.0 unx     3241 b- defN 22-Feb-21 22:11 rna_tools/tools/pymol_preview_generator/pymol_preview_generator.py
--rwxr-xr-x  2.0 unx      996 b- defN 22-Feb-21 22:11 rna_tools/tools/pymol_preview_generator/pymol_preview_install.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/renum_pdb_to_aln/__init__.py
--rw-r--r--  2.0 unx     3954 b- defN 22-Feb-21 22:11 rna_tools/tools/renum_pdb_to_aln/renum_pdb_to_aln.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/__init__.py
--rw-r--r--  2.0 unx    10159 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/random_assignment_of_nucleotides.py
--rw-r--r--  2.0 unx     2877 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/rna_align_distance_to_seq.py
--rw-r--r--  2.0 unx      650 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/rna_align_find_core.py
--rw-r--r--  2.0 unx     2174 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/rna_align_find_seq_in_alignment.py
--rw-r--r--  2.0 unx    13333 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/rna_align_foldability.py
--rw-r--r--  2.0 unx      953 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/rna_align_get_ss_from_fasta.py
--rw-r--r--  2.0 unx     1340 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/rna_align_get_ss_from_stk.py
--rw-r--r--  2.0 unx     1675 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/rna_align_seq_to_alignment.py
--rw-r--r--  2.0 unx    45244 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/rna_alignment.py
--rw-r--r--  2.0 unx     1786 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/rna_alignment_r2r.py
--rw-r--r--  2.0 unx     2361 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/test.py
--rw-r--r--  2.0 unx     6910 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/test_rna_alignment.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/utils/__init__.py
--rw-r--r--  2.0 unx     2336 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/utils/mfa-get-seq-for-species.py
--rw-r--r--  2.0 unx      725 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/utils/rna_align_strip_stk.py
--rw-r--r--  2.0 unx     8104 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/utils/rna_alignment_calc_energy.py
--rw-r--r--  2.0 unx     9721 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/utils/rna_alignment_get_species.py
--rw-r--r--  2.0 unx     4358 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/utils/rna_alignment_merge_by_species.py
--rw-r--r--  2.0 unx     1932 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/utils/rna_alignment_process_id.py
--rw-r--r--  2.0 unx     2314 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/utils/rna_alignment_r2r.py
--rw-r--r--  2.0 unx      959 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_alignment/utils/rna_alignment_sto_remove_uniq_ids.py
--rwxr-xr-x  2.0 unx      954 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_binding_affinity/rna_dG2kd.py
--rwxr-xr-x  2.0 unx     1012 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_binding_affinity/rna_kd2dG.py
--rwxr-xr-x  2.0 unx      718 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_binding_affinity/rna_kd2pkd.py
--rwxr-xr-x  2.0 unx      899 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_binding_affinity/rna_pkd2dG.py
--rwxr-xr-x  2.0 unx      796 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_binding_affinity/rna_pkd2kd.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_bp/__init__.py
--rw-r--r--  2.0 unx     7281 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_bp/base_pair.py
--rw-r--r--  2.0 unx     2737 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_evo_rmsd/RNAalignment.py
--rw-r--r--  2.0 unx     5639 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_evo_rmsd/RNAmodel.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_evo_rmsd/__init__.py
--rw-r--r--  2.0 unx     6818 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_evo_rmsd/rna_calc_evo_rmsd.py
--rwxr-xr-x  2.0 unx     2448 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_hbonds/rna_calc_hbonds.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_inf/__init__.py
--rw-r--r--  2.0 unx     5638 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_inf/rna_calc_dinf.py
--rw-r--r--  2.0 unx    10008 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_inf/rna_calc_inf.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_rmsd/__init__.py
--rw-r--r--  2.0 unx    13443 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_rmsd/rna_calc_rmsd.py
--rw-r--r--  2.0 unx     4199 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_rmsd/rna_calc_rmsd_all_vs_all.py
--rw-r--r--  2.0 unx    11477 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_rmsd/rna_calc_rmsd_biopython.py
--rw-r--r--  2.0 unx     3978 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_rmsd/rna_calc_rmsd_multi_targets.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_rmsd/lib/__init__.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_rmsd/lib/rmsd/__init__.py
--rw-r--r--  2.0 unx     6911 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_rmsd/lib/rmsd/calculate_rmsd.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_rmsd_trafl/__init__.py
--rw-r--r--  2.0 unx     1331 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_rmsd_trafl/rna_cal_rmsd_trafl_plot.py
--rw-r--r--  2.0 unx     7170 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_calc_rmsd_trafl/rna_calc_rmsd_trafl.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_convert_pseudoknot_formats/__init__.py
--rw-r--r--  2.0 unx     1052 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_convert_pseudoknot_formats/rna_pk_simrna_to_one_line.py
--rw-r--r--  2.0 unx     1334 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_convert_pseudoknot_formats/rna_ss_pk_to_simrna.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/__init__.py
--rw-r--r--  2.0 unx      637 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/pymol_dists.py
--rw-r--r--  2.0 unx      878 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_csv2pairs.py
--rw-r--r--  2.0 unx    12319 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_dca_mapping.py
--rw-r--r--  2.0 unx     1166 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_dca_select_interactions.py
--rw-r--r--  2.0 unx     2806 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_ec2x.py
--rw-r--r--  2.0 unx    11906 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_filter.py
--rw-r--r--  2.0 unx    11051 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_get_distances.py
--rw-r--r--  2.0 unx     9035 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_get_dists.py
--rw-r--r--  2.0 unx     1802 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_pairs2SimRNArestrs.py
--rw-r--r--  2.0 unx     1524 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_pairs2rfilter.py
--rw-r--r--  2.0 unx     1397 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_pairs_diff.py
--rw-r--r--  2.0 unx     1373 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_restrs2SimRNArestrs.py
--rw-r--r--  2.0 unx     1299 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_restrs2mqapRNA.py
--rw-r--r--  2.0 unx     1658 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/rna_ss_get_bps.py
--rw-r--r--  2.0 unx      884 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/simrna_get_restraints.py
--rw-r--r--  2.0 unx      990 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_filter/test_rna_filter.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_helix_vis/__init__.py
--rw-r--r--  2.0 unx     4068 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_helix_vis/rna_helix_vis.py
--rwxr-xr-x  2.0 unx      428 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_multimodels/rna_pdb_merge_into_one.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_pdb_edit_occupancy_bfactor/__init__.py
--rw-r--r--  2.0 unx     6158 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_pdb_edit_occupancy_bfactor/rna_pdb_edit_occupancy_bfactor.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_pdb_merge_structure_with_fragments/__init__.py
--rw-r--r--  2.0 unx     4918 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_pdb_merge_structure_with_fragments/rna_pdb_merge_structure_with_fragments.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_prediction_significance/__init__.py
--rw-r--r--  2.0 unx     1088 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_prediction_significance/rna_prediction_significance.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_refinement/__init__.py
--rw-r--r--  2.0 unx     7043 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_refinement/rna_refinement.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_rosetta/__init__.py
--rw-r--r--  2.0 unx     6083 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_rosetta/rna_rosetta_check_progress.py
--rw-r--r--  2.0 unx     5475 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_rosetta/rna_rosetta_cluster.py
--rw-r--r--  2.0 unx     1536 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_rosetta/rna_rosetta_easy_cat.py
--rw-r--r--  2.0 unx     1536 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_rosetta/rna_rosetta_extract_lowscore_decoys.py
--rw-r--r--  2.0 unx     3990 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_rosetta/rna_rosetta_head.py
--rw-r--r--  2.0 unx     4916 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_rosetta/rna_rosetta_min.py
--rw-r--r--  2.0 unx     1058 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_rosetta/rna_rosetta_n.py
--rw-r--r--  2.0 unx    10108 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_rosetta/rna_rosetta_run.py
--rw-r--r--  2.0 unx      649 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_rosetta/test.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_sali2dotbracket/__init__.py
--rw-r--r--  2.0 unx     3463 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_sali2dotbracket/rna_sali2dotbracket.py
--rw-r--r--  2.0 unx     3958 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_seq_search_BLASTn_outfmt-6/select_seq_fromBLAStn_6outfm.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_x3dna/__init__.py
--rw-r--r--  2.0 unx     6727 b- defN 22-Feb-21 22:11 rna_tools/tools/rna_x3dna/rna_x3dna.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/simrna_trajectory/__init__.py
--rw-r--r--  2.0 unx     1235 b- defN 22-Feb-21 22:11 rna_tools/tools/simrna_trajectory/rna_simrna_cluster.py
--rw-r--r--  2.0 unx      848 b- defN 22-Feb-21 22:11 rna_tools/tools/simrna_trajectory/rna_simrna_energy_plot.py
--rw-r--r--  2.0 unx     3649 b- defN 22-Feb-21 22:11 rna_tools/tools/simrna_trajectory/rna_simrna_extract.py
--rw-r--r--  2.0 unx     1278 b- defN 22-Feb-21 22:11 rna_tools/tools/simrna_trajectory/rna_simrna_lowest.py
--rw-r--r--  2.0 unx      723 b- defN 22-Feb-21 22:11 rna_tools/tools/simrna_trajectory/rna_simrna_subset.py
--rw-r--r--  2.0 unx    10181 b- defN 22-Feb-21 22:11 rna_tools/tools/simrna_trajectory/simrna_trajectory.py
--rw-r--r--  2.0 unx      406 b- defN 22-Feb-21 22:11 rna_tools/tools/simrna_trajectory/test_rmsd.py
--rw-r--r--  2.0 unx      232 b- defN 22-Feb-21 22:11 rna_tools/tools/simrna_trajectory/test_rna_simrna.py
--rwxr-xr-x  2.0 unx    16459 b- defN 22-Feb-21 22:11 rna_tools/tools/triplexibility/triplexibility.py
--rwxr-xr-x  2.0 unx    13684 b- defN 22-Feb-21 22:11 rna_tools/tools/triplexibility/triplexibility2.py
--rwxr-xr-x  2.0 unx     1551 b- defN 22-Feb-21 22:11 rna_tools/tools/triplexibility/trx_mutgen.py
--rwxr-xr-x  2.0 unx     3295 b- defN 22-Feb-21 22:11 rna_tools/tools/triplexibility/trx_score.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/__init__.py
--rw-r--r--  2.0 unx     2640 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/checker.py
--rw-r--r--  2.0 unx     4553 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/cleanup.py
--rw-r--r--  2.0 unx    11839 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/daemon.py
--rw-r--r--  2.0 unx      246 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/manage.py
--rw-r--r--  2.0 unx     2712 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/sendmail.py
--rw-r--r--  2.0 unx      632 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/wsgi.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/app/__init__.py
--rw-r--r--  2.0 unx      508 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/app/admin.py
--rw-r--r--  2.0 unx     8356 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/app/models.py
--rw-r--r--  2.0 unx    20146 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/app/views.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/web/__init__.py
--rw-r--r--  2.0 unx     6078 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/web/settings.py
--rw-r--r--  2.0 unx     1943 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/web/urls.py
--rw-r--r--  2.0 unx     1410 b- defN 22-Feb-21 22:11 rna_tools/tools/webserver-engine/web/wsgi.py
--rwxr-xr-x  2.0 unx     2638 b- defN 22-Feb-21 22:11 rna_tools/tools/workflow/rna_workflow.py
--rwxr-xr-x  2.0 unx    46835 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/clarna.py
--rwxr-xr-x  2.0 unx     5672 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/clarna_app.py
--rwxr-xr-x  2.0 unx     4636 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/diffpdb.py
--rwxr-xr-x  2.0 unx     5979 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/pdbs_measure_atom_dists.py
--rwxr-xr-x  2.0 unx     2091 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/pymol_color_by_conserv.py
--rwxr-xr-x  2.0 unx     3227 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/pymol_preview_generator.py
--rwxr-xr-x  2.0 unx      983 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/pymol_preview_install.py
--rwxr-xr-x  2.0 unx    10146 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/random_assignment_of_nucleotides.py
--rwxr-xr-x  2.0 unx     3941 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/renum_pdb_to_aln.py
--rwxr-xr-x  2.0 unx      141 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/reverse.py
--rwxr-xr-x  2.0 unx     2381 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_add_chain.py
--rwxr-xr-x  2.0 unx      637 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_align_find_core.py
--rwxr-xr-x  2.0 unx     2161 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_align_find_seq_in_alignment.py
--rwxr-xr-x  2.0 unx    13320 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_align_foldability.py
--rwxr-xr-x  2.0 unx     1327 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_align_get_ss_from_stk.py
--rwxr-xr-x  2.0 unx     1662 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_align_seq_to_alignment.py
--rwxr-xr-x  2.0 unx      712 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_align_strip_stk.py
--rwxr-xr-x  2.0 unx     9708 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_alignment_get_species.py
--rwxr-xr-x  2.0 unx     1919 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_alignment_process_id.py
--rwxr-xr-x  2.0 unx     2301 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_alignment_r2r.py
--rwxr-xr-x  2.0 unx     1318 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_cal_rmsd_trafl_plot.py
--rwxr-xr-x  2.0 unx     6805 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_calc_evo_rmsd.py
--rwxr-xr-x  2.0 unx     2453 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_calc_fenergy.py
--rwxr-xr-x  2.0 unx     2435 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_calc_hbonds.py
--rwxr-xr-x  2.0 unx     9995 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_calc_inf.py
--rwxr-xr-x  2.0 unx    13430 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_calc_rmsd.py
--rwxr-xr-x  2.0 unx     4186 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_calc_rmsd_all_vs_all.py
--rwxr-xr-x  2.0 unx    11464 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_calc_rmsd_biopython.py
--rwxr-xr-x  2.0 unx     3965 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_calc_rmsd_multi_targets.py
--rwxr-xr-x  2.0 unx     7157 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_calc_rmsd_trafl.py
--rwxr-xr-x  2.0 unx    21520 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_clanstix.py
--rwxr-xr-x  2.0 unx    48049 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_clarna_compare.py
--rwxr-xr-x  2.0 unx    45334 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_clarna_run.py
--rwxr-xr-x  2.0 unx      903 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_csv_sort.py
--rwxr-xr-x  2.0 unx      941 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_dG2kd.py
--rwxr-xr-x  2.0 unx     2435 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_dot2ct.py
--rwxr-xr-x  2.0 unx     5418 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_draw_edges.py
--rwxr-xr-x  2.0 unx     2793 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_ec2x.py
--rwxr-xr-x  2.0 unx    11893 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_filter.py
--rwxr-xr-x  2.0 unx      919 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_gromacs_ready.py
--rwxr-xr-x  2.0 unx     4055 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_helix_vis.py
--rwxr-xr-x  2.0 unx      999 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_kd2dG.py
--rwxr-xr-x  2.0 unx      705 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_kd2pkd.py
--rwxr-xr-x  2.0 unx     5927 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_md.py
--rwxr-xr-x  2.0 unx     1194 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_merge_dfs.py
--rwxr-xr-x  2.0 unx     8069 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_merge_two_dfs.py
--rwxr-xr-x  2.0 unx     3263 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_minimize.py
--rwxr-xr-x  2.0 unx     1169 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_clashscore.py
--rwxr-xr-x  2.0 unx    24580 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_collect.py
--rwxr-xr-x  2.0 unx     1216 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_dfire.py
--rwxr-xr-x  2.0 unx      908 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_eSCORE.py
--rwxr-xr-x  2.0 unx     2428 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_exp.py
--rwxr-xr-x  2.0 unx     2435 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_farfar2.py
--rwxr-xr-x  2.0 unx     5044 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_farna.py
--rwxr-xr-x  2.0 unx     1428 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_plot.py
--rwxr-xr-x  2.0 unx     1371 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_rasp.py
--rwxr-xr-x  2.0 unx      290 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_remote.sh
--rwxr-xr-x  2.0 unx      949 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_rna3dcnn.py
--rwxr-xr-x  2.0 unx     1707 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_rnakb.py
--rwxr-xr-x  2.0 unx      912 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_rsRNASP.py
--rwxr-xr-x  2.0 unx     4635 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_score.py
--rwxr-xr-x  2.0 unx     1546 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_simrna.py
--rwxr-xr-x  2.0 unx     2713 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_mq_wx_collect.py
--rwxr-xr-x  2.0 unx       64 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_ok
--rwxr-xr-x  2.0 unx     1369 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_ok_mail.py
--rwxr-xr-x  2.0 unx     1789 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_pairs2SimRNArestrs.py
--rwxr-xr-x  2.0 unx     1384 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_pairs_diff.py
--rwxr-xr-x  2.0 unx     6145 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_pdb_edit_occupancy_bfactor.py
--rwxr-xr-x  2.0 unx      415 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_pdb_merge_into_one.py
--rwxr-xr-x  2.0 unx     4905 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_pdb_merge_structure_with_fragments.py
--rwxr-xr-x  2.0 unx     1743 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_pdb_replace.py
--rwxr-xr-x  2.0 unx    41410 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_pdb_toolsx.py
--rwxr-xr-x  2.0 unx     1039 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_pk_simrna_to_one_line.py
--rwxr-xr-x  2.0 unx      886 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_pkd2dG.py
--rwxr-xr-x  2.0 unx      783 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_pkd2kd.py
--rwxr-xr-x  2.0 unx     2263 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_plot_boxplotlike.py
--rwxr-xr-x  2.0 unx     1610 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_plot_dendogram.py
--rwxr-xr-x  2.0 unx     1780 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_plot_density.py
--rwxr-xr-x  2.0 unx     2024 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_plot_heatmap.py
--rwxr-xr-x  2.0 unx     1946 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_plot_hist.py
--rwxr-xr-x  2.0 unx     1079 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_prediction_significance.py
--rwxr-xr-x  2.0 unx     7029 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_refinement.py
--rwxr-xr-x  2.0 unx     6070 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_rosetta_check_progress.py
--rwxr-xr-x  2.0 unx     5462 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_rosetta_cluster.py
--rwxr-xr-x  2.0 unx     1523 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_rosetta_extract_lowscore_decoys.py
--rwxr-xr-x  2.0 unx     3977 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_rosetta_head.py
--rwxr-xr-x  2.0 unx     4903 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_rosetta_min.py
--rwxr-xr-x  2.0 unx     1045 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_rosetta_n.py
--rwxr-xr-x  2.0 unx    10095 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_rosetta_run.py
--rwxr-xr-x  2.0 unx     3450 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_sali2dotbracket.py
--rwxr-xr-x  2.0 unx     2159 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_secondary_structure_prediction.py
--rwxr-xr-x  2.0 unx     1221 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_simrna_cluster.py
--rwxr-xr-x  2.0 unx     3636 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_simrna_extract.py
--rwxr-xr-x  2.0 unx     1265 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_simrna_lowest.py
--rwxr-xr-x  2.0 unx    12460 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_simrnaweb_download_job.py
--rwxr-xr-x  2.0 unx     1645 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_ss_get_bps.py
--rwxr-xr-x  2.0 unx      678 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_tools_which.py
--rwxr-xr-x  2.0 unx     1182 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_traj_n.py
--rwxr-xr-x  2.0 unx     1151 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_traj_nohetm.py
--rwxr-xr-x  2.0 unx     1798 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_traj_range.py
--rwxr-xr-x  2.0 unx     2625 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_workflow.py
--rwxr-xr-x  2.0 unx     6714 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/rna_x3dna.py
--rwxr-xr-x  2.0 unx     3945 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/select_seq_fromBLAStn_6outfm.py
--rwxr-xr-x  2.0 unx      489 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/translate.py
--rwxr-xr-x  2.0 unx    16446 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/triplexibility.py
--rwxr-xr-x  2.0 unx    13671 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/triplexibility2.py
--rwxr-xr-x  2.0 unx     1538 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/trx_mutgen.py
--rwxr-xr-x  2.0 unx     3282 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.data/scripts/trx_score.py
--rw-r--r--  2.0 unx        0 b- defN 22-Feb-21 22:11 tests/__init__.py
--rw-r--r--  2.0 unx    35142 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.dist-info/LICENSE
--rw-r--r--  2.0 unx    33691 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.dist-info/METADATA
--rw-r--r--  2.0 unx       92 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.dist-info/WHEEL
--rw-r--r--  2.0 unx       16 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.dist-info/top_level.txt
-?rw-rw-r--  2.0 unx    53375 b- defN 22-Feb-21 22:11 rna_tools-3.8.5.dist-info/RECORD
-509 files, 3831526 bytes uncompressed, 986140 bytes compressed:  74.3%
+Zip file size: 1079851 bytes, number of entries: 513
+-rw-r--r--  2.0 unx     1305 b- defN 22-Mar-07 11:12 rna_tools/BlastPDB.py
+-rw-r--r--  2.0 unx     3815 b- defN 22-Mar-07 11:12 rna_tools/RfamSearch.py
+-rw-r--r--  2.0 unx     3819 b- defN 22-Mar-07 11:12 rna_tools/SecondaryStructure.py
+-rw-r--r--  2.0 unx    27464 b- defN 22-Mar-07 11:12 rna_tools/Seq.py
+-rw-r--r--  2.0 unx       93 b- defN 22-Mar-07 11:12 rna_tools/__init__.py
+-rw-r--r--  2.0 unx      496 b- defN 22-Mar-07 11:12 rna_tools/_version.py
+-rw-r--r--  2.0 unx     2466 b- defN 22-Mar-07 11:12 rna_tools/rna_calc_fenergy.py
+-rw-r--r--  2.0 unx     2448 b- defN 22-Mar-07 11:12 rna_tools/rna_dot2ct.py
+-rw-r--r--  2.0 unx     1756 b- defN 22-Mar-07 11:12 rna_tools/rna_pdb_replace.py
+-rw-r--r--  2.0 unx    41423 b- defN 22-Mar-07 11:12 rna_tools/rna_pdb_toolsx.py
+-rw-r--r--  2.0 unx     2172 b- defN 22-Mar-07 11:12 rna_tools/rna_secondary_structure_prediction.py
+-rw-r--r--  2.0 unx    12473 b- defN 22-Mar-07 11:12 rna_tools/rna_simrnaweb_download_job.py
+-rw-r--r--  2.0 unx     3731 b- defN 22-Mar-07 11:12 rna_tools/rna_tools_config.py
+-rw-r--r--  2.0 unx    82236 b- defN 22-Mar-07 11:12 rna_tools/rna_tools_lib.py
+-rw-r--r--  2.0 unx     1342 b- defN 22-Mar-07 11:12 rna_tools/rna_tools_test_all.py
+-rw-r--r--  2.0 unx      709 b- defN 22-Mar-07 11:12 rna_tools/update_readme.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/__init__.py
+-rw-r--r--  2.0 unx     3080 b- defN 22-Mar-07 11:12 rna_tools/tools/ClashCalc/ClashCalc.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/ClashCalc/__init__.py
+-rw-r--r--  2.0 unx    38955 b- defN 22-Mar-07 11:12 rna_tools/tools/PyMOL4RNA/PyMOL4RNA.py
+-rw-r--r--  2.0 unx    15676 b- defN 22-Mar-07 11:12 rna_tools/tools/PyMOL4RNA/PyMOL4Spliceosome.py
+-rw-r--r--  2.0 unx     6836 b- defN 22-Mar-07 11:12 rna_tools/tools/PyMOL4RNA/Spliceosome_PyMOL.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/PyMOL4RNA/__init__.py
+-rw-r--r--  2.0 unx    15836 b- defN 22-Mar-07 11:12 rna_tools/tools/PyMOL4RNA/code_for_color_spl.py
+-rw-r--r--  2.0 unx     7898 b- defN 22-Mar-07 11:12 rna_tools/tools/PyMOL4RNA/code_for_spl.py
+-rw-r--r--  2.0 unx    11652 b- defN 22-Mar-07 11:12 rna_tools/tools/PyMOL4RNA/external_flatten_object.py
+-rw-r--r--  2.0 unx     5431 b- defN 22-Mar-07 11:12 rna_tools/tools/PyMOL4RNA/rna_draw_edges.py
+-rw-r--r--  2.0 unx     3208 b- defN 22-Mar-07 11:12 rna_tools/tools/PyMOL4RNA/rnapuzzles.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/PyMOL4RNA/libs/__init__.py
+-rw-r--r--  2.0 unx     7612 b- defN 22-Mar-07 11:12 rna_tools/tools/PyMOL4RNA/libs/get_raw_distances.py
+-rw-r--r--  2.0 unx    15707 b- defN 22-Mar-07 11:12 rna_tools/tools/PyMOL4RNA/libs/show_contacts.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/clanstix/__init__.py
+-rw-r--r--  2.0 unx    21529 b- defN 22-Mar-07 11:12 rna_tools/tools/clanstix/rna_clanstix.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_app/__init__.py
+-rw-r--r--  2.0 unx     5685 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_app/clarna_app.py
+-rw-r--r--  2.0 unx      603 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_app/test.py
+-rw-r--r--  2.0 unx    14950 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/PDBSeq.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/__init__.py
+-rw-r--r--  2.0 unx    48062 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/rna_clarna_compare.py
+-rw-r--r--  2.0 unx    45347 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/rna_clarna_run.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/__init__.py
+-rw-r--r--  2.0 unx    13864 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/aggregate-doublets-dict.py
+-rw-r--r--  2.0 unx     2945 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/align-structures.py
+-rw-r--r--  2.0 unx   782660 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/bp_limits.py
+-rw-r--r--  2.0 unx      803 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/check-descriptions-dict.py
+-rw-r--r--  2.0 unx       33 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/cl_settings.py
+-rw-r--r--  2.0 unx     1911 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/cl_settings_global.py
+-rw-r--r--  2.0 unx    46848 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/clarna.py
+-rw-r--r--  2.0 unx    10101 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/cluster-doublets.py
+-rw-r--r--  2.0 unx    28593 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/combine-contact-graphs.py
+-rw-r--r--  2.0 unx     4658 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/compute-bp-limits.py
+-rw-r--r--  2.0 unx    63976 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/compute-classifier.py
+-rw-r--r--  2.0 unx     2609 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/cross-validation-groups.py
+-rw-r--r--  2.0 unx     6898 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/cross-validation-groups2.py
+-rw-r--r--  2.0 unx    43738 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/distances.py
+-rw-r--r--  2.0 unx    14757 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/doublet-params.py
+-rw-r--r--  2.0 unx     5309 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/draw-contact-graph.py
+-rw-r--r--  2.0 unx     5111 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/eval-graphs.py
+-rw-r--r--  2.0 unx     4267 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/export-ref-classes.py
+-rw-r--r--  2.0 unx    15896 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/extract-contacts.py
+-rw-r--r--  2.0 unx     2215 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/extract-doublets-from-graph.py
+-rw-r--r--  2.0 unx    98501 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/fabfile.py
+-rw-r--r--  2.0 unx    23171 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/fix-reference-groups.py
+-rw-r--r--  2.0 unx     9592 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/gen-other-groups.py
+-rw-r--r--  2.0 unx    12132 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/gen-pdb.py
+-rw-r--r--  2.0 unx     3245 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/graph-to-secondary-structure.py
+-rw-r--r--  2.0 unx     3193 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/json-tool.py
+-rw-r--r--  2.0 unx     4098 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/make-residue-dict.py
+-rw-r--r--  2.0 unx     3380 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/make-reverse-descriptions-dict.py
+-rw-r--r--  2.0 unx     2452 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/merge-dicts.py
+-rw-r--r--  2.0 unx    11530 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/moderna-stackings.py
+-rw-r--r--  2.0 unx     3656 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/morph-doublet.py
+-rw-r--r--  2.0 unx     4083 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/normalize_pdb.py
+-rw-r--r--  2.0 unx    12645 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/progressbar.py
+-rw-r--r--  2.0 unx     2300 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/show-doublet-info.py
+-rw-r--r--  2.0 unx     9025 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/structure_ciach.py
+-rw-r--r--  2.0 unx    21974 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/supp-files.py
+-rw-r--r--  2.0 unx     7801 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/testcase.py
+-rw-r--r--  2.0 unx     1867 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/translate-doublets-ids.py
+-rw-r--r--  2.0 unx    48150 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/utils.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/get_pairs/__init__.py
+-rw-r--r--  2.0 unx     2616 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/get_pairs/get_pairs.py
+-rw-r--r--  2.0 unx      157 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/ClaRNAlib/get_pairs/test.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/lib/__init__.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/lib/rna_pdb_tools/__init__.py
+-rw-r--r--  2.0 unx    38027 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/lib/rna_pdb_tools/pdb_parser_lib.py
+-rw-r--r--  2.0 unx     6354 b- defN 22-Mar-07 11:12 rna_tools/tools/clarna_play/lib/rna_pdb_tools/rna-pdb-tools.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/cluster_load/__init__.py
+-rw-r--r--  2.0 unx     2943 b- defN 22-Mar-07 11:12 rna_tools/tools/cluster_load/cluster_load.py
+-rw-r--r--  2.0 unx      319 b- defN 22-Mar-07 11:12 rna_tools/tools/cluster_load/cluster_mini_free.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/diffpdb/__init__.py
+-rw-r--r--  2.0 unx     4645 b- defN 22-Mar-07 11:12 rna_tools/tools/diffpdb/diffpdb.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/extra_functions/__init__.py
+-rw-r--r--  2.0 unx      506 b- defN 22-Mar-07 11:12 rna_tools/tools/extra_functions/gen_numbering.py
+-rw-r--r--  2.0 unx     3744 b- defN 22-Mar-07 11:12 rna_tools/tools/extra_functions/select_fragment.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/md/__init__.py
+-rw-r--r--  2.0 unx     1405 b- defN 22-Mar-07 11:12 rna_tools/tools/md/rna_get_energy.py
+-rw-r--r--  2.0 unx      932 b- defN 22-Mar-07 11:12 rna_tools/tools/md/rna_gromacs_ready.py
+-rw-r--r--  2.0 unx     5940 b- defN 22-Mar-07 11:12 rna_tools/tools/md/rna_md.py
+-rw-r--r--  2.0 unx     3276 b- defN 22-Mar-07 11:12 rna_tools/tools/md/rna_minimize.py
+-rw-r--r--  2.0 unx     1195 b- defN 22-Mar-07 11:12 rna_tools/tools/md/rna_traj_n.py
+-rw-r--r--  2.0 unx     1164 b- defN 22-Mar-07 11:12 rna_tools/tools/md/rna_traj_nohetm.py
+-rw-r--r--  2.0 unx     1811 b- defN 22-Mar-07 11:12 rna_tools/tools/md/rna_traj_range.py
+-rw-r--r--  2.0 unx    20335 b- defN 22-Mar-07 11:12 rna_tools/tools/md/rnakb_utils.py
+-rw-r--r--  2.0 unx     1717 b- defN 22-Mar-07 11:12 rna_tools/tools/md/test.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/__init__.py
+-rw-r--r--  2.0 unx    19630 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/CheckPdb.py
+-rw-r--r--  2.0 unx     4810 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/Constants.py
+-rw-r--r--  2.0 unx     1089 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/FragmentInsertion.py
+-rw-r--r--  2.0 unx     6839 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/Helix.py
+-rw-r--r--  2.0 unx    10220 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/ModelingRecipe.py
+-rw-r--r--  2.0 unx    25493 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/ModernaFragment.py
+-rw-r--r--  2.0 unx    11822 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/ModernaStructure.py
+-rw-r--r--  2.0 unx     3043 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/ModernaSuperimposer.py
+-rw-r--r--  2.0 unx     9166 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/PdbConverter.py
+-rw-r--r--  2.0 unx     9262 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/RNAChain.py
+-rw-r--r--  2.0 unx    17244 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/RNAModel.py
+-rw-r--r--  2.0 unx    12712 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/RNAResidue.py
+-rw-r--r--  2.0 unx     5992 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/Renumerator.py
+-rw-r--r--  2.0 unx     3052 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/Renumerator_new.py
+-rw-r--r--  2.0 unx     1837 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/Template.py
+-rw-r--r--  2.0 unx      564 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/__init__.py
+-rw-r--r--  2.0 unx     7626 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/_version.py
+-rw-r--r--  2.0 unx    44552 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/commands.py
+-rw-r--r--  2.0 unx     4642 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/moderna.py
+-rw-r--r--  2.0 unx     5696 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/BasePairCalculator.py
+-rw-r--r--  2.0 unx     7527 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/BaseRecognizer.py
+-rw-r--r--  2.0 unx     4459 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/ChainConnectivity.py
+-rw-r--r--  2.0 unx     7680 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/ClashRecognizer.py
+-rw-r--r--  2.0 unx     5299 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/Constants.py
+-rw-r--r--  2.0 unx     3157 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/GeometryAnalyzer.py
+-rw-r--r--  2.0 unx     9854 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/GeometryParameters.py
+-rw-r--r--  2.0 unx    13260 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/GeometryStatistics.py
+-rw-r--r--  2.0 unx     6691 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/HBondCalculator.py
+-rw-r--r--  2.0 unx     8995 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/MolGraphParser.py
+-rw-r--r--  2.0 unx     8941 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/MolParameters.py
+-rw-r--r--  2.0 unx    28177 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/MolTopologies.py
+-rw-r--r--  2.0 unx    23655 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/MolecularGraph.py
+-rw-r--r--  2.0 unx     3526 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/PuckerCalculator.py
+-rw-r--r--  2.0 unx     3178 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/RNASuites.py
+-rw-r--r--  2.0 unx     8097 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/StackingCalculator.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/analyze/__init__.py
+-rw-r--r--  2.0 unx     7625 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/builder/BackboneBuilder.py
+-rw-r--r--  2.0 unx     1026 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/builder/ChiRotator.py
+-rw-r--r--  2.0 unx      401 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/builder/Constants.py
+-rw-r--r--  2.0 unx     1974 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/builder/CoordBuilder.py
+-rw-r--r--  2.0 unx     5086 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/builder/FCCDLoopCloser.py
+-rw-r--r--  2.0 unx    12133 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/builder/PhosphateBuilder.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/builder/__init__.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/examples/__init__.py
+-rw-r--r--  2.0 unx     6477 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/examples/usage_examples.py
+-rw-r--r--  2.0 unx     8489 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/fragment_library/LIR.py
+-rw-r--r--  2.0 unx    11310 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/fragment_library/LIRdb.py
+-rw-r--r--  2.0 unx    21344 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/fragment_library/SearchLIR.py
+-rw-r--r--  2.0 unx     4552 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/fragment_library/StructureLibrary.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/fragment_library/__init__.py
+-rw-r--r--  2.0 unx     3674 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/isosteric/IsostericModelling.py
+-rw-r--r--  2.0 unx    10949 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/isosteric/Isostericity.py
+-rw-r--r--  2.0 unx     3035 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/isosteric/IsostericityMatrices.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/isosteric/__init__.py
+-rw-r--r--  2.0 unx     3089 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/modifications/BaseExchanger.py
+-rw-r--r--  2.0 unx     4052 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/modifications/ModificationAdder.py
+-rw-r--r--  2.0 unx     4170 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/modifications/ModificationRemover.py
+-rw-r--r--  2.0 unx     1468 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/modifications/ModifyResidue.py
+-rw-r--r--  2.0 unx     4826 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/modifications/ResidueEditor.py
+-rw-r--r--  2.0 unx      494 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/modifications/__init__.py
+-rw-r--r--  2.0 unx     7245 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/sequence/AlignmentMatcher.py
+-rw-r--r--  2.0 unx     6090 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/sequence/ModernaAlphabet.py
+-rw-r--r--  2.0 unx     6619 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/sequence/ModernaSequence.py
+-rw-r--r--  2.0 unx    17350 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/sequence/RNAAlignment.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/sequence/__init__.py
+-rw-r--r--  2.0 unx     3546 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/sequence/commands.py
+-rw-r--r--  2.0 unx     2271 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/util/Errors.py
+-rw-r--r--  2.0 unx     3075 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/util/LogFile.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/util/__init__.py
+-rw-r--r--  2.0 unx     2210 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/util/decorators.py
+-rw-r--r--  2.0 unx     5837 b- defN 22-Mar-07 11:12 rna_tools/tools/mini_moderna3/moderna/util/validators.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/misc/__init__.py
+-rw-r--r--  2.0 unx      154 b- defN 22-Mar-07 11:12 rna_tools/tools/misc/reverse.py
+-rw-r--r--  2.0 unx     2394 b- defN 22-Mar-07 11:12 rna_tools/tools/misc/rna_add_chain.py
+-rw-r--r--  2.0 unx      916 b- defN 22-Mar-07 11:12 rna_tools/tools/misc/rna_csv_sort.py
+-rw-r--r--  2.0 unx     1207 b- defN 22-Mar-07 11:12 rna_tools/tools/misc/rna_merge_dfs.py
+-rw-r--r--  2.0 unx     8082 b- defN 22-Mar-07 11:12 rna_tools/tools/misc/rna_merge_two_dfs.py
+-rwxr-xr-x  2.0 unx       64 b- defN 22-Mar-07 11:12 rna_tools/tools/misc/rna_ok
+-rw-r--r--  2.0 unx     1378 b- defN 22-Mar-07 11:12 rna_tools/tools/misc/rna_ok_mail.py
+-rw-r--r--  2.0 unx      691 b- defN 22-Mar-07 11:12 rna_tools/tools/misc/rna_tools_which.py
+-rw-r--r--  2.0 unx      502 b- defN 22-Mar-07 11:12 rna_tools/tools/misc/translate.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/__init__.py
+-rw-r--r--  2.0 unx    24593 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/rna_mq_collect.py
+-rw-r--r--  2.0 unx     2441 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/rna_mq_exp.py
+-rw-r--r--  2.0 unx     1441 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/rna_mq_plot.py
+-rwxr-xr-x  2.0 unx      290 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/rna_mq_remote.sh
+-rw-r--r--  2.0 unx     4648 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/rna_mq_score.py
+-rw-r--r--  2.0 unx     2726 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/rna_mq_wx_collect.py
+-rw-r--r--  2.0 unx     4458 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/AnalyzeGeometry/AnalyzeGeometry.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/AnalyzeGeometry/__init__.py
+-rw-r--r--  2.0 unx     3050 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/ClashScore/ClashScore.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/ClashScore/__init__.py
+-rw-r--r--  2.0 unx     1182 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/ClashScore/rna_mq_clashscore.py
+-rw-r--r--  2.0 unx     2735 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/Dfire/Dfire.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/Dfire/__init__.py
+-rw-r--r--  2.0 unx     1229 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/Dfire/rna_mq_dfire.py
+-rw-r--r--  2.0 unx     9627 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/FARFAR2/FARFAR2.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/FARFAR2/__init__.py
+-rw-r--r--  2.0 unx     2448 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/FARFAR2/rna_mq_farfar2.py
+-rw-r--r--  2.0 unx       19 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/FARFAR2/test/__init__.py
+-rw-r--r--  2.0 unx     8873 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/FARFAR2/test/test.py
+-rw-r--r--  2.0 unx     7252 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/FARNA/FARNA.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/FARNA/__init__.py
+-rw-r--r--  2.0 unx     5057 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/FARNA/rna_mq_farna.py
+-rw-r--r--  2.0 unx       19 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/FARNA/test_data/__init__.py
+-rw-r--r--  2.0 unx     8873 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/FARNA/test_data/test.py
+-rw-r--r--  2.0 unx     5902 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/QRNAS/QRNAS.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/QRNAS/__init__.py
+-rw-r--r--  2.0 unx       19 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/QRNAS/test/__init__.py
+-rw-r--r--  2.0 unx     1699 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/QRNAS/test/test.py
+-rw-r--r--  2.0 unx     5234 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RASP/RASP.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RASP/__init__.py
+-rw-r--r--  2.0 unx     1384 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RASP/rna_mq_rasp.py
+-rw-r--r--  2.0 unx      382 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RASP/test.py
+-rw-r--r--  2.0 unx       19 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RASP/test/__init__.py
+-rw-r--r--  2.0 unx     6408 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RASP/test/test.py
+-rw-r--r--  2.0 unx     4639 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RNA3DCNN/RNA3DCNN.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RNA3DCNN/__init__.py
+-rw-r--r--  2.0 unx      962 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RNA3DCNN/rna_mq_rna3dcnn.py
+-rw-r--r--  2.0 unx    14264 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RNAkb/RNAkb.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RNAkb/__init__.py
+-rw-r--r--  2.0 unx     1720 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RNAkb/rna_mq_rnakb.py
+-rw-r--r--  2.0 unx     2783 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RNAscore/RNAscore.py
+-rw-r--r--  2.0 unx        1 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/RNAscore/__init__.py
+-rw-r--r--  2.0 unx     4953 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/SimRNA/SimRNA.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/SimRNA/__init__.py
+-rw-r--r--  2.0 unx     1559 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/SimRNA/rna_mq_simrna.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/eSCORE/__init__.py
+-rw-r--r--  2.0 unx     2752 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/eSCORE/eSCORE.py
+-rw-r--r--  2.0 unx      921 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/eSCORE/rna_mq_eSCORE.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/lib/__init__.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/lib/shellgraphics/__init__.py
+-rw-r--r--  2.0 unx    10890 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/lib/shellgraphics/shellgraphics.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/lib/timex/__init__.py
+-rw-r--r--  2.0 unx     1341 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/lib/timex/timex.py
+-rw-r--r--  2.0 unx     2582 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/lib/wrappers/SubprocessUtils.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/lib/wrappers/__init__.py
+-rw-r--r--  2.0 unx     8823 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/lib/wrappers/base_wrappers.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/lib/wrappers/test/__init__.py
+-rw-r--r--  2.0 unx     1226 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/lib/wrappers/test/test.py
+-rw-r--r--  2.0 unx        1 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/rsRNASP/__init__.py
+-rw-r--r--  2.0 unx      925 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/rsRNASP/rna_mq_rsRNASP.py
+-rw-r--r--  2.0 unx     2718 b- defN 22-Mar-07 11:12 rna_tools/tools/mq/rsRNASP/rsRNASP.py
+-rw-r--r--  2.0 unx    12536 b- defN 22-Mar-07 11:12 rna_tools/tools/pdb_formatix/PDBFile.py
+-rw-r--r--  2.0 unx     1964 b- defN 22-Mar-07 11:12 rna_tools/tools/pdb_formatix/RosettaUtils.py
+-rw-r--r--  2.0 unx     1492 b- defN 22-Mar-07 11:12 rna_tools/tools/pdb_formatix/SingleLineUtils.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/pdb_formatix/__init__.py
+-rw-r--r--  2.0 unx     2612 b- defN 22-Mar-07 11:12 rna_tools/tools/pdb_formatix/rebuilder.py
+-rw-r--r--  2.0 unx      623 b- defN 22-Mar-07 11:12 rna_tools/tools/pdb_formatix/test-quick-and-dirty.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/pdb_formatix/test/__init__.py
+-rw-r--r--  2.0 unx     5106 b- defN 22-Mar-07 11:12 rna_tools/tools/pdb_formatix/test/test.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/pdbs_measure_atom_dists/__init__.py
+-rw-r--r--  2.0 unx     5992 b- defN 22-Mar-07 11:12 rna_tools/tools/pdbs_measure_atom_dists/pdbs_measure_atom_dists.py
+-rw-r--r--  2.0 unx        1 b- defN 22-Mar-07 11:12 rna_tools/tools/plotting/__init__.py
+-rw-r--r--  2.0 unx     2276 b- defN 22-Mar-07 11:12 rna_tools/tools/plotting/rna_plot_boxplotlike.py
+-rw-r--r--  2.0 unx     1623 b- defN 22-Mar-07 11:12 rna_tools/tools/plotting/rna_plot_dendogram.py
+-rw-r--r--  2.0 unx     1793 b- defN 22-Mar-07 11:12 rna_tools/tools/plotting/rna_plot_density.py
+-rw-r--r--  2.0 unx     2037 b- defN 22-Mar-07 11:12 rna_tools/tools/plotting/rna_plot_heatmap.py
+-rw-r--r--  2.0 unx     1959 b- defN 22-Mar-07 11:12 rna_tools/tools/plotting/rna_plot_hist.py
+-rwxr-xr-x  2.0 unx     2104 b- defN 22-Mar-07 11:12 rna_tools/tools/pymol_color_by_conserv/pymol_color_by_conserv.py
+-rw-r--r--  2.0 unx        1 b- defN 22-Mar-07 11:12 rna_tools/tools/pymol_drawing/__init__.py
+-rw-r--r--  2.0 unx     3539 b- defN 22-Mar-07 11:12 rna_tools/tools/pymol_drawing/pymol_drawing.py
+-rwxr-xr-x  2.0 unx     3368 b- defN 22-Mar-07 11:12 rna_tools/tools/pymol_preview_generator/pymol_preview_generator.py
+-rwxr-xr-x  2.0 unx      996 b- defN 22-Mar-07 11:12 rna_tools/tools/pymol_preview_generator/pymol_preview_install.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/renum_pdb_to_aln/__init__.py
+-rw-r--r--  2.0 unx     3954 b- defN 22-Mar-07 11:12 rna_tools/tools/renum_pdb_to_aln/renum_pdb_to_aln.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/__init__.py
+-rw-r--r--  2.0 unx    10159 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/random_assignment_of_nucleotides.py
+-rw-r--r--  2.0 unx     2877 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/rna_align_distance_to_seq.py
+-rw-r--r--  2.0 unx      650 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/rna_align_find_core.py
+-rw-r--r--  2.0 unx     2174 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/rna_align_find_seq_in_alignment.py
+-rw-r--r--  2.0 unx    13333 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/rna_align_foldability.py
+-rw-r--r--  2.0 unx      953 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/rna_align_get_ss_from_fasta.py
+-rw-r--r--  2.0 unx     1340 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/rna_align_get_ss_from_stk.py
+-rw-r--r--  2.0 unx     1675 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/rna_align_seq_to_alignment.py
+-rw-r--r--  2.0 unx    45244 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/rna_alignment.py
+-rw-r--r--  2.0 unx     1786 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/rna_alignment_r2r.py
+-rw-r--r--  2.0 unx     2361 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/test.py
+-rw-r--r--  2.0 unx     6910 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/test_rna_alignment.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/utils/__init__.py
+-rw-r--r--  2.0 unx     2336 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/utils/mfa-get-seq-for-species.py
+-rw-r--r--  2.0 unx      725 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/utils/rna_align_strip_stk.py
+-rw-r--r--  2.0 unx     8104 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/utils/rna_alignment_calc_energy.py
+-rw-r--r--  2.0 unx     9721 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/utils/rna_alignment_get_species.py
+-rw-r--r--  2.0 unx     4358 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/utils/rna_alignment_merge_by_species.py
+-rw-r--r--  2.0 unx     1932 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/utils/rna_alignment_process_id.py
+-rw-r--r--  2.0 unx     2314 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/utils/rna_alignment_r2r.py
+-rw-r--r--  2.0 unx      959 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_alignment/utils/rna_alignment_sto_remove_uniq_ids.py
+-rwxr-xr-x  2.0 unx      954 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_binding_affinity/rna_dG2kd.py
+-rwxr-xr-x  2.0 unx     1012 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_binding_affinity/rna_kd2dG.py
+-rwxr-xr-x  2.0 unx      718 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_binding_affinity/rna_kd2pkd.py
+-rwxr-xr-x  2.0 unx      899 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_binding_affinity/rna_pkd2dG.py
+-rwxr-xr-x  2.0 unx      796 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_binding_affinity/rna_pkd2kd.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_bp/__init__.py
+-rw-r--r--  2.0 unx     7281 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_bp/base_pair.py
+-rw-r--r--  2.0 unx     2737 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_evo_rmsd/RNAalignment.py
+-rw-r--r--  2.0 unx     5639 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_evo_rmsd/RNAmodel.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_evo_rmsd/__init__.py
+-rw-r--r--  2.0 unx     6818 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_evo_rmsd/rna_calc_evo_rmsd.py
+-rwxr-xr-x  2.0 unx     2448 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_hbonds/rna_calc_hbonds.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_inf/__init__.py
+-rw-r--r--  2.0 unx     5638 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_inf/rna_calc_dinf.py
+-rw-r--r--  2.0 unx    10008 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_inf/rna_calc_inf.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_rmsd/__init__.py
+-rw-r--r--  2.0 unx    13443 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_rmsd/rna_calc_rmsd.py
+-rw-r--r--  2.0 unx     4199 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_rmsd/rna_calc_rmsd_all_vs_all.py
+-rw-r--r--  2.0 unx    11477 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_rmsd/rna_calc_rmsd_biopython.py
+-rw-r--r--  2.0 unx     3978 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_rmsd/rna_calc_rmsd_multi_targets.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_rmsd/lib/__init__.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_rmsd/lib/rmsd/__init__.py
+-rw-r--r--  2.0 unx     6911 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_rmsd/lib/rmsd/calculate_rmsd.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_rmsd_trafl/__init__.py
+-rw-r--r--  2.0 unx     1331 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_rmsd_trafl/rna_cal_rmsd_trafl_plot.py
+-rw-r--r--  2.0 unx     7170 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_calc_rmsd_trafl/rna_calc_rmsd_trafl.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_convert_pseudoknot_formats/__init__.py
+-rw-r--r--  2.0 unx     1052 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_convert_pseudoknot_formats/rna_pk_simrna_to_one_line.py
+-rw-r--r--  2.0 unx     1334 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_convert_pseudoknot_formats/rna_ss_pk_to_simrna.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/__init__.py
+-rw-r--r--  2.0 unx      637 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/pymol_dists.py
+-rw-r--r--  2.0 unx      878 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_csv2pairs.py
+-rw-r--r--  2.0 unx    12319 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_dca_mapping.py
+-rw-r--r--  2.0 unx     1166 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_dca_select_interactions.py
+-rw-r--r--  2.0 unx     2806 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_ec2x.py
+-rw-r--r--  2.0 unx    11906 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_filter.py
+-rw-r--r--  2.0 unx    11051 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_get_distances.py
+-rw-r--r--  2.0 unx     9035 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_get_dists.py
+-rw-r--r--  2.0 unx     1802 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_pairs2SimRNArestrs.py
+-rw-r--r--  2.0 unx     1524 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_pairs2rfilter.py
+-rw-r--r--  2.0 unx     1397 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_pairs_diff.py
+-rw-r--r--  2.0 unx     1373 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_restrs2SimRNArestrs.py
+-rw-r--r--  2.0 unx     1299 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_restrs2mqapRNA.py
+-rw-r--r--  2.0 unx     1658 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/rna_ss_get_bps.py
+-rw-r--r--  2.0 unx      884 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/simrna_get_restraints.py
+-rw-r--r--  2.0 unx      990 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_filter/test_rna_filter.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_helix_vis/__init__.py
+-rw-r--r--  2.0 unx     4068 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_helix_vis/rna_helix_vis.py
+-rwxr-xr-x  2.0 unx      428 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_multimodels/rna_pdb_merge_into_one.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_pdb_edit_occupancy_bfactor/__init__.py
+-rw-r--r--  2.0 unx     6158 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_pdb_edit_occupancy_bfactor/rna_pdb_edit_occupancy_bfactor.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_pdb_merge_structure_with_fragments/__init__.py
+-rw-r--r--  2.0 unx     4918 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_pdb_merge_structure_with_fragments/rna_pdb_merge_structure_with_fragments.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_prediction_significance/__init__.py
+-rw-r--r--  2.0 unx     1088 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_prediction_significance/rna_prediction_significance.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_refinement/__init__.py
+-rw-r--r--  2.0 unx     7043 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_refinement/rna_refinement.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_rosetta/__init__.py
+-rw-r--r--  2.0 unx     6083 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_rosetta/rna_rosetta_check_progress.py
+-rw-r--r--  2.0 unx     5475 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_rosetta/rna_rosetta_cluster.py
+-rw-r--r--  2.0 unx     1536 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_rosetta/rna_rosetta_easy_cat.py
+-rw-r--r--  2.0 unx     1536 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_rosetta/rna_rosetta_extract_lowscore_decoys.py
+-rw-r--r--  2.0 unx     3990 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_rosetta/rna_rosetta_head.py
+-rw-r--r--  2.0 unx     4916 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_rosetta/rna_rosetta_min.py
+-rw-r--r--  2.0 unx     1058 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_rosetta/rna_rosetta_n.py
+-rw-r--r--  2.0 unx    10108 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_rosetta/rna_rosetta_run.py
+-rw-r--r--  2.0 unx      649 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_rosetta/test.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_sali2dotbracket/__init__.py
+-rw-r--r--  2.0 unx     3463 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_sali2dotbracket/rna_sali2dotbracket.py
+-rw-r--r--  2.0 unx     3958 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_seq_search_BLASTn_outfmt-6/select_seq_fromBLAStn_6outfm.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_x3dna/__init__.py
+-rw-r--r--  2.0 unx     6727 b- defN 22-Mar-07 11:12 rna_tools/tools/rna_x3dna/rna_x3dna.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/simrna_trajectory/__init__.py
+-rw-r--r--  2.0 unx     1235 b- defN 22-Mar-07 11:12 rna_tools/tools/simrna_trajectory/rna_simrna_cluster.py
+-rw-r--r--  2.0 unx      848 b- defN 22-Mar-07 11:12 rna_tools/tools/simrna_trajectory/rna_simrna_energy_plot.py
+-rw-r--r--  2.0 unx     3649 b- defN 22-Mar-07 11:12 rna_tools/tools/simrna_trajectory/rna_simrna_extract.py
+-rw-r--r--  2.0 unx     1278 b- defN 22-Mar-07 11:12 rna_tools/tools/simrna_trajectory/rna_simrna_lowest.py
+-rw-r--r--  2.0 unx      723 b- defN 22-Mar-07 11:12 rna_tools/tools/simrna_trajectory/rna_simrna_subset.py
+-rw-r--r--  2.0 unx    10181 b- defN 22-Mar-07 11:12 rna_tools/tools/simrna_trajectory/simrna_trajectory.py
+-rw-r--r--  2.0 unx      406 b- defN 22-Mar-07 11:12 rna_tools/tools/simrna_trajectory/test_rmsd.py
+-rw-r--r--  2.0 unx      232 b- defN 22-Mar-07 11:12 rna_tools/tools/simrna_trajectory/test_rna_simrna.py
+-rwxr-xr-x  2.0 unx    16459 b- defN 22-Mar-07 11:12 rna_tools/tools/triplexibility/triplexibility.py
+-rwxr-xr-x  2.0 unx    13684 b- defN 22-Mar-07 11:12 rna_tools/tools/triplexibility/triplexibility2.py
+-rwxr-xr-x  2.0 unx     1671 b- defN 22-Mar-07 11:12 rna_tools/tools/triplexibility/trx_mutagen_allvsall.py
+-rwxr-xr-x  2.0 unx     1557 b- defN 22-Mar-07 11:12 rna_tools/tools/triplexibility/trx_mutgen.py
+-rwxr-xr-x  2.0 unx     3334 b- defN 22-Mar-07 11:12 rna_tools/tools/triplexibility/trx_score.py
+-rwxr-xr-x  2.0 unx     3288 b- defN 22-Mar-07 11:12 rna_tools/tools/triplexibility/trx_score_allvsall.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/__init__.py
+-rw-r--r--  2.0 unx     2640 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/checker.py
+-rw-r--r--  2.0 unx     4553 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/cleanup.py
+-rw-r--r--  2.0 unx    11839 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/daemon.py
+-rw-r--r--  2.0 unx      246 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/manage.py
+-rw-r--r--  2.0 unx     2712 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/sendmail.py
+-rw-r--r--  2.0 unx      632 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/wsgi.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/app/__init__.py
+-rw-r--r--  2.0 unx      508 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/app/admin.py
+-rw-r--r--  2.0 unx     8356 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/app/models.py
+-rw-r--r--  2.0 unx    20146 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/app/views.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/web/__init__.py
+-rw-r--r--  2.0 unx     6078 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/web/settings.py
+-rw-r--r--  2.0 unx     1943 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/web/urls.py
+-rw-r--r--  2.0 unx     1410 b- defN 22-Mar-07 11:12 rna_tools/tools/webserver-engine/web/wsgi.py
+-rwxr-xr-x  2.0 unx     2638 b- defN 22-Mar-07 11:12 rna_tools/tools/workflow/rna_workflow.py
+-rwxr-xr-x  2.0 unx    46835 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/clarna.py
+-rwxr-xr-x  2.0 unx     5672 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/clarna_app.py
+-rwxr-xr-x  2.0 unx     4636 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/diffpdb.py
+-rwxr-xr-x  2.0 unx     5979 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/pdbs_measure_atom_dists.py
+-rwxr-xr-x  2.0 unx     2091 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/pymol_color_by_conserv.py
+-rwxr-xr-x  2.0 unx     3354 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/pymol_preview_generator.py
+-rwxr-xr-x  2.0 unx      983 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/pymol_preview_install.py
+-rwxr-xr-x  2.0 unx    10146 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/random_assignment_of_nucleotides.py
+-rwxr-xr-x  2.0 unx     3941 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/renum_pdb_to_aln.py
+-rwxr-xr-x  2.0 unx      141 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/reverse.py
+-rwxr-xr-x  2.0 unx     2381 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_add_chain.py
+-rwxr-xr-x  2.0 unx      637 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_align_find_core.py
+-rwxr-xr-x  2.0 unx     2161 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_align_find_seq_in_alignment.py
+-rwxr-xr-x  2.0 unx    13320 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_align_foldability.py
+-rwxr-xr-x  2.0 unx     1327 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_align_get_ss_from_stk.py
+-rwxr-xr-x  2.0 unx     1662 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_align_seq_to_alignment.py
+-rwxr-xr-x  2.0 unx      712 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_align_strip_stk.py
+-rwxr-xr-x  2.0 unx     9708 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_alignment_get_species.py
+-rwxr-xr-x  2.0 unx     1919 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_alignment_process_id.py
+-rwxr-xr-x  2.0 unx     2301 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_alignment_r2r.py
+-rwxr-xr-x  2.0 unx     1318 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_cal_rmsd_trafl_plot.py
+-rwxr-xr-x  2.0 unx     6805 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_calc_evo_rmsd.py
+-rwxr-xr-x  2.0 unx     2453 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_calc_fenergy.py
+-rwxr-xr-x  2.0 unx     2435 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_calc_hbonds.py
+-rwxr-xr-x  2.0 unx     9995 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_calc_inf.py
+-rwxr-xr-x  2.0 unx    13430 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_calc_rmsd.py
+-rwxr-xr-x  2.0 unx     4186 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_calc_rmsd_all_vs_all.py
+-rwxr-xr-x  2.0 unx    11464 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_calc_rmsd_biopython.py
+-rwxr-xr-x  2.0 unx     3965 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_calc_rmsd_multi_targets.py
+-rwxr-xr-x  2.0 unx     7157 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_calc_rmsd_trafl.py
+-rwxr-xr-x  2.0 unx    21520 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_clanstix.py
+-rwxr-xr-x  2.0 unx    48049 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_clarna_compare.py
+-rwxr-xr-x  2.0 unx    45334 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_clarna_run.py
+-rwxr-xr-x  2.0 unx      903 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_csv_sort.py
+-rwxr-xr-x  2.0 unx      941 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_dG2kd.py
+-rwxr-xr-x  2.0 unx     2435 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_dot2ct.py
+-rwxr-xr-x  2.0 unx     5418 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_draw_edges.py
+-rwxr-xr-x  2.0 unx     2793 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_ec2x.py
+-rwxr-xr-x  2.0 unx    11893 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_filter.py
+-rwxr-xr-x  2.0 unx      919 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_gromacs_ready.py
+-rwxr-xr-x  2.0 unx     4055 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_helix_vis.py
+-rwxr-xr-x  2.0 unx      999 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_kd2dG.py
+-rwxr-xr-x  2.0 unx      705 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_kd2pkd.py
+-rwxr-xr-x  2.0 unx     5927 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_md.py
+-rwxr-xr-x  2.0 unx     1194 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_merge_dfs.py
+-rwxr-xr-x  2.0 unx     8069 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_merge_two_dfs.py
+-rwxr-xr-x  2.0 unx     3263 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_minimize.py
+-rwxr-xr-x  2.0 unx     1169 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_clashscore.py
+-rwxr-xr-x  2.0 unx    24580 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_collect.py
+-rwxr-xr-x  2.0 unx     1216 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_dfire.py
+-rwxr-xr-x  2.0 unx      908 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_eSCORE.py
+-rwxr-xr-x  2.0 unx     2428 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_exp.py
+-rwxr-xr-x  2.0 unx     2435 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_farfar2.py
+-rwxr-xr-x  2.0 unx     5044 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_farna.py
+-rwxr-xr-x  2.0 unx     1428 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_plot.py
+-rwxr-xr-x  2.0 unx     1371 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_rasp.py
+-rwxr-xr-x  2.0 unx      290 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_remote.sh
+-rwxr-xr-x  2.0 unx      949 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_rna3dcnn.py
+-rwxr-xr-x  2.0 unx     1707 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_rnakb.py
+-rwxr-xr-x  2.0 unx      912 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_rsRNASP.py
+-rwxr-xr-x  2.0 unx     4635 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_score.py
+-rwxr-xr-x  2.0 unx     1546 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_simrna.py
+-rwxr-xr-x  2.0 unx     2713 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_mq_wx_collect.py
+-rwxr-xr-x  2.0 unx       64 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_ok
+-rwxr-xr-x  2.0 unx     1369 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_ok_mail.py
+-rwxr-xr-x  2.0 unx     1789 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_pairs2SimRNArestrs.py
+-rwxr-xr-x  2.0 unx     1384 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_pairs_diff.py
+-rwxr-xr-x  2.0 unx     6145 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_pdb_edit_occupancy_bfactor.py
+-rwxr-xr-x  2.0 unx      415 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_pdb_merge_into_one.py
+-rwxr-xr-x  2.0 unx     4905 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_pdb_merge_structure_with_fragments.py
+-rwxr-xr-x  2.0 unx     1743 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_pdb_replace.py
+-rwxr-xr-x  2.0 unx    41410 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_pdb_toolsx.py
+-rwxr-xr-x  2.0 unx     1039 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_pk_simrna_to_one_line.py
+-rwxr-xr-x  2.0 unx      886 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_pkd2dG.py
+-rwxr-xr-x  2.0 unx      783 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_pkd2kd.py
+-rwxr-xr-x  2.0 unx     2263 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_plot_boxplotlike.py
+-rwxr-xr-x  2.0 unx     1610 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_plot_dendogram.py
+-rwxr-xr-x  2.0 unx     1780 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_plot_density.py
+-rwxr-xr-x  2.0 unx     2024 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_plot_heatmap.py
+-rwxr-xr-x  2.0 unx     1946 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_plot_hist.py
+-rwxr-xr-x  2.0 unx     1079 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_prediction_significance.py
+-rwxr-xr-x  2.0 unx     7029 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_refinement.py
+-rwxr-xr-x  2.0 unx     6070 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_rosetta_check_progress.py
+-rwxr-xr-x  2.0 unx     5462 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_rosetta_cluster.py
+-rwxr-xr-x  2.0 unx     1523 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_rosetta_extract_lowscore_decoys.py
+-rwxr-xr-x  2.0 unx     3977 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_rosetta_head.py
+-rwxr-xr-x  2.0 unx     4903 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_rosetta_min.py
+-rwxr-xr-x  2.0 unx     1045 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_rosetta_n.py
+-rwxr-xr-x  2.0 unx    10095 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_rosetta_run.py
+-rwxr-xr-x  2.0 unx     3450 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_sali2dotbracket.py
+-rwxr-xr-x  2.0 unx     2159 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_secondary_structure_prediction.py
+-rwxr-xr-x  2.0 unx     1221 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_simrna_cluster.py
+-rwxr-xr-x  2.0 unx     3636 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_simrna_extract.py
+-rwxr-xr-x  2.0 unx     1265 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_simrna_lowest.py
+-rwxr-xr-x  2.0 unx    12460 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_simrnaweb_download_job.py
+-rwxr-xr-x  2.0 unx     1645 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_ss_get_bps.py
+-rwxr-xr-x  2.0 unx      678 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_tools_which.py
+-rwxr-xr-x  2.0 unx     1182 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_traj_n.py
+-rwxr-xr-x  2.0 unx     1151 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_traj_nohetm.py
+-rwxr-xr-x  2.0 unx     1798 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_traj_range.py
+-rwxr-xr-x  2.0 unx     2625 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_workflow.py
+-rwxr-xr-x  2.0 unx     6714 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/rna_x3dna.py
+-rwxr-xr-x  2.0 unx     3945 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/select_seq_fromBLAStn_6outfm.py
+-rwxr-xr-x  2.0 unx      489 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/translate.py
+-rwxr-xr-x  2.0 unx    16446 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/triplexibility.py
+-rwxr-xr-x  2.0 unx    13671 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/triplexibility2.py
+-rwxr-xr-x  2.0 unx     1658 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/trx_mutagen_allvsall.py
+-rwxr-xr-x  2.0 unx     1544 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/trx_mutgen.py
+-rwxr-xr-x  2.0 unx     3321 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/trx_score.py
+-rwxr-xr-x  2.0 unx     3275 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.data/scripts/trx_score_allvsall.py
+-rw-r--r--  2.0 unx        0 b- defN 22-Mar-07 11:12 tests/__init__.py
+-rw-r--r--  2.0 unx    35142 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.dist-info/LICENSE
+-rw-r--r--  2.0 unx    33691 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.dist-info/METADATA
+-rw-r--r--  2.0 unx       92 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.dist-info/WHEEL
+-rw-r--r--  2.0 unx       16 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.dist-info/top_level.txt
+?rw-rw-r--  2.0 unx    53811 b- defN 22-Mar-07 11:12 rna_tools-3.8.6.dist-info/RECORD
+513 files, 3848136 bytes uncompressed, 991159 bytes compressed:  74.2%
```

## zipnote {}

```diff
@@ -1125,20 +1125,26 @@
 
 Filename: rna_tools/tools/triplexibility/triplexibility.py
 Comment: 
 
 Filename: rna_tools/tools/triplexibility/triplexibility2.py
 Comment: 
 
+Filename: rna_tools/tools/triplexibility/trx_mutagen_allvsall.py
+Comment: 
+
 Filename: rna_tools/tools/triplexibility/trx_mutgen.py
 Comment: 
 
 Filename: rna_tools/tools/triplexibility/trx_score.py
 Comment: 
 
+Filename: rna_tools/tools/triplexibility/trx_score_allvsall.py
+Comment: 
+
 Filename: rna_tools/tools/webserver-engine/__init__.py
 Comment: 
 
 Filename: rna_tools/tools/webserver-engine/checker.py
 Comment: 
 
 Filename: rna_tools/tools/webserver-engine/cleanup.py
@@ -1179,350 +1185,356 @@
 
 Filename: rna_tools/tools/webserver-engine/web/wsgi.py
 Comment: 
 
 Filename: rna_tools/tools/workflow/rna_workflow.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/clarna.py
+Filename: rna_tools-3.8.6.data/scripts/clarna.py
+Comment: 
+
+Filename: rna_tools-3.8.6.data/scripts/clarna_app.py
+Comment: 
+
+Filename: rna_tools-3.8.6.data/scripts/diffpdb.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/clarna_app.py
+Filename: rna_tools-3.8.6.data/scripts/pdbs_measure_atom_dists.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/diffpdb.py
+Filename: rna_tools-3.8.6.data/scripts/pymol_color_by_conserv.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/pdbs_measure_atom_dists.py
+Filename: rna_tools-3.8.6.data/scripts/pymol_preview_generator.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/pymol_color_by_conserv.py
+Filename: rna_tools-3.8.6.data/scripts/pymol_preview_install.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/pymol_preview_generator.py
+Filename: rna_tools-3.8.6.data/scripts/random_assignment_of_nucleotides.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/pymol_preview_install.py
+Filename: rna_tools-3.8.6.data/scripts/renum_pdb_to_aln.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/random_assignment_of_nucleotides.py
+Filename: rna_tools-3.8.6.data/scripts/reverse.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/renum_pdb_to_aln.py
+Filename: rna_tools-3.8.6.data/scripts/rna_add_chain.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/reverse.py
+Filename: rna_tools-3.8.6.data/scripts/rna_align_find_core.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_add_chain.py
+Filename: rna_tools-3.8.6.data/scripts/rna_align_find_seq_in_alignment.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_align_find_core.py
+Filename: rna_tools-3.8.6.data/scripts/rna_align_foldability.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_align_find_seq_in_alignment.py
+Filename: rna_tools-3.8.6.data/scripts/rna_align_get_ss_from_stk.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_align_foldability.py
+Filename: rna_tools-3.8.6.data/scripts/rna_align_seq_to_alignment.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_align_get_ss_from_stk.py
+Filename: rna_tools-3.8.6.data/scripts/rna_align_strip_stk.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_align_seq_to_alignment.py
+Filename: rna_tools-3.8.6.data/scripts/rna_alignment_get_species.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_align_strip_stk.py
+Filename: rna_tools-3.8.6.data/scripts/rna_alignment_process_id.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_alignment_get_species.py
+Filename: rna_tools-3.8.6.data/scripts/rna_alignment_r2r.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_alignment_process_id.py
+Filename: rna_tools-3.8.6.data/scripts/rna_cal_rmsd_trafl_plot.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_alignment_r2r.py
+Filename: rna_tools-3.8.6.data/scripts/rna_calc_evo_rmsd.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_cal_rmsd_trafl_plot.py
+Filename: rna_tools-3.8.6.data/scripts/rna_calc_fenergy.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_calc_evo_rmsd.py
+Filename: rna_tools-3.8.6.data/scripts/rna_calc_hbonds.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_calc_fenergy.py
+Filename: rna_tools-3.8.6.data/scripts/rna_calc_inf.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_calc_hbonds.py
+Filename: rna_tools-3.8.6.data/scripts/rna_calc_rmsd.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_calc_inf.py
+Filename: rna_tools-3.8.6.data/scripts/rna_calc_rmsd_all_vs_all.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_calc_rmsd.py
+Filename: rna_tools-3.8.6.data/scripts/rna_calc_rmsd_biopython.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_calc_rmsd_all_vs_all.py
+Filename: rna_tools-3.8.6.data/scripts/rna_calc_rmsd_multi_targets.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_calc_rmsd_biopython.py
+Filename: rna_tools-3.8.6.data/scripts/rna_calc_rmsd_trafl.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_calc_rmsd_multi_targets.py
+Filename: rna_tools-3.8.6.data/scripts/rna_clanstix.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_calc_rmsd_trafl.py
+Filename: rna_tools-3.8.6.data/scripts/rna_clarna_compare.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_clanstix.py
+Filename: rna_tools-3.8.6.data/scripts/rna_clarna_run.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_clarna_compare.py
+Filename: rna_tools-3.8.6.data/scripts/rna_csv_sort.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_clarna_run.py
+Filename: rna_tools-3.8.6.data/scripts/rna_dG2kd.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_csv_sort.py
+Filename: rna_tools-3.8.6.data/scripts/rna_dot2ct.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_dG2kd.py
+Filename: rna_tools-3.8.6.data/scripts/rna_draw_edges.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_dot2ct.py
+Filename: rna_tools-3.8.6.data/scripts/rna_ec2x.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_draw_edges.py
+Filename: rna_tools-3.8.6.data/scripts/rna_filter.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_ec2x.py
+Filename: rna_tools-3.8.6.data/scripts/rna_gromacs_ready.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_filter.py
+Filename: rna_tools-3.8.6.data/scripts/rna_helix_vis.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_gromacs_ready.py
+Filename: rna_tools-3.8.6.data/scripts/rna_kd2dG.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_helix_vis.py
+Filename: rna_tools-3.8.6.data/scripts/rna_kd2pkd.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_kd2dG.py
+Filename: rna_tools-3.8.6.data/scripts/rna_md.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_kd2pkd.py
+Filename: rna_tools-3.8.6.data/scripts/rna_merge_dfs.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_md.py
+Filename: rna_tools-3.8.6.data/scripts/rna_merge_two_dfs.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_merge_dfs.py
+Filename: rna_tools-3.8.6.data/scripts/rna_minimize.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_merge_two_dfs.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_clashscore.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_minimize.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_collect.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_clashscore.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_dfire.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_collect.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_eSCORE.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_dfire.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_exp.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_eSCORE.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_farfar2.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_exp.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_farna.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_farfar2.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_plot.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_farna.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_rasp.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_plot.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_remote.sh
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_rasp.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_rna3dcnn.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_remote.sh
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_rnakb.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_rna3dcnn.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_rsRNASP.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_rnakb.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_score.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_rsRNASP.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_simrna.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_score.py
+Filename: rna_tools-3.8.6.data/scripts/rna_mq_wx_collect.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_simrna.py
+Filename: rna_tools-3.8.6.data/scripts/rna_ok
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_mq_wx_collect.py
+Filename: rna_tools-3.8.6.data/scripts/rna_ok_mail.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_ok
+Filename: rna_tools-3.8.6.data/scripts/rna_pairs2SimRNArestrs.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_ok_mail.py
+Filename: rna_tools-3.8.6.data/scripts/rna_pairs_diff.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_pairs2SimRNArestrs.py
+Filename: rna_tools-3.8.6.data/scripts/rna_pdb_edit_occupancy_bfactor.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_pairs_diff.py
+Filename: rna_tools-3.8.6.data/scripts/rna_pdb_merge_into_one.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_pdb_edit_occupancy_bfactor.py
+Filename: rna_tools-3.8.6.data/scripts/rna_pdb_merge_structure_with_fragments.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_pdb_merge_into_one.py
+Filename: rna_tools-3.8.6.data/scripts/rna_pdb_replace.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_pdb_merge_structure_with_fragments.py
+Filename: rna_tools-3.8.6.data/scripts/rna_pdb_toolsx.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_pdb_replace.py
+Filename: rna_tools-3.8.6.data/scripts/rna_pk_simrna_to_one_line.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_pdb_toolsx.py
+Filename: rna_tools-3.8.6.data/scripts/rna_pkd2dG.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_pk_simrna_to_one_line.py
+Filename: rna_tools-3.8.6.data/scripts/rna_pkd2kd.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_pkd2dG.py
+Filename: rna_tools-3.8.6.data/scripts/rna_plot_boxplotlike.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_pkd2kd.py
+Filename: rna_tools-3.8.6.data/scripts/rna_plot_dendogram.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_plot_boxplotlike.py
+Filename: rna_tools-3.8.6.data/scripts/rna_plot_density.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_plot_dendogram.py
+Filename: rna_tools-3.8.6.data/scripts/rna_plot_heatmap.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_plot_density.py
+Filename: rna_tools-3.8.6.data/scripts/rna_plot_hist.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_plot_heatmap.py
+Filename: rna_tools-3.8.6.data/scripts/rna_prediction_significance.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_plot_hist.py
+Filename: rna_tools-3.8.6.data/scripts/rna_refinement.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_prediction_significance.py
+Filename: rna_tools-3.8.6.data/scripts/rna_rosetta_check_progress.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_refinement.py
+Filename: rna_tools-3.8.6.data/scripts/rna_rosetta_cluster.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_rosetta_check_progress.py
+Filename: rna_tools-3.8.6.data/scripts/rna_rosetta_extract_lowscore_decoys.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_rosetta_cluster.py
+Filename: rna_tools-3.8.6.data/scripts/rna_rosetta_head.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_rosetta_extract_lowscore_decoys.py
+Filename: rna_tools-3.8.6.data/scripts/rna_rosetta_min.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_rosetta_head.py
+Filename: rna_tools-3.8.6.data/scripts/rna_rosetta_n.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_rosetta_min.py
+Filename: rna_tools-3.8.6.data/scripts/rna_rosetta_run.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_rosetta_n.py
+Filename: rna_tools-3.8.6.data/scripts/rna_sali2dotbracket.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_rosetta_run.py
+Filename: rna_tools-3.8.6.data/scripts/rna_secondary_structure_prediction.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_sali2dotbracket.py
+Filename: rna_tools-3.8.6.data/scripts/rna_simrna_cluster.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_secondary_structure_prediction.py
+Filename: rna_tools-3.8.6.data/scripts/rna_simrna_extract.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_simrna_cluster.py
+Filename: rna_tools-3.8.6.data/scripts/rna_simrna_lowest.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_simrna_extract.py
+Filename: rna_tools-3.8.6.data/scripts/rna_simrnaweb_download_job.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_simrna_lowest.py
+Filename: rna_tools-3.8.6.data/scripts/rna_ss_get_bps.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_simrnaweb_download_job.py
+Filename: rna_tools-3.8.6.data/scripts/rna_tools_which.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_ss_get_bps.py
+Filename: rna_tools-3.8.6.data/scripts/rna_traj_n.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_tools_which.py
+Filename: rna_tools-3.8.6.data/scripts/rna_traj_nohetm.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_traj_n.py
+Filename: rna_tools-3.8.6.data/scripts/rna_traj_range.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_traj_nohetm.py
+Filename: rna_tools-3.8.6.data/scripts/rna_workflow.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_traj_range.py
+Filename: rna_tools-3.8.6.data/scripts/rna_x3dna.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_workflow.py
+Filename: rna_tools-3.8.6.data/scripts/select_seq_fromBLAStn_6outfm.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/rna_x3dna.py
+Filename: rna_tools-3.8.6.data/scripts/translate.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/select_seq_fromBLAStn_6outfm.py
+Filename: rna_tools-3.8.6.data/scripts/triplexibility.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/translate.py
+Filename: rna_tools-3.8.6.data/scripts/triplexibility2.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/triplexibility.py
+Filename: rna_tools-3.8.6.data/scripts/trx_mutagen_allvsall.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/triplexibility2.py
+Filename: rna_tools-3.8.6.data/scripts/trx_mutgen.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/trx_mutgen.py
+Filename: rna_tools-3.8.6.data/scripts/trx_score.py
 Comment: 
 
-Filename: rna_tools-3.8.5.data/scripts/trx_score.py
+Filename: rna_tools-3.8.6.data/scripts/trx_score_allvsall.py
 Comment: 
 
 Filename: tests/__init__.py
 Comment: 
 
-Filename: rna_tools-3.8.5.dist-info/LICENSE
+Filename: rna_tools-3.8.6.dist-info/LICENSE
 Comment: 
 
-Filename: rna_tools-3.8.5.dist-info/METADATA
+Filename: rna_tools-3.8.6.dist-info/METADATA
 Comment: 
 
-Filename: rna_tools-3.8.5.dist-info/WHEEL
+Filename: rna_tools-3.8.6.dist-info/WHEEL
 Comment: 
 
-Filename: rna_tools-3.8.5.dist-info/top_level.txt
+Filename: rna_tools-3.8.6.dist-info/top_level.txt
 Comment: 
 
-Filename: rna_tools-3.8.5.dist-info/RECORD
+Filename: rna_tools-3.8.6.dist-info/RECORD
 Comment: 
 
 Zip file comment:
```

## rna_tools/_version.py

```diff
@@ -4,18 +4,18 @@
 # unpacked source archive. Distribution tarballs contain a pre-generated copy
 # of this file.
 
 import json
 
 version_json = '''
 {
- "date": "2022-02-21T22:55:01+0100",
+ "date": "2022-03-07T11:55:43+0100",
  "dirty": true,
  "error": null,
- "full-revisionid": "e142af908ca11bfedca337fbd11db1171f8f4fa0",
- "version": "3.8.5"
+ "full-revisionid": "781b1507eea17cda6d9f5c1035c3f0c890d6ced5",
+ "version": "3.8.6"
 }
 '''  # END VERSION_JSON
 
 
 def get_versions():
     return json.loads(version_json)
```

## rna_tools/rna_tools_config.py

```diff
@@ -46,15 +46,15 @@
     os.mkdir(tmp)
 except:
     pass
 
 LOG_DIRECTORY = tmp
 TMP_PATH = tmp
 SANDBOX_PATH = tmp
-BIN_PATH = ""
+BIN_PATH = "/opt/homebrew/bin/"
 
 WRAPPERS_PATH = RT + "/rna_tools/tools/mq/"
 
 dfire_PATH = ""      # "/Users/magnus/work/papers/mqaprna/mqaprna_env/mqapRNA/opt/dfire_rna/"
 
 RNA3DCNN_PATH=""     # "/Users/magnus/work/papers/mqaprna/mqaprna_env/mqapRNA/opt/RNA3DCNN"
 PYTHON3_PATH=""      # "/Users/magnus/miniconda2/envs/py37/bin/python"
```

## rna_tools/tools/PyMOL4RNA/PyMOL4RNA.py

```diff
@@ -78,14 +78,26 @@
     if verbose: print('cmd:' + cmd)
     o = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
                          executable=EXECUTABLE)
     out = o.stdout.read().strip().decode()
     err = o.stderr.read().strip().decode()
     return out, err
 
+def spla():
+    cmd.do("color forest, chain A")
+    cmd.do("color firebrick, chain B")
+cmd.extend('spla', spla)
+print('spla - color A and B')
+
+def spln():
+    cmd.do("color forest, chain 2")
+    cmd.do("color firebrick, chain 6")
+cmd.extend('spln', spln)
+print('spln - color 2 and 6')
+      
 def color_protein():
     cmd.do("color blue, resn ARG+LYS+HIS and (sele)")
     cmd.do("color red, resn ASP+GLU and (sele)")
     cmd.do("color green, resn GLY+ALA+VAL+LEU+ILE+MET+PHE and (sele)")
     cmd.do("color yellow, resn TYR+TRP and (sele)")
     cmd.do("color forest, resn SER+THR+CYS+ASN+GLN and (sele)")
     cmd.do("color pink, resn PRO and (sele)")
@@ -118,14 +130,23 @@
     """Helper function used for color-coding based on residue indexes ranges."""
     for t in txt.strip().split('\n'):
         print(t)
         color, resi = t.replace('color ', '').split(',')
         print((color, resi))
         cmd.color(color.strip(), resi.strip())
 
+def cmd_text(txt):
+    """Helper function used for color-coding based on residue indexes ranges."""
+    for t in txt.strip().split('\n'):
+        cmd.do(t)
+
+def delete_all():
+    cmd.delete('*')
+cmd.extend('dall', delete_all)
+print('dall - delete_all')
 
 def rp():
     """Represent your RNA."""
     cmd.hide("sticks", "all")
     cmd.hide("lines", "all")
     cmd.show("cartoon", "all")
     cmd.set("cartoon_ring_mode", 3)
@@ -303,14 +324,15 @@
 
     c = 1
     for r1 in models:
         for r2 in models:
             if r1 == r2:
                 rmsd = 0
             else:
+                print(r1, r2)
                 values = cmd.align(r1, r2, cycles=cycles)
                 # RaR [1]       RbR [3]
                 # RaR  #AA  CoR               RbR  #AbR RS   AR'
                 # (0.668652355670929, 241, 5, 1.1646124124526978, 293, 199.0, 38)
                 rmsd = round(values[0], 3)
             t += str(rmsd) + ' '
         #print('...', c, r1)
@@ -389,14 +411,28 @@
     for obj in obj_list:
         print("  ", obj, colours[i])
         cmd.color(colours[i], obj)
         i = i+1
         if(i == ncolours):
            i = 0
 
+def da():  # donors accepters
+    t = """
+    h_add;
+    set sphere_scale, 0.2, (all)
+    set sphere_transparency, 0
+    #color blue, donors;
+    #color green, acceptors;
+    show sphere, donors;
+    show sphere, acceptors;
+    color gray,  name H*; # color atoms from white to gray
+    """
+    cmd.do(t)
+cmd.extend('da', da)
+    
 def pdb():
     """Get PDB content of selection.
 
     .. image:: ../../rna_tools/utils/PyMOL4RNA/doc/pdb.png"""
     tmpfn = '/tmp/pymol_get_pdb.pdb'
     cmd.save(tmpfn, '(sele)')
     s = RNAStructure(tmpfn)
@@ -529,18 +565,27 @@
     out, err = exe(cmdline)
     print('\n'.join(out.split('\n')[2:]))  # to remove first line of py3dna /tmp/xxx
     if err:
         print(err)
     f.close()
 
 
-def mutate(mutation, selection):
-    """Get Secondary Structure of (sele) based on py3dna.py.
 
-    .. image:: ../../rna_tools/utils/PyMOL4RNA/doc/ss.png
+def rtrun(cmd, selection, suffix):
+    f = tempfile.NamedTemporaryFile(delete=False) # True)
+    output = os.path.dirname(f.name) + os.sep +  selection + '.pdb'
+    output2 = os.path.dirname(f.name) + os.sep + selection + '_mut.pdb'
+    exe(cmdline)
+    print(cmdline)
+    cmd.save(output, selection)
+
+    # 'A:1A+2A+3A+4A' 
+
+def mutate(mutation, selection):
+    """
     """
     f = tempfile.NamedTemporaryFile(delete=False) # True)
     output = os.path.dirname(f.name) + os.sep +  selection + '.pdb'
     output2 = os.path.dirname(f.name) + os.sep + selection + '_mut.pdb'
     cmdline = "rna_pdb_toolsx.py --mutate " + mutation + ' ' + output + ' > ' + output2
     print(cmdline)
     exe(cmdline)
@@ -724,15 +769,15 @@
 
 def edges(selection):
     """Save selection into a file in a temp folder and run rna_draw_edges.py on it and load it into this session"""
     f = tempfile.TemporaryDirectory()
     tmpf = f.name + os.sep + strip_selection_name(selection) + '.pdb'
     outf = f.name + '/output.py'
     cmd.save(tmpf, selection)
-    cmdline = 'rna_draw_edges.py --name %s %s > %s' % (strip_selection_name(selection), tmpf, outf)
+    cmdline = '/Users/magnus/miniconda3/bin/rna_draw_edges.py --name %s %s > %s' % (strip_selection_name(selection), tmpf, outf)
     print(cmdline)
     out, err = exe(cmdline)
     if err:
         print(err)
     cmd.load(outf)
     cmd.orient()
 
@@ -740,31 +785,20 @@
 
 
 def ino():
     """Sphare and yellow inorganic, such us Mg.
 
     .. image:: ../../rna_tools/utils/PyMOL4RNA/doc/ion.png"""
     cmd.show("spheres", "inorganic")
-    #cmd.set('sphere_scale', '0.25', '(all)')
-    cmd.set('sphere_scale', '1', '(all)')
+    cmd.set('sphere_scale', '0.25', '(all)')
+    #cmd.set('sphere_scale', '1', '(all)')
     cmd.color("yellow", "inorganic")
 
 mapping = [[u'PRP8', 'A', u'skyblue'], [u'BRR2', 'B', u'grey60'], [u'BUD31', 'C', u'dirtyviolet'], [u'CEF1', 'D', u'raspberry'], [u'CLF1', 'E', u'raspberry'], [u'CWC15', 'F', u'dirtyviolet'], [u'CWC16/YJU2', 'G', u'lightteal'], [u'CWC2', 'H', u'ruby'], [u'CWC21', 'I', u'violetpurple'], [u'CWC22', 'J', u'bluewhite'], [u'CWC25', 'K', u'deepteal'], [u'Intron', 'L', u'black'], [u'ISY1', 'M', u'dirtyviolet'], [u'LEA1', 'N', u'palegreen'], [u'Msl1', 'O', u'palegreen'], [u'PRP45', 'P', u'lightpink'], [u'PRP16', 'Q', u'smudge'], [u'CDC40\xa0(PRP17, SLU4, XRS2)', 'R', u'dirtyviolet'], [u'PRP19 (PSO4)', 'S', u'grey70'], [u'PRP46', 'T', u'lightblue'], [u'SLT11/ECM2', 'U', u'chocolate'], [u'SNT309', 'V', u'grey70'], [u'SNU114', 'W', u'slate'], [u'SYF2', 'X', u'brightorange'], [u'SYF1', 'Y', u'brightorange'], [u'U2', 'Z', u'forest'], [u'U5', 'a', u'density'], [u'U5_SmRNP', 'b', u'deepblue'], [u'U6', 'c', u'firebrick'], [u'Intron', 'r', u'grey50'], [u'Exon', 'z', u'yellow'], [u'exon-3', 'y', u'yellow'], [u'exon-5', 'z', u'yellow'], [u'PRP4 ', 'd', u'grey50'], [u'PRP31', 'e', u'grey50'], [u'PRP6', 'f', u'grey50'], [u'PRP3', 'g', u'grey50'], [u'DIB1', 'h', u'grey50'], [u'SNU13', 'i', u'grey50'], [u'LSM8', 'j', u'grey50'], [u'LSM2', 'k', u'grey50'], [u'LSM3', 'l', u'grey50'], [u'LSM6', 'm', u'grey50'], [u'LSM5', 'n', u'grey50'], [u'LSM7', 'o', u'grey50'], [u'LSM4', 'p', u'grey50'], [u'SNU66', 'q', u'grey50'], [u'RNA (intron or U6 snRNA)', 'r', u'grey50'], [u'5EXON', 's', u'grey50'], [u'BUD13', 't', u'grey60'], [u'CLF2', 'u', u'rasberry'], [u'Cus1', 'v', u'palegreen'], [u'CWC24', 'w', u'grey60'], [u'CWC27', 'x', u'grey60'], [u'HSH155', '1', u'smudge'], [u'HSH49', '2', u'sand'], [u'PML1', '3', u'grey60'], [u'PRP11', '4', u'palegreen'], [u'PRP2', '5', u'palegreen'], [u'RDS3', '6', u'palegreen'], [u'RSE1', '7', u'smudge'], [u'SNU17', '8', u'grey60'], [u'Ysf3', '9', u'palegreen'], [u'cwc23', 'd', u'grey50'], [u'SPP382\xa0(CCF8, NTR1)', 'e', u'grey50'], [u'NTR2', 'f', u'grey50'], [u'PRP43', 'g', u'grey50'], [u'SMB1', 'h', u'grey50'], [u'SME1', 'i', u'grey50'], [u'SMX3', 'j', u'grey50'], [u'SMX2\xa0(SNP2)', 'k', u'grey50'], [u'SMD3', 'l', u'grey50'], [u'SMD1', 'm', u'grey50'], [u'SMD2', 'n', u'grey50'], [u'PRP22', 'o', u'grey50'], [u'PRP18', 'p', u'grey50'], [u'SLU7', 'q', u'grey50'], [u'SMF', 'd', u'grey50'], [u'SMG', 'e', u'grey50'], [u'PRP9', 'f', u'grey50'], [u'PRP21', 'g', u'grey50'], [u'SNU23', 'r', u'grey50'], [u'PRP38', 's', u'grey50'], [u'SPP381', 'w', u'grey50']]
 
-def g2():
-    txt = """color gray, all;
-    color yellow, chain B;
-    color red, resi 788-824;
-    color yellow, resi 828-866;
-    color forest, resi 476-490;
-    """
-    color_by_text(txt)
-    cmd.color('pink', 'resi 120')
-    cmd.color('pink', 'resi 2')
-
 
 def rgyration(selection='(all)', quiet=1):
     '''
 
 [PyMOL] RES: radius of gyration
 From: Tsjerk Wassenaar <tsjerkw@gm...> - 2011-03-31 14:07:03
 https://sourceforge.net/p/pymol/mailman/message/27288491/
@@ -833,14 +867,20 @@
          hit = re.findall('chain: (?P<chain>\w+).*\# (?P<residue>\d+)', l)
          if hit: # [('X', '1')]
              cmd.color('red', name + ' and chain ' + hit[0][0] + ' and resi ' + hit[0][1])
     print(out)
 
 cmd.extend('inspect', inspect)
     
+def ll():
+    cmd.do('hide all')
+    cmd.do('show lines')
+cmd.extend('ll', ll)
+print('ll - show lines only')
+
 def rpr(selection):
     """rpr"""
     f = tempfile.NamedTemporaryFile(delete=False)
     input = os.path.dirname(f.name) + os.sep +  selection + '.pdb'
     cmd.save(input, selection)
 
     output = os.path.dirname(f.name) + os.sep +  selection + '_rpr.pdb'
@@ -860,114 +900,154 @@
     cmdline = 'diffpdb.py ' + input + ' ' + output + ' &'
     print(cmdline)
     #os.system(cmdline)
     exe(cmdline)
 
 cmd.extend('diff', diff)
 
-def mini(f):
+def mini(f):  # min with qrna
 
     #os.system('/home/magnus/opt/qrnas/QRNA02/QRNA -i ' + f + ' -c /home/magnus/opt/qrnas/QRNA02/configfile.txt -o out.pdb')
     os.system('~/opt/qrnas/QRNA02/QRNA -i ' + f + ' -c ~/opt/qrnas/QRNA02/configfile.txt -o out.pdb')
     cmd.delete('mini')
     cmd.load('out.pdb', 'mini')
     print('end')
 
 
 def reload():
     """Reload ~/.pymolrc and all included there packages (e.g. with run <foo.py>)"""
     cmd.do('@ ~/.pymolrc')
 
-def clr():
-  """clr - make white bg and structure black"""
-  cmd.bg_color( "white" )
-  color_by_text('color black, all')
-
 
 def rlabel():
     cmd = "n. C1'", '"%s %s" % (resn, resi)'
     print('label ' + cmd)
     cmd.label(cmd)
 
 
-def sav(name):
+
+def sav(name): #sav
     # cmd.bg_color( "white" )
     tf = tempfile.NamedTemporaryFile(delete=False)
     fn = tf.name + '.png'
     tf = tempfile.NamedTemporaryFile(delete=False)
     cfn = tf.name + '.png'
 
     psefn = '~/Desktop/' + name + '.pse'
     cmd.save(psefn)
 
     cmd.save(fn)
 
-    cmdline= "/usr/local/bin/convert " + fn + " -gravity center -crop 3:3 +repage " + cfn
+    cmdline= "/opt/homebrew/bin/convert " + fn + " -gravity center -crop 3:3 +repage " + cfn
+    print(cmdline)
+    os.system(cmdline)
+    cmdline = '/opt/homebrew/bin/fileicon set ' + psefn + ' ' + cfn
+    print(cmdline)
     os.system(cmdline)
-
-    os.system('/usr/local/bin/fileicon set ' + psefn + ' ' + cfn)
 
     #cmd.png(coverfn, 576,576)
     #cmd.ray(576,576)
+cmd.extend('sav', sav)
+
 def hide_rna():
     cmd.hide('(polymer.nucleic)')
 cmd.extend('rna-hide', hide_rna)
 
 def show_rna():
     cmd.show('(polymer.nucleic)')
 cmd.extend('rna-show', show_rna)
 
+def clr():
+    cmd.do('delete *')
+cmd.extend('clr', clr)
+print('clr - delete all')
+
+def bw():
+  """clr - make white bg and structure black"""
+  cmd.bg_color( "white" )
+  color_by_text('color black, all')
+cmd.extend('bw', bw)
+print('bw - white bg, black all')
+
 def select_rna():
     cmd.select('polymer.nucleic')
 cmd.extend('select-rna', select_rna)
 
 def hide_protein():
     cmd.hide('(polymer.protein)')
 #cmd.extend('protein-hide', hide_protein)
 #cmd.extend('rp-hide', hide_protein)
-
 def select_protein():
     cmd.select('polymer.protein')
 cmd.extend('protein-select', select_protein)
 
 def tmp():
     cmd.save('/home/' + user + '/Desktop/' + tmp + '.png')
     cmd.save('/home/' + user + '/Desktop/' + tmp + '.pse')
 
+def tp(): #tp temp pse
+    """tp here"""
+    import datetime
+
+    # cmd.bg_color( "white" )
+    tf = tempfile.NamedTemporaryFile(delete=False)
+    fn = tf.name + '.png'
+    tf = tempfile.NamedTemporaryFile(delete=False)
+    cfn = tf.name + '.png'
+
+    date = datetime.datetime.today().strftime('%Y-%m-%d.%H%M%S')
+    psefn = '~/Desktop/' + date + '.pse'
+    cmd.save(psefn)
+
+    cmd.save(fn)
+
+    cmdline= "/opt/homebrew/bin/convert " + fn + " -gravity center -crop 3:3 +repage " + cfn
+    print(cmdline)
+    os.system(cmdline)
+    cmdline = '/opt/homebrew/bin/fileicon set ' + psefn + ' ' + cfn
+    print(cmdline)
+    os.system(cmdline)
+
+cmd.extend('tp', tp)
 
-################################################################################
 def sav_tmp():
     from shutil import copyfile
     import datetime
     try:
         TMP_FOLDER + ' '
     except:
         print("Error: Set up TMP_FOLDER in your ~/.pymolrc, e.g. TMP_FOLDER = '/home/magnus/Desktop/PyMOL/'")
         return
 
     try:
         os.mkdir(TMP_FOLDER)
     except:
         pass
 
-    date = datetime.datetime.today().strftime('%Y-%m-%d.%S')
+    date = datetime.datetime.today().strftime('%Y-%m-%d.%H%M%S')
     try:
         fn = TMP_FOLDER +  os.sep + id + '_' + date + '.pse'
     except TypeError:
         fn = TMP_FOLDER +  os.sep + '_' + date + '.pse'
     cmd.save(fn)
     print('Save...' + fn)
     cmd.save(fn.replace('.pse', '.png'))
     copyfile(fn, TMP_FOLDER + '/last.pse')
 
 def load_tmp():
     print('Load...')
     cmd.load(TMP_FOLDER + '/last.pse')
 
 
+def trim():
+    cmd.do("remove solvent")
+    cmd.do("remove resn NA")
+    
+cmd.extend('trim', trim)
+    
 def get_resi():
     """
     PyMOL>get_resi()
     358+376+288+290+359+383+386+382+289+287+384+357+385+360+377+361
     """
     stored.residues = set()
     cmd.iterate('(sele)', "stored.residues.add(resi)") # ('289') # residue only
@@ -979,24 +1059,92 @@
     #cmd.iterate('(sele)', 'l.append([resn, resi])')
     #rint(l)
 
 def findN(r):
     c = 'select br. all within ' + str(r) + ' of (sele)'
     cmd.do(c)
 
+def white():
+    cmd.set('dash_color', 'black')
+    cmd.set('dash_width', 2)
+    cmd.bg_color( "white" )
+cmd.extend('w', white)
 
 def desc(t='', width=80):
     print()
     print()
     print()
     print(t.center(int(width)))
     print()
     print()
     print()
 
+def s(): # quick save selected to tmp.pdb
+    cmd.do('save tmp.pdb, (sele)')
+cmd.extend('s', s)
+print('s - quick save selected to tmp.pdb')
+
+def draw():
+    #cmd.select("name C1'")
+    t = """
+
+
+    set sphere_scale, 0.4, name C1'
+    show sphere, name C1'
+    color black, name C1'
+
+    remove name OP2+OP1+O5'+P+C5'+O2'+O3'+C4'+C3'+C2'+O4'
+
+    h_add
+
+    select (bound_to name C1') and name H*
+    remove (sele)
+
+    set spec_reflect, 0
+    # https://pymolwiki.org/index.php/Spec_reflect
+
+    #set sphere_transparency, 0
+    #color blue, donors;
+    #color green, acceptors;
+    #show sphere, donors;
+    #show sphere, acceptors;
+
+    color gray,  name H*; # color atoms from white to gray
+
+
+set dash_color, black
+set dash_width, 1; set dash_gap,  0.2
+bg white;
+#color black;
+"""
+    cmd_text(t)
+
+cmd.extend('draw', draw)
+cmd.extend('dr', draw)
+
+def se(): # save
+    cmd.do('save tmp.pdb, (enabled)')
+cmd.extend('se', se)
+print('se - quick save enabled to tmp.pdb')
+      
+def ha():
+    """
+        cmd.do('h_add')
+    """
+    cmd.do('h_add')
+cmd.extend('ha', ha)
+    
+def hb(): # hydrogen bonds
+    cmd.do('contacts *,*')
+cmd.extend('hb', hb)
+
+
+def pn(): # pn
+    cmd_text('save ~/Desktop/tmp.png')
+cmd.extend('pn', pn) # pn
 def quickref():
     print('   PyMOL4RNA (rna-tools)  ')
     print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
     print('Quickref `qr`: ')
     print('  alter (sele), chain="B" ')
     print('  alter (sele), resv -= 4')
     print('  alter (chain B), resv -= 44 ')
@@ -1031,16 +1179,14 @@
     cmd.set_key('CTRL-S', sav_tmp)
     cmd.set_key('CTRL-Z', load_tmp)  # ostatni wrzucam tutaj
     #cmd.load, ['/home/magnus/Desktop/tmp.pse'])
     # main code #
 
     cmd.extend('quickref', quickref)
     cmd.extend('qr', quickref)
-    cmd.extend('rg2', g2)
-    cmd.extend('x', g2)
 
     cmd.extend('rp', rp)
     cmd.extend('p', p)
     cmd.extend('pdb', pdb)
     cmd.extend('seq', seq)
     cmd.extend('seqsel', seqsel)
     cmd.extend('rseq', seq)
@@ -1057,33 +1203,43 @@
     cmd.extend('x3dna', x3dna)
     cmd.extend("rgyration", rgyration)
     cmd.extend('rlabel', 'rlabel')
 
     cmd.extend('grid_on', grid_on)
     cmd.extend('grid_off', grid_off)
     cmd.extend('reload', reload)
+    cmd.extend('rl', reload)
 
     cmd.extend('color_aa_types', color_aa_types)
 
     cmd.extend('names', names)
 
     cmd.extend('findX', findN)
 
-    # set dash lines
-    cmd.set('dash_color', 'red')
-    cmd.set('dash_width', 4)
-
-    cmd.extend('sav', sav)
+    # set dash lines #hbonds #hydrogen
+    cmd.set('dash_color', 'white')
+    cmd.set('dash_width', 2)
+    cmd.set('cartoon_tube_radius', 0.5)
+    
     cmd.extend('save_transformed', save_transformed)
     cmd.extend('savt', save_transformed)
     cmd.extend('show_all_at_once', show_all_at_once)
 
     cmd.set('ignore_case', 'off')
     #cmd.set('cartoon_ring_mode', '3')
     #cmd.set('cartoon_ring_finder', '2')
     #cmd.extend('spl_select', spl_select)
     #    print('ignore_case made off')
     print('\_ PYMOL4RNA loading .... [ok]')
 
     cmd.extend('desc', desc)
     #cmd.do('set overlay, 1')
 
+    #### change do desktop
+    user = getpass.getuser()
+    cw = os.path.abspath(os.getcwd())
+    print(cw)
+    if cw == '/Users/magnus':
+        print('change to Desktop')
+        os.chdir('/Users/magnus/Desktop/')
+    cw = os.path.abspath(os.getcwd())
+    print(cw)
```

## rna_tools/tools/PyMOL4RNA/PyMOL4Spliceosome.py

```diff
@@ -7,14 +7,39 @@
 
 try:
     from pymol import cmd
 except ImportError:
     print("PyMOL Python lib is missing")
     # sys.exit(0)
 
+def spl_help():
+    print("""
+PyMOL4Spliceosome
+-----------------------
+extract all (ea)  - show
+colors            - list all colors
+
+spl hprp8
+spl prp8
+spl yprp8 - "color skyblue, PRP8_* and resi 885-1251"
+spl hprp28
+
+spl chains
+ color blue, chain 5
+ color red, chain 6
+ color forest, chain 2
+
+set cartoon_ring_mode to 3 ...
+
+""")
+spl_help()
+
+print("""
+spl g2 # coloring for 6chr g2 introns
+""")
 
 cmd.set('transparency', 0.25)
 cmd.set('cartoon_ring_mode', 3)
 # colors taken from https://github.com/maxewilkinson/Spliceosome-PyMOL-sessions
 cmd.set_color('lightgreen', [144, 238, 144])
 cmd.set_color('darkgreen', [0, 100, 0])
 cmd.set_color('darkseagreen', [143, 188, 143])
@@ -34,36 +59,105 @@
 cmd.set_color('violetBlue', [40, 0, 120])
 cmd.set_color('mediumpurple', [147,112,219])
 cmd.set_color('lavenderblush', [255,240,245])
 cmd.set_color('lavender', [230,230,250])
 cmd.set_color('thistle', [216,191,216])
 
 
+
+def color_by_text(txt):
+    """Helper function used for color-coding based on residue indexes ranges."""
+    for t in txt.strip().split('\n'):
+        print(t)
+        color, resi = t.replace('color ', '').split(',')
+        print((color, resi))
+        cmd.color(color.strip(), resi.strip())
+
 def spl(arg=''):
     """
     action='', name=''
     """
+    #reload()
+    print(arg)
     if ' ' in arg:
         action, name = arg.split()
         name = name.lower()
     else:
         action = arg
         name = ''
     #import pandas as pd
     #df = pd.read_excel("/home/magnus/Desktop/pyMoL_colors-EMX.xlsx")
     if not action or action == 'help':
         spl_help()
+        cmd.do('color red, chain A')        
+        cmd.do('color forest, chain B')
+        cmd.do('color grey, chain C')        
+        cmd.do('color grey, chain D')
+
+        cmd.do('color grey, chain I')        
+        cmd.do('color grey, chain J')
+
+        cmd.do('color purple, chain D and resi 4')        
+        cmd.do('color purple, resn MG')
+        # exon
+        cmd.do('color yellow, chain G')
+        cmd.do('color yellow, chain E')
+        
+        cmd.do('color blue, chain H')
+        cmd.do('color blue, chain U')
+
+        cmd.show("spheres", "inorganic")
+        cmd.set('sphere_scale', '1', '(all)')
+        #cmd.set('sphere_scale', '1', '(all)')
+        cmd.color("yellow", "inorganic")
+
+    elif arg == 'g22':
+        print('g22')
+        t = """
+        color gray, resi 1-600;
+        color red, resi 550-586;
+        color pink, resi 587-630;
+        color brown, resi 507-549;
+        color marine, resi 350-424;
+        color yellow, resi 425-507;
+        color forest, resi 1-26;
+        color forest, resi 327-350;
+        color yellow, chain B;
+        """
+        color_by_text(t)
+
+    elif arg == 'g21':
+        print('g21')
+        t = """
+        color gray, resi 1-600;
+        color red, resi 300-400
+        color marine, resi 288-289;
+        """
+        color_by_text(t)
+
     elif action == 'color' or arg=='c':
         code_for_color_spl.spl_color()
     elif arg == 'extract all' or arg == 'ea' or arg == 'e':
         code_for_spl.spl_extract()
     elif arg == 'cc': # colorchains
         cmd.do('color blue, chain 5')
         cmd.do('color red, chain 6')        
         cmd.do('color forest, chain 2')
+
+        cmd.do('color red, chain V')        
+        cmd.do('color forest, chain Z')
+
+    elif arg == 'ab': # colorchains
+        cmd.do('color red, chain B')        
+        cmd.do('color forest, chain A')
+        cmd.do('color grey, chain C')        
+        cmd.do('color grey, chain D')
+        cmd.do('color purple, chain D and resi 4')        
+        cmd.do('color purple, resn MG')
+
     elif arg == 'trim':
         cmd.do("""    	remove chain 5;
 	remove chain 2 and resi 1-19;
 	remove chain 2 and resi 50-200;
 	remove chain 6 and resi 87-200;
 	remove chain 6 and resi 1-40;
 	remove chain P and resi 45-200;
@@ -166,35 +260,24 @@
     align /5ylz//D, /5lj3//V;
     """)
     else:
         spl_help()
 
 cmd.extend('spl', spl)
 
-def spl_help():
-    print("""
-PyMOL4Spliceosome
------------------------
-extract all (ea)  - show
-colors            - list all colors
-
-spl hprp8
-spl prp8
-spl yprp8 - "color skyblue, PRP8_* and resi 885-1251"
-spl hprp28
-
-spl chains
- color blue, chain 5
- color red, chain 6
- color forest, chain 2
-
-set cartoon_ring_mode to 3 ...
-
-""")
-spl_help()
+def g2():
+    txt = """color gray, all;
+    color yellow, chain B;
+    color red, resi 788-824;
+    color yellow, resi 828-866;
+    color forest, resi 476-490;
+    """
+    color_by_text(txt)
+    cmd.color('pink', 'resi 120')
+    cmd.color('pink', 'resi 2')
 
 def __spl_color():
     for m in mapping:
         protein = m[0]
         chain = m[1]
         color = m[2]
         print('\_' + ' '.join([protein, chain, color]))
```

## rna_tools/tools/pymol_preview_generator/pymol_preview_generator.py

```diff
@@ -6,14 +6,15 @@
 """
 import sys
 import argparse
 import tempfile
 import os
 import subprocess
 import platform
+from rna_tools.rna_tools_config import BIN_PATH
 
 def get_parser():
     parser = argparse.ArgumentParser(
         description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
     parser.add_argument("-v", "--verbose",
                         action="store_true", help="be verbose")
     parser.add_argument("-b", "--black",
@@ -23,15 +24,16 @@
     parser.add_argument("-r", "--rainbow",
                         action="store_true", help="rainbow")
     parser.add_argument("files", help="", default="", nargs='+')
     return parser
 
 if 'Darwin' == platform.system():
     is_mac = True
-    BIN = '/usr/local/bin' 
+    #BIN = '/usr/local/bin' # TODO: fix it
+    BIN = BIN_PATH # "/opt/homebrew/bin/"
 else:
     is_mac = False
     BIN = '/usr/bin'
     
 if __name__ == '__main__':
     parser = get_parser()
     args = parser.parse_args()
@@ -50,15 +52,15 @@
                 o = subprocess.Popen(
                     cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
                 out = o.stdout.read().strip().decode()
                 err = o.stderr.read().strip().decode()
                 return out, err
             cmd = BIN + '/pymol -c ' + file + " -d 'print(len([x for x in cmd.get_model().atom]))'"
             out, err = exe(cmd)
-            # ugly
+            # ugly way to get n of atoms
             """
             PyMOL>run ~/work/src/rna-tools/rna_tools/tools/PyMOL4RNA/bucket/mlk4.py
             65
             """
             n = int(out.split('\n')[-1])
             # if something short
             # aaa this will not work for pse files (!)
```

## rna_tools/tools/triplexibility/trx_mutgen.py

```diff
@@ -36,10 +36,10 @@
                  os.system(cmd)
                  f3 = f2.replace('.pdb', '_rpr.pdb')
                  cmd = "rna_pdb_toolsx.py --rpr " + f2 + " > " + f3
                  print(cmd)
                  os.system(cmd)
                  # --files ../../db/triples-all-v2-rpr/*
                  csvf = a + b + c + ".csv"
-                 cmd = "triplexibility2.py -t " + f3 + "  --sort --result " + csvf
+                 cmd = "triplexibility2.py -t " + f3 + " --save --sort --result " + csvf
                  print(cmd)
                  os.system(cmd)
```

## rna_tools/tools/triplexibility/trx_score.py

```diff
@@ -13,15 +13,15 @@
     parser = argparse.ArgumentParser(
         description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
     parser.add_argument("-v", "--verbose",
                         action="store_true", help="be verbose")
     parser.add_argument("--fig", help="", default="fig.py") # nargs='+')
     parser.add_argument("ref", help="", default="") # nargs='+')
     parser.add_argument("-e", "--edge", help="", default= "Triple_cWW_cHS") # nargs='+')
-    parser.add_argument("-t", "--threshold", help="", default="", type=int) # nargs='+')
+    parser.add_argument("-t", "--threshold", help="if t not none then if score > t is 1 else 0", type=int) # nargs='+')
 
     return parser
 
 if __name__ == '__main__':
     parser = get_parser()
     args = parser.parse_args()
 
@@ -91,12 +91,12 @@
 
     df = pd.DataFrame(sl)
     df = df.T
     df.to_csv('sl.csv', index=False)
 
     df = pd.DataFrame(sle)
     df = df.T
-    df.to_csv('sle.csv', index=False)
+    df.to_csv('slexemplary.csv', index=False)
 
     df = pd.DataFrame(sledge)
     df = df.T
     df.to_csv('sledge.csv', index=False)
```

## Comparing `rna_tools-3.8.5.data/scripts/clarna.py` & `rna_tools-3.8.6.data/scripts/clarna.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/clarna_app.py` & `rna_tools-3.8.6.data/scripts/clarna_app.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/diffpdb.py` & `rna_tools-3.8.6.data/scripts/diffpdb.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/pdbs_measure_atom_dists.py` & `rna_tools-3.8.6.data/scripts/pdbs_measure_atom_dists.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/pymol_color_by_conserv.py` & `rna_tools-3.8.6.data/scripts/pymol_color_by_conserv.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/pymol_preview_generator.py` & `rna_tools-3.8.6.data/scripts/pymol_preview_generator.py`

 * *Files 5% similar despite different names*

```diff
@@ -6,14 +6,15 @@
 """
 import sys
 import argparse
 import tempfile
 import os
 import subprocess
 import platform
+from rna_tools.rna_tools_config import BIN_PATH
 
 def get_parser():
     parser = argparse.ArgumentParser(
         description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
     parser.add_argument("-v", "--verbose",
                         action="store_true", help="be verbose")
     parser.add_argument("-b", "--black",
@@ -23,15 +24,16 @@
     parser.add_argument("-r", "--rainbow",
                         action="store_true", help="rainbow")
     parser.add_argument("files", help="", default="", nargs='+')
     return parser
 
 if 'Darwin' == platform.system():
     is_mac = True
-    BIN = '/usr/local/bin' 
+    #BIN = '/usr/local/bin' # TODO: fix it
+    BIN = BIN_PATH # "/opt/homebrew/bin/"
 else:
     is_mac = False
     BIN = '/usr/bin'
     
 if __name__ == '__main__':
     parser = get_parser()
     args = parser.parse_args()
@@ -50,15 +52,15 @@
                 o = subprocess.Popen(
                     cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
                 out = o.stdout.read().strip().decode()
                 err = o.stderr.read().strip().decode()
                 return out, err
             cmd = BIN + '/pymol -c ' + file + " -d 'print(len([x for x in cmd.get_model().atom]))'"
             out, err = exe(cmd)
-            # ugly
+            # ugly way to get n of atoms
             """
             PyMOL>run ~/work/src/rna-tools/rna_tools/tools/PyMOL4RNA/bucket/mlk4.py
             65
             """
             n = int(out.split('\n')[-1])
             # if something short
             # aaa this will not work for pse files (!)
```

## Comparing `rna_tools-3.8.5.data/scripts/pymol_preview_install.py` & `rna_tools-3.8.6.data/scripts/pymol_preview_install.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/random_assignment_of_nucleotides.py` & `rna_tools-3.8.6.data/scripts/random_assignment_of_nucleotides.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/renum_pdb_to_aln.py` & `rna_tools-3.8.6.data/scripts/renum_pdb_to_aln.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_add_chain.py` & `rna_tools-3.8.6.data/scripts/rna_add_chain.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_align_find_core.py` & `rna_tools-3.8.6.data/scripts/rna_align_find_core.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_align_find_seq_in_alignment.py` & `rna_tools-3.8.6.data/scripts/rna_align_find_seq_in_alignment.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_align_foldability.py` & `rna_tools-3.8.6.data/scripts/rna_align_foldability.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_align_get_ss_from_stk.py` & `rna_tools-3.8.6.data/scripts/rna_align_get_ss_from_stk.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_align_seq_to_alignment.py` & `rna_tools-3.8.6.data/scripts/rna_align_seq_to_alignment.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_align_strip_stk.py` & `rna_tools-3.8.6.data/scripts/rna_align_strip_stk.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_alignment_get_species.py` & `rna_tools-3.8.6.data/scripts/rna_alignment_get_species.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_alignment_process_id.py` & `rna_tools-3.8.6.data/scripts/rna_alignment_process_id.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_alignment_r2r.py` & `rna_tools-3.8.6.data/scripts/rna_alignment_r2r.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_cal_rmsd_trafl_plot.py` & `rna_tools-3.8.6.data/scripts/rna_cal_rmsd_trafl_plot.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_calc_evo_rmsd.py` & `rna_tools-3.8.6.data/scripts/rna_calc_evo_rmsd.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_calc_fenergy.py` & `rna_tools-3.8.6.data/scripts/rna_calc_fenergy.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_calc_hbonds.py` & `rna_tools-3.8.6.data/scripts/rna_calc_hbonds.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_calc_inf.py` & `rna_tools-3.8.6.data/scripts/rna_calc_inf.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_calc_rmsd.py` & `rna_tools-3.8.6.data/scripts/rna_calc_rmsd.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_calc_rmsd_all_vs_all.py` & `rna_tools-3.8.6.data/scripts/rna_calc_rmsd_all_vs_all.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_calc_rmsd_biopython.py` & `rna_tools-3.8.6.data/scripts/rna_calc_rmsd_biopython.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_calc_rmsd_multi_targets.py` & `rna_tools-3.8.6.data/scripts/rna_calc_rmsd_multi_targets.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_calc_rmsd_trafl.py` & `rna_tools-3.8.6.data/scripts/rna_calc_rmsd_trafl.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_clanstix.py` & `rna_tools-3.8.6.data/scripts/rna_clanstix.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_clarna_compare.py` & `rna_tools-3.8.6.data/scripts/rna_clarna_compare.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_clarna_run.py` & `rna_tools-3.8.6.data/scripts/rna_clarna_run.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_csv_sort.py` & `rna_tools-3.8.6.data/scripts/rna_csv_sort.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_dG2kd.py` & `rna_tools-3.8.6.data/scripts/rna_dG2kd.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_dot2ct.py` & `rna_tools-3.8.6.data/scripts/rna_dot2ct.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_draw_edges.py` & `rna_tools-3.8.6.data/scripts/rna_draw_edges.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_ec2x.py` & `rna_tools-3.8.6.data/scripts/rna_ec2x.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_filter.py` & `rna_tools-3.8.6.data/scripts/rna_filter.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_gromacs_ready.py` & `rna_tools-3.8.6.data/scripts/rna_gromacs_ready.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_helix_vis.py` & `rna_tools-3.8.6.data/scripts/rna_helix_vis.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_kd2dG.py` & `rna_tools-3.8.6.data/scripts/rna_kd2dG.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_kd2pkd.py` & `rna_tools-3.8.6.data/scripts/rna_kd2pkd.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_md.py` & `rna_tools-3.8.6.data/scripts/rna_md.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_merge_dfs.py` & `rna_tools-3.8.6.data/scripts/rna_merge_dfs.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_merge_two_dfs.py` & `rna_tools-3.8.6.data/scripts/rna_merge_two_dfs.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_minimize.py` & `rna_tools-3.8.6.data/scripts/rna_minimize.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_clashscore.py` & `rna_tools-3.8.6.data/scripts/rna_mq_clashscore.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_collect.py` & `rna_tools-3.8.6.data/scripts/rna_mq_collect.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_dfire.py` & `rna_tools-3.8.6.data/scripts/rna_mq_dfire.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_eSCORE.py` & `rna_tools-3.8.6.data/scripts/rna_mq_eSCORE.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_exp.py` & `rna_tools-3.8.6.data/scripts/rna_mq_exp.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_farfar2.py` & `rna_tools-3.8.6.data/scripts/rna_mq_farfar2.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_farna.py` & `rna_tools-3.8.6.data/scripts/rna_mq_farna.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_plot.py` & `rna_tools-3.8.6.data/scripts/rna_mq_plot.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_rasp.py` & `rna_tools-3.8.6.data/scripts/rna_mq_rasp.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_rna3dcnn.py` & `rna_tools-3.8.6.data/scripts/rna_mq_rna3dcnn.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_rnakb.py` & `rna_tools-3.8.6.data/scripts/rna_mq_rnakb.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_rsRNASP.py` & `rna_tools-3.8.6.data/scripts/rna_mq_rsRNASP.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_score.py` & `rna_tools-3.8.6.data/scripts/rna_mq_score.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_simrna.py` & `rna_tools-3.8.6.data/scripts/rna_mq_simrna.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_mq_wx_collect.py` & `rna_tools-3.8.6.data/scripts/rna_mq_wx_collect.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_ok_mail.py` & `rna_tools-3.8.6.data/scripts/rna_ok_mail.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_pairs2SimRNArestrs.py` & `rna_tools-3.8.6.data/scripts/rna_pairs2SimRNArestrs.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_pairs_diff.py` & `rna_tools-3.8.6.data/scripts/rna_pairs_diff.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_pdb_edit_occupancy_bfactor.py` & `rna_tools-3.8.6.data/scripts/rna_pdb_edit_occupancy_bfactor.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_pdb_merge_structure_with_fragments.py` & `rna_tools-3.8.6.data/scripts/rna_pdb_merge_structure_with_fragments.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_pdb_replace.py` & `rna_tools-3.8.6.data/scripts/rna_pdb_replace.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_pdb_toolsx.py` & `rna_tools-3.8.6.data/scripts/rna_pdb_toolsx.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_pk_simrna_to_one_line.py` & `rna_tools-3.8.6.data/scripts/rna_pk_simrna_to_one_line.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_pkd2dG.py` & `rna_tools-3.8.6.data/scripts/rna_pkd2dG.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_pkd2kd.py` & `rna_tools-3.8.6.data/scripts/rna_pkd2kd.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_plot_boxplotlike.py` & `rna_tools-3.8.6.data/scripts/rna_plot_boxplotlike.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_plot_dendogram.py` & `rna_tools-3.8.6.data/scripts/rna_plot_dendogram.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_plot_density.py` & `rna_tools-3.8.6.data/scripts/rna_plot_density.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_plot_heatmap.py` & `rna_tools-3.8.6.data/scripts/rna_plot_heatmap.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_plot_hist.py` & `rna_tools-3.8.6.data/scripts/rna_plot_hist.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_prediction_significance.py` & `rna_tools-3.8.6.data/scripts/rna_prediction_significance.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_refinement.py` & `rna_tools-3.8.6.data/scripts/rna_refinement.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_rosetta_check_progress.py` & `rna_tools-3.8.6.data/scripts/rna_rosetta_check_progress.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_rosetta_cluster.py` & `rna_tools-3.8.6.data/scripts/rna_rosetta_cluster.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_rosetta_extract_lowscore_decoys.py` & `rna_tools-3.8.6.data/scripts/rna_rosetta_extract_lowscore_decoys.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_rosetta_head.py` & `rna_tools-3.8.6.data/scripts/rna_rosetta_head.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_rosetta_min.py` & `rna_tools-3.8.6.data/scripts/rna_rosetta_min.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_rosetta_n.py` & `rna_tools-3.8.6.data/scripts/rna_rosetta_n.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_rosetta_run.py` & `rna_tools-3.8.6.data/scripts/rna_rosetta_run.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_sali2dotbracket.py` & `rna_tools-3.8.6.data/scripts/rna_sali2dotbracket.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_secondary_structure_prediction.py` & `rna_tools-3.8.6.data/scripts/rna_secondary_structure_prediction.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_simrna_cluster.py` & `rna_tools-3.8.6.data/scripts/rna_simrna_cluster.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_simrna_extract.py` & `rna_tools-3.8.6.data/scripts/rna_simrna_extract.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_simrna_lowest.py` & `rna_tools-3.8.6.data/scripts/rna_simrna_lowest.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_simrnaweb_download_job.py` & `rna_tools-3.8.6.data/scripts/rna_simrnaweb_download_job.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_ss_get_bps.py` & `rna_tools-3.8.6.data/scripts/rna_ss_get_bps.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_tools_which.py` & `rna_tools-3.8.6.data/scripts/rna_tools_which.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_traj_n.py` & `rna_tools-3.8.6.data/scripts/rna_traj_n.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_traj_nohetm.py` & `rna_tools-3.8.6.data/scripts/rna_traj_nohetm.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_traj_range.py` & `rna_tools-3.8.6.data/scripts/rna_traj_range.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_workflow.py` & `rna_tools-3.8.6.data/scripts/rna_workflow.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/rna_x3dna.py` & `rna_tools-3.8.6.data/scripts/rna_x3dna.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/select_seq_fromBLAStn_6outfm.py` & `rna_tools-3.8.6.data/scripts/select_seq_fromBLAStn_6outfm.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/triplexibility.py` & `rna_tools-3.8.6.data/scripts/triplexibility.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/triplexibility2.py` & `rna_tools-3.8.6.data/scripts/triplexibility2.py`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.data/scripts/trx_mutgen.py` & `rna_tools-3.8.6.data/scripts/trx_mutgen.py`

 * *Files 2% similar despite different names*

```diff
@@ -36,10 +36,10 @@
                  os.system(cmd)
                  f3 = f2.replace('.pdb', '_rpr.pdb')
                  cmd = "rna_pdb_toolsx.py --rpr " + f2 + " > " + f3
                  print(cmd)
                  os.system(cmd)
                  # --files ../../db/triples-all-v2-rpr/*
                  csvf = a + b + c + ".csv"
-                 cmd = "triplexibility2.py -t " + f3 + "  --sort --result " + csvf
+                 cmd = "triplexibility2.py -t " + f3 + " --save --sort --result " + csvf
                  print(cmd)
                  os.system(cmd)
```

## Comparing `rna_tools-3.8.5.data/scripts/trx_score.py` & `rna_tools-3.8.6.data/scripts/trx_score.py`

 * *Files 2% similar despite different names*

```diff
@@ -13,15 +13,15 @@
     parser = argparse.ArgumentParser(
         description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
     parser.add_argument("-v", "--verbose",
                         action="store_true", help="be verbose")
     parser.add_argument("--fig", help="", default="fig.py") # nargs='+')
     parser.add_argument("ref", help="", default="") # nargs='+')
     parser.add_argument("-e", "--edge", help="", default= "Triple_cWW_cHS") # nargs='+')
-    parser.add_argument("-t", "--threshold", help="", default="", type=int) # nargs='+')
+    parser.add_argument("-t", "--threshold", help="if t not none then if score > t is 1 else 0", type=int) # nargs='+')
 
     return parser
 
 if __name__ == '__main__':
     parser = get_parser()
     args = parser.parse_args()
 
@@ -91,12 +91,12 @@
 
     df = pd.DataFrame(sl)
     df = df.T
     df.to_csv('sl.csv', index=False)
 
     df = pd.DataFrame(sle)
     df = df.T
-    df.to_csv('sle.csv', index=False)
+    df.to_csv('slexemplary.csv', index=False)
 
     df = pd.DataFrame(sledge)
     df = df.T
     df.to_csv('sledge.csv', index=False)
```

## Comparing `rna_tools-3.8.5.dist-info/LICENSE` & `rna_tools-3.8.6.dist-info/LICENSE`

 * *Files identical despite different names*

## Comparing `rna_tools-3.8.5.dist-info/METADATA` & `rna_tools-3.8.6.dist-info/METADATA`

 * *Files 0% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: rna-tools
-Version: 3.8.5
+Version: 3.8.6
 Summary: a toolbox to analyze structures and simulations of RNA
 Home-page: https://github.com/mmagnus/rna-tools
 Author: Marcin Magnus
 Author-email: mag_dex@o2.pl
 License: GPLv3
 Platform: UNKNOWN
 Description-Content-Type: text/markdown
```

### html2text {}

```diff
@@ -1,8 +1,8 @@
-Metadata-Version: 2.1 Name: rna-tools Version: 3.8.5 Summary: a toolbox to
+Metadata-Version: 2.1 Name: rna-tools Version: 3.8.6 Summary: a toolbox to
 analyze structures and simulations of RNA Home-page: https://github.com/
 mmagnus/rna-tools Author: Marcin Magnus Author-email: mag_dex@o2.pl License:
 GPLv3 Platform: UNKNOWN Description-Content-Type: text/markdown Requires-Dist:
 numpy Requires-Dist: biopython Requires-Dist: progressbar2 Requires-Dist:
 csvsort Requires-Dist: urllib3 Requires-Dist: versioneer Requires-Dist:
 configparser Requires-Dist: icecream
                             ************ rrnnaa--ttoooollss ************
```

## Comparing `rna_tools-3.8.5.dist-info/RECORD` & `rna_tools-3.8.6.dist-info/RECORD`

 * *Files 3% similar despite different names*

```diff
@@ -1,28 +1,28 @@
 rna_tools/BlastPDB.py,sha256=lxDLV6Mo8bxPIMvwTU5qo5tj4dA3AbZCeZ7sgACSsZ0,1305
 rna_tools/RfamSearch.py,sha256=4-R5ovtJcyOn3FZ48jct1ESBFy19EJ7IvM_xFkM1H6g,3815
 rna_tools/SecondaryStructure.py,sha256=Lmpg_q-Dd5yTaeM-ffn0jcChnFUVvSV-CWjzKs0P9R0,3819
 rna_tools/Seq.py,sha256=R333PZcFNeP2E88jA7jGKDDgeY80jJFRqfeLcj_l1m4,27464
 rna_tools/__init__.py,sha256=JOl41NviMN-qDV0Z8ZPmhNIxvgyauGGJHdB4A-8MhqM,93
-rna_tools/_version.py,sha256=ZRnrRf-vaPybV__ocoRUS3MOEFSVEbARKJhjRr0O20k,496
+rna_tools/_version.py,sha256=hCf_wfmz06Zw3UCyGC98_u7yD2i85WvPvOeDjqNiUzo,496
 rna_tools/rna_calc_fenergy.py,sha256=Dx32hkd6KOAcjw6ILRjpSQkPjBbHbhGNFIGU_9pbP50,2466
 rna_tools/rna_dot2ct.py,sha256=mBhknxHAYdU4xglOAp_BSvxyVYNofNXrdvXd2xVd27M,2448
 rna_tools/rna_pdb_replace.py,sha256=WbWBkdZg0yHoIclj-64cKfafHj8JZp-xu4ri0WgNszo,1756
 rna_tools/rna_pdb_toolsx.py,sha256=neQ_es_iPoo0G1b6I79duo4LcvX3p2DjrUsMheM0fzE,41423
 rna_tools/rna_secondary_structure_prediction.py,sha256=FvXkU_GlLUOij1P8t0zfVl1RHSOoudtpKRbMxfFBVz8,2172
 rna_tools/rna_simrnaweb_download_job.py,sha256=T_Nfwmnq-JS5d2-UOfEge0TrQINz6sMIUGXcHKPGycI,12473
-rna_tools/rna_tools_config.py,sha256=fZ1wQls5vtxzVBZEO2IbkAlXxsep1gjXDSr-RQ8seSM,3713
+rna_tools/rna_tools_config.py,sha256=QVUH3xorH_ofAuRtx8Khh8YiZXK4aUJ34pBebbST8AU,3731
 rna_tools/rna_tools_lib.py,sha256=nDSPTmtjwPBvqfblD7HAvr6dOEMxSjGMaqhLKGVmL0E,82236
 rna_tools/rna_tools_test_all.py,sha256=OsszeEz9cY7GC32C0puZSV74jW942oYrIwrXFILCnvQ,1342
 rna_tools/update_readme.py,sha256=15nJY4esqygRtn97KGBXmjaoQNiKVCv3neux2mAKYuQ,709
 rna_tools/tools/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 rna_tools/tools/ClashCalc/ClashCalc.py,sha256=kZYhh4QplRwtpK2oX14NV9DymzleQfQ9CqcPGOzhaOI,3080
 rna_tools/tools/ClashCalc/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
-rna_tools/tools/PyMOL4RNA/PyMOL4RNA.py,sha256=xYMmrAu1IXoBzXTs_jVr9eNrkxFvwm65IM105CMZX90,35375
-rna_tools/tools/PyMOL4RNA/PyMOL4Spliceosome.py,sha256=NQckekkbx1NrVwGKDB9bNpicJXDg4z5S7PB8h1PLP6g,13336
+rna_tools/tools/PyMOL4RNA/PyMOL4RNA.py,sha256=AwZIhN-T8e4iXBlR0mPnfxXQ-nalZcJW3gSYIRoAWxs,38955
+rna_tools/tools/PyMOL4RNA/PyMOL4Spliceosome.py,sha256=tAPqBpXi2v27RFzWQ6ZxqoYt6jjOaja0JwqXbengAwE,15676
 rna_tools/tools/PyMOL4RNA/Spliceosome_PyMOL.py,sha256=zUWhrVzUWtUSJb7PMXGT5DjJuniflSKMdkoNMfZb9Vk,6836
 rna_tools/tools/PyMOL4RNA/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 rna_tools/tools/PyMOL4RNA/code_for_color_spl.py,sha256=Z4DdY83aZZWSX_VqwSfjRFRowRmSuaCZYt2H14uxg1g,15836
 rna_tools/tools/PyMOL4RNA/code_for_spl.py,sha256=Hevn5lCcZFVI0XRMEKMj2WOjboSzGzkAMiRggs4yKHA,7898
 rna_tools/tools/PyMOL4RNA/external_flatten_object.py,sha256=qWRyGfQmHyLuTPYll-oXFOo3pFg0ieRcTn19cWP1-Fs,11652
 rna_tools/tools/PyMOL4RNA/rna_draw_edges.py,sha256=qIX_lf8zHT20L5e7a7UJmd-IevQ-L6GNMW7bwJtBKTI,5431
 rna_tools/tools/PyMOL4RNA/rnapuzzles.py,sha256=sq0LS6y59baBOLB_k335ZLnLOlRwSYgXp_tzohkTt2k,3208
@@ -262,15 +262,15 @@
 rna_tools/tools/plotting/rna_plot_dendogram.py,sha256=RdGCSrIeveiuuXU60hE95cARNkzVDty7XN2OVhfySpY,1623
 rna_tools/tools/plotting/rna_plot_density.py,sha256=UTLZaj3hjmE6nSgjfdYSwoGWpuYaJQuItTuNSm1v010,1793
 rna_tools/tools/plotting/rna_plot_heatmap.py,sha256=Mn-WATNNyVG-tLTmQZ9QCU5HJxWiUfPkbFvC885DrZw,2037
 rna_tools/tools/plotting/rna_plot_hist.py,sha256=hKs2ykUwazHcJkQkDFzHUWQXXFC-2vtX9Bvy7zwxolE,1959
 rna_tools/tools/pymol_color_by_conserv/pymol_color_by_conserv.py,sha256=7dVAgXb4Pz2IL8KqnlClrfbhofWc4b0IA95puOgRnqA,2104
 rna_tools/tools/pymol_drawing/__init__.py,sha256=AbpHGcgLb-kRsJGnwFEktk7uzpZOCcBY74-YBdrKVGs,1
 rna_tools/tools/pymol_drawing/pymol_drawing.py,sha256=2GFSofXonpurKZnYvuAOvwCPpx_c6PVhJXKsoe1n4Hk,3539
-rna_tools/tools/pymol_preview_generator/pymol_preview_generator.py,sha256=w42xkqTOT3f_9Cegj97IVGAGFpQDxX-75nih76NS7_Y,3241
+rna_tools/tools/pymol_preview_generator/pymol_preview_generator.py,sha256=YuRKArBRdWDdz54trdJBLfwO0H1zlPEsgHcA0mSO_Xw,3368
 rna_tools/tools/pymol_preview_generator/pymol_preview_install.py,sha256=REmgEnu035eOUB_ht6q5FAbF2oxIgbA5VgIcfF6kT1M,996
 rna_tools/tools/renum_pdb_to_aln/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 rna_tools/tools/renum_pdb_to_aln/renum_pdb_to_aln.py,sha256=UIvE8XgNtjFfwDxDiJ8Pl41miKQ2Dhzg102EpjPgz8Y,3954
 rna_tools/tools/rna_alignment/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 rna_tools/tools/rna_alignment/random_assignment_of_nucleotides.py,sha256=9bxlSvfZY5l3WsUWq28BqXUTabJ23awpzQILACV0fZs,10159
 rna_tools/tools/rna_alignment/rna_align_distance_to_seq.py,sha256=CxPriVff9nBdS7k1pml1dvVNP-eArz7sYGelDR6WDTo,2877
 rna_tools/tools/rna_alignment/rna_align_find_core.py,sha256=6KgpAeL80pxNUr6iwLzFt2Fy6uMXwp60Jo9A7bEViQc,650
@@ -371,16 +371,18 @@
 rna_tools/tools/simrna_trajectory/rna_simrna_lowest.py,sha256=AxF-1KC_ijiWqCAWXSg2jZzwqRcJckIRi6G9DU9hcVM,1278
 rna_tools/tools/simrna_trajectory/rna_simrna_subset.py,sha256=wQzdZac2WpF5HnKVQjUIvOu7FsYwt2P-5qnkAcLlJm8,723
 rna_tools/tools/simrna_trajectory/simrna_trajectory.py,sha256=_vezn07UuBbQA1_h0Nu8mSQwpcJ4w-SVJvMnDswTWmc,10181
 rna_tools/tools/simrna_trajectory/test_rmsd.py,sha256=BRCuo3vhFy2dAIujh6hj5-OHgcyOB4dgau3Apuwkr8Y,406
 rna_tools/tools/simrna_trajectory/test_rna_simrna.py,sha256=kQqOx02PgsV8TtO8dGN9voQb_Q73QP11AWBNcApqdHk,232
 rna_tools/tools/triplexibility/triplexibility.py,sha256=10z-xePKLumPEBj4jS2a8A9kUQnai04eAUXNQ7Aolmw,16459
 rna_tools/tools/triplexibility/triplexibility2.py,sha256=NbB5CpO93sQe60DyM6_J4iWY0y8XWczFo1pzMnrQAj4,13684
-rna_tools/tools/triplexibility/trx_mutgen.py,sha256=tqj5ZYHdA0VtZlAePhzteo0RBSChW17sRwetXJfHaiM,1551
-rna_tools/tools/triplexibility/trx_score.py,sha256=FeMVV1m3UizG1u3Iu7bwCtOF1m9LTCDXROw_aj_-EOQ,3295
+rna_tools/tools/triplexibility/trx_mutagen_allvsall.py,sha256=S8qL3FXAM4esAlhhLb5wY7DKpfq6jPjuGXrKaGvM7tc,1671
+rna_tools/tools/triplexibility/trx_mutgen.py,sha256=FJQFznfZ4SLCmXGbTtkWHtME0oLGau1CR1mGgX1ZcZA,1557
+rna_tools/tools/triplexibility/trx_score.py,sha256=sWI_rPiRaWHR6tMuvKTYMUM4AIY07unHzoJ3rYLKR8I,3334
+rna_tools/tools/triplexibility/trx_score_allvsall.py,sha256=X6YbraZF4-DODUakLLF6S4yVqGNCgf6NaVpqWBO4zNM,3288
 rna_tools/tools/webserver-engine/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 rna_tools/tools/webserver-engine/checker.py,sha256=SDaW3UogS6wHjjjFn27LmYxANLn5LHMr5Zz6M5iVWIY,2640
 rna_tools/tools/webserver-engine/cleanup.py,sha256=CiLVtMUhbfyivs-G4Zse3Ra4Z7xzOGiDNYQe-bw-odY,4553
 rna_tools/tools/webserver-engine/daemon.py,sha256=kDbtMCAidQ2-g15gja09EONLnywoODsGRsqVNdS2yeQ,11839
 rna_tools/tools/webserver-engine/manage.py,sha256=sEAKPtry6YPjzk5RXxFv8vfKs6j2RSNzLqbnVNZv0Cg,246
 rna_tools/tools/webserver-engine/sendmail.py,sha256=R-VZHLhGB33qZsdwb68TI0gKC76FKot8Fn16rBVembs,2712
 rna_tools/tools/webserver-engine/wsgi.py,sha256=H1k2ZFj_ocirHI0THPWyFgGu9ivQGclTQpwH-MD9TDI,632
@@ -389,121 +391,123 @@
 rna_tools/tools/webserver-engine/app/models.py,sha256=K6mkEa-rPwK37cI57ksY5FrdDoRITclf32Qwg5fHysk,8356
 rna_tools/tools/webserver-engine/app/views.py,sha256=le4_14N_nKNHEmbkIWU_7EpjgWmQHqKM0AgjloCcfE4,20146
 rna_tools/tools/webserver-engine/web/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 rna_tools/tools/webserver-engine/web/settings.py,sha256=B86rN7-9SCcoitXytG9PhvsFYhUu5wvi4Ckalcs6qIA,6078
 rna_tools/tools/webserver-engine/web/urls.py,sha256=qdiLgeVEDme-klygBvSeFyP7udvU6qqM7lq5q6lXmog,1943
 rna_tools/tools/webserver-engine/web/wsgi.py,sha256=vi2hE27YLitbiqb1bJ1dZ40sEU_AQ4JRyvd8_xfpVPo,1410
 rna_tools/tools/workflow/rna_workflow.py,sha256=AOkctbBCsvpKIetd-mqzsSxlG3zSuIlZBgjOgPfivvw,2638
-rna_tools-3.8.5.data/scripts/clarna.py,sha256=IV_CzDAy1OXiBLrF1NyUoSt0MCDl3RE9SH4D1wIpvsc,46835
-rna_tools-3.8.5.data/scripts/clarna_app.py,sha256=vJiFj1Ue-0oIEFj6S4cyQpjIC4e7yRIiJE5WUlunKaQ,5672
-rna_tools-3.8.5.data/scripts/diffpdb.py,sha256=g4wZz4SEw0gMfhoB3FmvNdjuuaeBwgivk-hPqEDVQxY,4636
-rna_tools-3.8.5.data/scripts/pdbs_measure_atom_dists.py,sha256=b37gL1BH63zACS8u37JuKNDBQ_Ex3UeaFaSrPNZgkjc,5979
-rna_tools-3.8.5.data/scripts/pymol_color_by_conserv.py,sha256=pmoDOvvFyv0geoW6_eCcGjjOyXs8ChqtVR_jYPujVL8,2091
-rna_tools-3.8.5.data/scripts/pymol_preview_generator.py,sha256=gV4gY5oVRicnQSIOiYau3xQH2lZtk5E-8GjxkOYZEyY,3227
-rna_tools-3.8.5.data/scripts/pymol_preview_install.py,sha256=fisyGVgRuBgMnxAJATs0QXIYoljVQymLPQGUnac9Hvs,983
-rna_tools-3.8.5.data/scripts/random_assignment_of_nucleotides.py,sha256=EuxMreZF--cjmCQ1CpErTEHly53YE8qz0H7gXbvd2fk,10146
-rna_tools-3.8.5.data/scripts/renum_pdb_to_aln.py,sha256=DCdpoZw-eTWHapARasFSlvBunNiAfGfhhQdlcnqkKtA,3941
-rna_tools-3.8.5.data/scripts/reverse.py,sha256=tjFE74uGoY7__Ms3Wnir4xNHiS-qFt_f4hTXCgFsyxE,141
-rna_tools-3.8.5.data/scripts/rna_add_chain.py,sha256=aQQOz-5oZU_jd3CdzZjDo35gxkrhNmqvSb3aIRTxKXk,2381
-rna_tools-3.8.5.data/scripts/rna_align_find_core.py,sha256=upZeHDrPOjj3IQA8nnE3jPuvsSRPQI7v8CNVNI-5n_o,637
-rna_tools-3.8.5.data/scripts/rna_align_find_seq_in_alignment.py,sha256=O_hMeconRaHufOc1PQReQn3yKTTSJ0K8tcKskmd5jhM,2161
-rna_tools-3.8.5.data/scripts/rna_align_foldability.py,sha256=ik1GiMmDp9c8VI1FAQp-fsHTJxvpl0yZMhfyMPaEcC4,13320
-rna_tools-3.8.5.data/scripts/rna_align_get_ss_from_stk.py,sha256=3Ud__SibvdnXoKv_24sJGJ6o9HpTtAiHkPrS3xLs5oE,1327
-rna_tools-3.8.5.data/scripts/rna_align_seq_to_alignment.py,sha256=wFqE4aa0T1BlXQUilJ9_YSk4fNb1boQzORLUkgxswyg,1662
-rna_tools-3.8.5.data/scripts/rna_align_strip_stk.py,sha256=nKFWOrCQtdqeI5vx4VdBwEQScCaxdnB3fQwO0h7NSNw,712
-rna_tools-3.8.5.data/scripts/rna_alignment_get_species.py,sha256=5-oPt_9nKK8m8NCQo6yb4ckuP-5WJdvXuRfrD7e0jQw,9708
-rna_tools-3.8.5.data/scripts/rna_alignment_process_id.py,sha256=m38TLU7iXvDWbZzV5sWhPvcQUuoBcKHGO4Zjm0EG2Xw,1919
-rna_tools-3.8.5.data/scripts/rna_alignment_r2r.py,sha256=dnXLDi1f54KyIO4yU95gbi_K-Phv2qmIr-B7QdCNAww,2301
-rna_tools-3.8.5.data/scripts/rna_cal_rmsd_trafl_plot.py,sha256=mCYsn3sYdBGKx0t6wmatOB9Gseb3omsd6vTOukZqQ1M,1318
-rna_tools-3.8.5.data/scripts/rna_calc_evo_rmsd.py,sha256=1BVTZlb8CEQYSyvtst5aGdpFv2U_ukbXB7e6cYenP80,6805
-rna_tools-3.8.5.data/scripts/rna_calc_fenergy.py,sha256=-8F2yJyLP9XYpu29MDa5fPGE5HVmMqNWHoZd9tBE4GE,2453
-rna_tools-3.8.5.data/scripts/rna_calc_hbonds.py,sha256=pjpBUQdMg61k794BPMHkH_krYU8tHhUCsM696B2kevM,2435
-rna_tools-3.8.5.data/scripts/rna_calc_inf.py,sha256=ZuZtErEKswe7qK8Lw_vP-UZH47i2mo3p17GEO7M9L2w,9995
-rna_tools-3.8.5.data/scripts/rna_calc_rmsd.py,sha256=EW-ZAiWxPeI7YMqmmibg55joK5VaLE28oX6qilY0uR8,13430
-rna_tools-3.8.5.data/scripts/rna_calc_rmsd_all_vs_all.py,sha256=jWe90O2PskmF1LROkQRPb5WA_PHB-0ptUICzH3Bv1dg,4186
-rna_tools-3.8.5.data/scripts/rna_calc_rmsd_biopython.py,sha256=wWoNFKGzBP6VJ023_SSDlW5PDgCqHaUQ-X_YH96TCyk,11464
-rna_tools-3.8.5.data/scripts/rna_calc_rmsd_multi_targets.py,sha256=38aZ3LM-GQGxlKEQT2-BRwSPPAlpv7M2H5ldDYrq3ls,3965
-rna_tools-3.8.5.data/scripts/rna_calc_rmsd_trafl.py,sha256=KrNAmO3b1mVY7LS2JbFbGXdBH5dn57ZKNitE-acsfw4,7157
-rna_tools-3.8.5.data/scripts/rna_clanstix.py,sha256=T8yf1aiBD6OEMzlp5dEhzZZ-Tu67AxEZz7XuQJO73ks,21520
-rna_tools-3.8.5.data/scripts/rna_clarna_compare.py,sha256=s681bN2ZdBJL_xmJaSbGkvV4N5f3N_DlGUHQAoTeXLc,48049
-rna_tools-3.8.5.data/scripts/rna_clarna_run.py,sha256=drvlZbd8H36DfXmDD9ahSEiDawJE2tH6GC5UG5VbKwQ,45334
-rna_tools-3.8.5.data/scripts/rna_csv_sort.py,sha256=fTC4Piv8uyWgUKdfvBVPfrtTuuAmjoCBcSuHgrPJMlU,903
-rna_tools-3.8.5.data/scripts/rna_dG2kd.py,sha256=kHd9ayXQ9CodgbeieS-Us3kWvjYO6qoBLc-zbxpHFc0,941
-rna_tools-3.8.5.data/scripts/rna_dot2ct.py,sha256=ePiy24VuPd0lwT206lw0F_yRK4YXqkkzqeA7_RdRkgM,2435
-rna_tools-3.8.5.data/scripts/rna_draw_edges.py,sha256=sKhLKw8AhYG9kcP668JV9XBnDimQi6G1fQkFV01PRjE,5418
-rna_tools-3.8.5.data/scripts/rna_ec2x.py,sha256=10qP2UBykjb3R-Sm1FRvXfonfjj18yW2ErDjbQYI1to,2793
-rna_tools-3.8.5.data/scripts/rna_filter.py,sha256=ZA6mMhugdJ5fihZwh6hCKC0kXVPgJeTCc52ccZWIjVQ,11893
-rna_tools-3.8.5.data/scripts/rna_gromacs_ready.py,sha256=R8jv59cP-vekhAzRWLMwId2LY1Fyg0jTR5Pxp6292eI,919
-rna_tools-3.8.5.data/scripts/rna_helix_vis.py,sha256=xVMDR-y1ECtIFMzUoP46Od1Aevb9ad5tERvMU4P8Z3s,4055
-rna_tools-3.8.5.data/scripts/rna_kd2dG.py,sha256=-zQjAAvconTIEjBs4wQRgPDJu3Z0aCG6qQsQGP9-79U,999
-rna_tools-3.8.5.data/scripts/rna_kd2pkd.py,sha256=F-vYL0WYaINEP1xCNDmrzRk2uhrqmeDTeW6AGcBhpRs,705
-rna_tools-3.8.5.data/scripts/rna_md.py,sha256=x3gjaopWueV2cEyHJPzokOxna9jNjHdSQdcS3s9rIWU,5927
-rna_tools-3.8.5.data/scripts/rna_merge_dfs.py,sha256=7ECjOGjJWki46S-oFPb-5UHexShtMjF2pvTDs9O_t-o,1194
-rna_tools-3.8.5.data/scripts/rna_merge_two_dfs.py,sha256=tic_ReQLT64DzjOsEnQRqBhdk4waTCcv1C81jKpLclQ,8069
-rna_tools-3.8.5.data/scripts/rna_minimize.py,sha256=48fPHFs-BxBJvP29HF5SR6bjizUmC8XWVcdJInDePGg,3263
-rna_tools-3.8.5.data/scripts/rna_mq_clashscore.py,sha256=bkYyFihDHp0YXBfeZprRFlJb94wLbuim_XW-DsfivoM,1169
-rna_tools-3.8.5.data/scripts/rna_mq_collect.py,sha256=QmJQQMS8EUbg4tVeqWGH-PAs6hVs-wloQWuLJU3lohw,24580
-rna_tools-3.8.5.data/scripts/rna_mq_dfire.py,sha256=4EtHQcKlRihoNBoTSCHU065BVe-GwUj2933bj7Lxroo,1216
-rna_tools-3.8.5.data/scripts/rna_mq_eSCORE.py,sha256=JIDGIi5FywNDrvfcNL8nMbGhDWV6COUgUekYfvR8Nf0,908
-rna_tools-3.8.5.data/scripts/rna_mq_exp.py,sha256=xONillBR5sYy-v2gRS727TRGX5AMTvliyDDmVXS1MzQ,2428
-rna_tools-3.8.5.data/scripts/rna_mq_farfar2.py,sha256=-Mmrmu9j5Ceb2Uv9IBT9pT1nXpIRmkHqmskruWytkP0,2435
-rna_tools-3.8.5.data/scripts/rna_mq_farna.py,sha256=5eyTBKEzJlc2F-0BUIwJlfHIdhu-uVeRS7yE8HoOHhU,5044
-rna_tools-3.8.5.data/scripts/rna_mq_plot.py,sha256=4cPhtRdUt2tTeSpaVFjCiNib5l1veY5rYEVp-nlsOl8,1428
-rna_tools-3.8.5.data/scripts/rna_mq_rasp.py,sha256=h6Xu7LLCWZNBy_KYq-gD4ONinDuVUUHoy_zakgN2FM8,1371
-rna_tools-3.8.5.data/scripts/rna_mq_remote.sh,sha256=fojmce5HxCKK7kNhjXOWqwRUTKIybQefR5TEhe3VovM,290
-rna_tools-3.8.5.data/scripts/rna_mq_rna3dcnn.py,sha256=TlYYi75HPHRaigzOBqar8OL6gHqJLz5o8OQbT5FNotA,949
-rna_tools-3.8.5.data/scripts/rna_mq_rnakb.py,sha256=gRtEOHZJ2MHW6n-Qau7LLMIdbi6a6VS8i6NZY1fua0E,1707
-rna_tools-3.8.5.data/scripts/rna_mq_rsRNASP.py,sha256=jSvYXnUvfoqq1m5_z69IGi6m8O4cTgaSSFsccbHEMRI,912
-rna_tools-3.8.5.data/scripts/rna_mq_score.py,sha256=zstVn5pykuI9jif86gNAgTPWKX-JgUX_0woLAb_8t9Q,4635
-rna_tools-3.8.5.data/scripts/rna_mq_simrna.py,sha256=V0qRvy1CkO1wRNm7dKk2OPmy71epEdz6q-pzQfadhwg,1546
-rna_tools-3.8.5.data/scripts/rna_mq_wx_collect.py,sha256=adJxw1zTjL6KKS9YpIbjfDvnuznfQU-BMrZH6gIfuQs,2713
-rna_tools-3.8.5.data/scripts/rna_ok,sha256=UMqzXXeuGbmgoRG9O-lR1EwqitZsXYE9AWinJg2TCGQ,64
-rna_tools-3.8.5.data/scripts/rna_ok_mail.py,sha256=iMLKNcD0xQmp7lc9j8OS5IND2Jc0TXtpNqksj7BSQLw,1369
-rna_tools-3.8.5.data/scripts/rna_pairs2SimRNArestrs.py,sha256=KQmaGk62dAOT77ZoR1QXbXQOLcOfeWC-l-udO8isljE,1789
-rna_tools-3.8.5.data/scripts/rna_pairs_diff.py,sha256=Q-_PhoPcuY-T1pbzKG6BTic6dZtXdPCslHEfo0v6TbE,1384
-rna_tools-3.8.5.data/scripts/rna_pdb_edit_occupancy_bfactor.py,sha256=jd_SqILeGndYjUVxX_WJQqiNTYs6OqUl-P3K6iCskfI,6145
-rna_tools-3.8.5.data/scripts/rna_pdb_merge_into_one.py,sha256=t0-CAwi6X9bfeNnGWzwguYW0geQi8yZs3-sTZMVkbFo,415
-rna_tools-3.8.5.data/scripts/rna_pdb_merge_structure_with_fragments.py,sha256=8skR7gd4jyP6IeTGhphw93sL_phn_AyQLAUzpWn5yAI,4905
-rna_tools-3.8.5.data/scripts/rna_pdb_replace.py,sha256=AGg8V-Yk0C3R8tzJzjQkBsS-uCbSyYVZsPCHAyoSt4s,1743
-rna_tools-3.8.5.data/scripts/rna_pdb_toolsx.py,sha256=WLEir8d9dlpGNBDEuuvdFrafOe-TdOOTebGuyQf76m8,41410
-rna_tools-3.8.5.data/scripts/rna_pk_simrna_to_one_line.py,sha256=qBy-nO1BCLXsqOWVjlg2aS8_iX_I1I-tdQKv4B2tepE,1039
-rna_tools-3.8.5.data/scripts/rna_pkd2dG.py,sha256=FYIZk8Mw3QSBklgR0O3ZW8pkzKYChcCOFSvOcnKFuRc,886
-rna_tools-3.8.5.data/scripts/rna_pkd2kd.py,sha256=qHIqibImjh6Sl7Y9hpsUsyGXf27qy6R0Q1aaFyAlXuY,783
-rna_tools-3.8.5.data/scripts/rna_plot_boxplotlike.py,sha256=E0c3sZIqexnl5deDPeCZ1uWriXjE4Il8pTxvKkbx5m4,2263
-rna_tools-3.8.5.data/scripts/rna_plot_dendogram.py,sha256=CVjg13_1ptnvAgzW7mGOb8iZgRO-Bf3khEPI02PzJT4,1610
-rna_tools-3.8.5.data/scripts/rna_plot_density.py,sha256=j3aNK2qQfgvANOyu9HVH9og9yx_CmIXbZp86Hn9aGOQ,1780
-rna_tools-3.8.5.data/scripts/rna_plot_heatmap.py,sha256=FdK_h6F-7XdysaLyXt8BbffiCVl695Ln_MmS68-Eimg,2024
-rna_tools-3.8.5.data/scripts/rna_plot_hist.py,sha256=ao_22HSl_ssyd9uYKIMOWS0ThkRkTh0laW2gYFx0IVc,1946
-rna_tools-3.8.5.data/scripts/rna_prediction_significance.py,sha256=q-PpYRCHYj85WU_-O8oB-GqROXLMxRYL0pXcJU83jEM,1079
-rna_tools-3.8.5.data/scripts/rna_refinement.py,sha256=yWhOHjD7QwUoPebuOL6vtb_2hKkrJb4duoHwjmKpd9U,7029
-rna_tools-3.8.5.data/scripts/rna_rosetta_check_progress.py,sha256=ZnnScybmyhAE4_GT1j7jmYvIJgb0-yeZbC8vgXvayXU,6070
-rna_tools-3.8.5.data/scripts/rna_rosetta_cluster.py,sha256=-ME8xtu3tZxh76SMQf4JKLAdJKrUoPhiXPR9yQSrf0M,5462
-rna_tools-3.8.5.data/scripts/rna_rosetta_extract_lowscore_decoys.py,sha256=rtOoYdXU1nH5zs-CQpq5jz6kQ2QtTSa_CXtyfG5n43A,1523
-rna_tools-3.8.5.data/scripts/rna_rosetta_head.py,sha256=puKYbGWowXfsWBkmvAKCYeZobTGbRXYWPfDgF_t-xmE,3977
-rna_tools-3.8.5.data/scripts/rna_rosetta_min.py,sha256=-_As6TK-nHddrBWzE_j6_ZESZ0DpL3bM3K3AZv41Vvo,4903
-rna_tools-3.8.5.data/scripts/rna_rosetta_n.py,sha256=UnGxNcLSXwRMmHMt38K9JT0MhnCvFKUpAM_F3ogpP0E,1045
-rna_tools-3.8.5.data/scripts/rna_rosetta_run.py,sha256=-hfi00IlE_1k43xvbdpDrNEFI1aYTtaPCTxB9DbzG6Q,10095
-rna_tools-3.8.5.data/scripts/rna_sali2dotbracket.py,sha256=wNjnSqr4nIEuUN4ZAmzd7hAPxnkWFXx11p9jSPfx-PA,3450
-rna_tools-3.8.5.data/scripts/rna_secondary_structure_prediction.py,sha256=GvfyLqtsyBGxmpv0bLaL7UcT5G95ppnebEVyPK2mjOg,2159
-rna_tools-3.8.5.data/scripts/rna_simrna_cluster.py,sha256=6XQTsrtlNjEdwSDltdpo5kFCrDJe7s_PT0zSJj3KwCo,1221
-rna_tools-3.8.5.data/scripts/rna_simrna_extract.py,sha256=5Dg076wBbu0usBOnTpq9aiDsGIepMjzMWcCHlebi7fw,3636
-rna_tools-3.8.5.data/scripts/rna_simrna_lowest.py,sha256=ogh0uqa7CYnpljbplYAhJJchmyBCSRbBZ7vKP8qIuUA,1265
-rna_tools-3.8.5.data/scripts/rna_simrnaweb_download_job.py,sha256=V7TjQi_MROq0elUKbg0Q-FPl6DP8fbLs6dvFk3HPhzk,12460
-rna_tools-3.8.5.data/scripts/rna_ss_get_bps.py,sha256=TQ6LRHgzoRw1V6HxcmgiFkIU8QQTtDDPDrhxCQihDgk,1645
-rna_tools-3.8.5.data/scripts/rna_tools_which.py,sha256=juO5A1cAFKrHynxkbW08jjv_Kgj0zf3rKlQTYJHKp44,678
-rna_tools-3.8.5.data/scripts/rna_traj_n.py,sha256=t1es4Kizu76mNG5KUBRPwE-YcWbAmQjPSD5Oa4VnIgg,1182
-rna_tools-3.8.5.data/scripts/rna_traj_nohetm.py,sha256=vXUScBzsx49Oghg8Npslx7Tvoy2I2Rw9OL90Rm3cC5A,1151
-rna_tools-3.8.5.data/scripts/rna_traj_range.py,sha256=hSZDz7QwLkxS1s36mCmbuxtxOofnRgpKxsqtKNr8-b4,1798
-rna_tools-3.8.5.data/scripts/rna_workflow.py,sha256=3WR5LVGWlGgCjWR-Z5Al2VW0mx14GUrFC6OqQICqvaM,2625
-rna_tools-3.8.5.data/scripts/rna_x3dna.py,sha256=_U9meKmqN83lIprc_SqZyyCkWEJq-Dmglo8A1RfMlr8,6714
-rna_tools-3.8.5.data/scripts/select_seq_fromBLAStn_6outfm.py,sha256=VQsBrgMsn_DajVmEk_XsPH-nw3cMIumwK4JcRSaiF2A,3945
-rna_tools-3.8.5.data/scripts/translate.py,sha256=4UgXoQMlG083J0IavivD-Czh91jC0aoPXTz0FJ8Dq_4,489
-rna_tools-3.8.5.data/scripts/triplexibility.py,sha256=VEKj79t1N2NZ4ZMIIq1FP3ZTfEDejPnrdrb_sYWvSxA,16446
-rna_tools-3.8.5.data/scripts/triplexibility2.py,sha256=XzA0XR0MhG2degMmDS-MKjHsTJTA7NVtrscBKu9i-fA,13671
-rna_tools-3.8.5.data/scripts/trx_mutgen.py,sha256=_X68edcmos9xPg3Wi5IDuoXh5usJmkT-AhnbHrEKcbw,1538
-rna_tools-3.8.5.data/scripts/trx_score.py,sha256=s23YU7oBqMzes8AlQg7nxRZepbRM5RWV3usmsqKg868,3282
+rna_tools-3.8.6.data/scripts/clarna.py,sha256=IV_CzDAy1OXiBLrF1NyUoSt0MCDl3RE9SH4D1wIpvsc,46835
+rna_tools-3.8.6.data/scripts/clarna_app.py,sha256=vJiFj1Ue-0oIEFj6S4cyQpjIC4e7yRIiJE5WUlunKaQ,5672
+rna_tools-3.8.6.data/scripts/diffpdb.py,sha256=g4wZz4SEw0gMfhoB3FmvNdjuuaeBwgivk-hPqEDVQxY,4636
+rna_tools-3.8.6.data/scripts/pdbs_measure_atom_dists.py,sha256=b37gL1BH63zACS8u37JuKNDBQ_Ex3UeaFaSrPNZgkjc,5979
+rna_tools-3.8.6.data/scripts/pymol_color_by_conserv.py,sha256=pmoDOvvFyv0geoW6_eCcGjjOyXs8ChqtVR_jYPujVL8,2091
+rna_tools-3.8.6.data/scripts/pymol_preview_generator.py,sha256=drE_KnNfanazI6tQciTEFPhXgXTaFUvqIh67NdiX6GY,3354
+rna_tools-3.8.6.data/scripts/pymol_preview_install.py,sha256=fisyGVgRuBgMnxAJATs0QXIYoljVQymLPQGUnac9Hvs,983
+rna_tools-3.8.6.data/scripts/random_assignment_of_nucleotides.py,sha256=EuxMreZF--cjmCQ1CpErTEHly53YE8qz0H7gXbvd2fk,10146
+rna_tools-3.8.6.data/scripts/renum_pdb_to_aln.py,sha256=DCdpoZw-eTWHapARasFSlvBunNiAfGfhhQdlcnqkKtA,3941
+rna_tools-3.8.6.data/scripts/reverse.py,sha256=tjFE74uGoY7__Ms3Wnir4xNHiS-qFt_f4hTXCgFsyxE,141
+rna_tools-3.8.6.data/scripts/rna_add_chain.py,sha256=aQQOz-5oZU_jd3CdzZjDo35gxkrhNmqvSb3aIRTxKXk,2381
+rna_tools-3.8.6.data/scripts/rna_align_find_core.py,sha256=upZeHDrPOjj3IQA8nnE3jPuvsSRPQI7v8CNVNI-5n_o,637
+rna_tools-3.8.6.data/scripts/rna_align_find_seq_in_alignment.py,sha256=O_hMeconRaHufOc1PQReQn3yKTTSJ0K8tcKskmd5jhM,2161
+rna_tools-3.8.6.data/scripts/rna_align_foldability.py,sha256=ik1GiMmDp9c8VI1FAQp-fsHTJxvpl0yZMhfyMPaEcC4,13320
+rna_tools-3.8.6.data/scripts/rna_align_get_ss_from_stk.py,sha256=3Ud__SibvdnXoKv_24sJGJ6o9HpTtAiHkPrS3xLs5oE,1327
+rna_tools-3.8.6.data/scripts/rna_align_seq_to_alignment.py,sha256=wFqE4aa0T1BlXQUilJ9_YSk4fNb1boQzORLUkgxswyg,1662
+rna_tools-3.8.6.data/scripts/rna_align_strip_stk.py,sha256=nKFWOrCQtdqeI5vx4VdBwEQScCaxdnB3fQwO0h7NSNw,712
+rna_tools-3.8.6.data/scripts/rna_alignment_get_species.py,sha256=5-oPt_9nKK8m8NCQo6yb4ckuP-5WJdvXuRfrD7e0jQw,9708
+rna_tools-3.8.6.data/scripts/rna_alignment_process_id.py,sha256=m38TLU7iXvDWbZzV5sWhPvcQUuoBcKHGO4Zjm0EG2Xw,1919
+rna_tools-3.8.6.data/scripts/rna_alignment_r2r.py,sha256=dnXLDi1f54KyIO4yU95gbi_K-Phv2qmIr-B7QdCNAww,2301
+rna_tools-3.8.6.data/scripts/rna_cal_rmsd_trafl_plot.py,sha256=mCYsn3sYdBGKx0t6wmatOB9Gseb3omsd6vTOukZqQ1M,1318
+rna_tools-3.8.6.data/scripts/rna_calc_evo_rmsd.py,sha256=1BVTZlb8CEQYSyvtst5aGdpFv2U_ukbXB7e6cYenP80,6805
+rna_tools-3.8.6.data/scripts/rna_calc_fenergy.py,sha256=-8F2yJyLP9XYpu29MDa5fPGE5HVmMqNWHoZd9tBE4GE,2453
+rna_tools-3.8.6.data/scripts/rna_calc_hbonds.py,sha256=pjpBUQdMg61k794BPMHkH_krYU8tHhUCsM696B2kevM,2435
+rna_tools-3.8.6.data/scripts/rna_calc_inf.py,sha256=ZuZtErEKswe7qK8Lw_vP-UZH47i2mo3p17GEO7M9L2w,9995
+rna_tools-3.8.6.data/scripts/rna_calc_rmsd.py,sha256=EW-ZAiWxPeI7YMqmmibg55joK5VaLE28oX6qilY0uR8,13430
+rna_tools-3.8.6.data/scripts/rna_calc_rmsd_all_vs_all.py,sha256=jWe90O2PskmF1LROkQRPb5WA_PHB-0ptUICzH3Bv1dg,4186
+rna_tools-3.8.6.data/scripts/rna_calc_rmsd_biopython.py,sha256=wWoNFKGzBP6VJ023_SSDlW5PDgCqHaUQ-X_YH96TCyk,11464
+rna_tools-3.8.6.data/scripts/rna_calc_rmsd_multi_targets.py,sha256=38aZ3LM-GQGxlKEQT2-BRwSPPAlpv7M2H5ldDYrq3ls,3965
+rna_tools-3.8.6.data/scripts/rna_calc_rmsd_trafl.py,sha256=KrNAmO3b1mVY7LS2JbFbGXdBH5dn57ZKNitE-acsfw4,7157
+rna_tools-3.8.6.data/scripts/rna_clanstix.py,sha256=T8yf1aiBD6OEMzlp5dEhzZZ-Tu67AxEZz7XuQJO73ks,21520
+rna_tools-3.8.6.data/scripts/rna_clarna_compare.py,sha256=s681bN2ZdBJL_xmJaSbGkvV4N5f3N_DlGUHQAoTeXLc,48049
+rna_tools-3.8.6.data/scripts/rna_clarna_run.py,sha256=drvlZbd8H36DfXmDD9ahSEiDawJE2tH6GC5UG5VbKwQ,45334
+rna_tools-3.8.6.data/scripts/rna_csv_sort.py,sha256=fTC4Piv8uyWgUKdfvBVPfrtTuuAmjoCBcSuHgrPJMlU,903
+rna_tools-3.8.6.data/scripts/rna_dG2kd.py,sha256=kHd9ayXQ9CodgbeieS-Us3kWvjYO6qoBLc-zbxpHFc0,941
+rna_tools-3.8.6.data/scripts/rna_dot2ct.py,sha256=ePiy24VuPd0lwT206lw0F_yRK4YXqkkzqeA7_RdRkgM,2435
+rna_tools-3.8.6.data/scripts/rna_draw_edges.py,sha256=sKhLKw8AhYG9kcP668JV9XBnDimQi6G1fQkFV01PRjE,5418
+rna_tools-3.8.6.data/scripts/rna_ec2x.py,sha256=10qP2UBykjb3R-Sm1FRvXfonfjj18yW2ErDjbQYI1to,2793
+rna_tools-3.8.6.data/scripts/rna_filter.py,sha256=ZA6mMhugdJ5fihZwh6hCKC0kXVPgJeTCc52ccZWIjVQ,11893
+rna_tools-3.8.6.data/scripts/rna_gromacs_ready.py,sha256=R8jv59cP-vekhAzRWLMwId2LY1Fyg0jTR5Pxp6292eI,919
+rna_tools-3.8.6.data/scripts/rna_helix_vis.py,sha256=xVMDR-y1ECtIFMzUoP46Od1Aevb9ad5tERvMU4P8Z3s,4055
+rna_tools-3.8.6.data/scripts/rna_kd2dG.py,sha256=-zQjAAvconTIEjBs4wQRgPDJu3Z0aCG6qQsQGP9-79U,999
+rna_tools-3.8.6.data/scripts/rna_kd2pkd.py,sha256=F-vYL0WYaINEP1xCNDmrzRk2uhrqmeDTeW6AGcBhpRs,705
+rna_tools-3.8.6.data/scripts/rna_md.py,sha256=x3gjaopWueV2cEyHJPzokOxna9jNjHdSQdcS3s9rIWU,5927
+rna_tools-3.8.6.data/scripts/rna_merge_dfs.py,sha256=7ECjOGjJWki46S-oFPb-5UHexShtMjF2pvTDs9O_t-o,1194
+rna_tools-3.8.6.data/scripts/rna_merge_two_dfs.py,sha256=tic_ReQLT64DzjOsEnQRqBhdk4waTCcv1C81jKpLclQ,8069
+rna_tools-3.8.6.data/scripts/rna_minimize.py,sha256=48fPHFs-BxBJvP29HF5SR6bjizUmC8XWVcdJInDePGg,3263
+rna_tools-3.8.6.data/scripts/rna_mq_clashscore.py,sha256=bkYyFihDHp0YXBfeZprRFlJb94wLbuim_XW-DsfivoM,1169
+rna_tools-3.8.6.data/scripts/rna_mq_collect.py,sha256=QmJQQMS8EUbg4tVeqWGH-PAs6hVs-wloQWuLJU3lohw,24580
+rna_tools-3.8.6.data/scripts/rna_mq_dfire.py,sha256=4EtHQcKlRihoNBoTSCHU065BVe-GwUj2933bj7Lxroo,1216
+rna_tools-3.8.6.data/scripts/rna_mq_eSCORE.py,sha256=JIDGIi5FywNDrvfcNL8nMbGhDWV6COUgUekYfvR8Nf0,908
+rna_tools-3.8.6.data/scripts/rna_mq_exp.py,sha256=xONillBR5sYy-v2gRS727TRGX5AMTvliyDDmVXS1MzQ,2428
+rna_tools-3.8.6.data/scripts/rna_mq_farfar2.py,sha256=-Mmrmu9j5Ceb2Uv9IBT9pT1nXpIRmkHqmskruWytkP0,2435
+rna_tools-3.8.6.data/scripts/rna_mq_farna.py,sha256=5eyTBKEzJlc2F-0BUIwJlfHIdhu-uVeRS7yE8HoOHhU,5044
+rna_tools-3.8.6.data/scripts/rna_mq_plot.py,sha256=4cPhtRdUt2tTeSpaVFjCiNib5l1veY5rYEVp-nlsOl8,1428
+rna_tools-3.8.6.data/scripts/rna_mq_rasp.py,sha256=h6Xu7LLCWZNBy_KYq-gD4ONinDuVUUHoy_zakgN2FM8,1371
+rna_tools-3.8.6.data/scripts/rna_mq_remote.sh,sha256=fojmce5HxCKK7kNhjXOWqwRUTKIybQefR5TEhe3VovM,290
+rna_tools-3.8.6.data/scripts/rna_mq_rna3dcnn.py,sha256=TlYYi75HPHRaigzOBqar8OL6gHqJLz5o8OQbT5FNotA,949
+rna_tools-3.8.6.data/scripts/rna_mq_rnakb.py,sha256=gRtEOHZJ2MHW6n-Qau7LLMIdbi6a6VS8i6NZY1fua0E,1707
+rna_tools-3.8.6.data/scripts/rna_mq_rsRNASP.py,sha256=jSvYXnUvfoqq1m5_z69IGi6m8O4cTgaSSFsccbHEMRI,912
+rna_tools-3.8.6.data/scripts/rna_mq_score.py,sha256=zstVn5pykuI9jif86gNAgTPWKX-JgUX_0woLAb_8t9Q,4635
+rna_tools-3.8.6.data/scripts/rna_mq_simrna.py,sha256=V0qRvy1CkO1wRNm7dKk2OPmy71epEdz6q-pzQfadhwg,1546
+rna_tools-3.8.6.data/scripts/rna_mq_wx_collect.py,sha256=adJxw1zTjL6KKS9YpIbjfDvnuznfQU-BMrZH6gIfuQs,2713
+rna_tools-3.8.6.data/scripts/rna_ok,sha256=UMqzXXeuGbmgoRG9O-lR1EwqitZsXYE9AWinJg2TCGQ,64
+rna_tools-3.8.6.data/scripts/rna_ok_mail.py,sha256=iMLKNcD0xQmp7lc9j8OS5IND2Jc0TXtpNqksj7BSQLw,1369
+rna_tools-3.8.6.data/scripts/rna_pairs2SimRNArestrs.py,sha256=KQmaGk62dAOT77ZoR1QXbXQOLcOfeWC-l-udO8isljE,1789
+rna_tools-3.8.6.data/scripts/rna_pairs_diff.py,sha256=Q-_PhoPcuY-T1pbzKG6BTic6dZtXdPCslHEfo0v6TbE,1384
+rna_tools-3.8.6.data/scripts/rna_pdb_edit_occupancy_bfactor.py,sha256=jd_SqILeGndYjUVxX_WJQqiNTYs6OqUl-P3K6iCskfI,6145
+rna_tools-3.8.6.data/scripts/rna_pdb_merge_into_one.py,sha256=t0-CAwi6X9bfeNnGWzwguYW0geQi8yZs3-sTZMVkbFo,415
+rna_tools-3.8.6.data/scripts/rna_pdb_merge_structure_with_fragments.py,sha256=8skR7gd4jyP6IeTGhphw93sL_phn_AyQLAUzpWn5yAI,4905
+rna_tools-3.8.6.data/scripts/rna_pdb_replace.py,sha256=AGg8V-Yk0C3R8tzJzjQkBsS-uCbSyYVZsPCHAyoSt4s,1743
+rna_tools-3.8.6.data/scripts/rna_pdb_toolsx.py,sha256=WLEir8d9dlpGNBDEuuvdFrafOe-TdOOTebGuyQf76m8,41410
+rna_tools-3.8.6.data/scripts/rna_pk_simrna_to_one_line.py,sha256=qBy-nO1BCLXsqOWVjlg2aS8_iX_I1I-tdQKv4B2tepE,1039
+rna_tools-3.8.6.data/scripts/rna_pkd2dG.py,sha256=FYIZk8Mw3QSBklgR0O3ZW8pkzKYChcCOFSvOcnKFuRc,886
+rna_tools-3.8.6.data/scripts/rna_pkd2kd.py,sha256=qHIqibImjh6Sl7Y9hpsUsyGXf27qy6R0Q1aaFyAlXuY,783
+rna_tools-3.8.6.data/scripts/rna_plot_boxplotlike.py,sha256=E0c3sZIqexnl5deDPeCZ1uWriXjE4Il8pTxvKkbx5m4,2263
+rna_tools-3.8.6.data/scripts/rna_plot_dendogram.py,sha256=CVjg13_1ptnvAgzW7mGOb8iZgRO-Bf3khEPI02PzJT4,1610
+rna_tools-3.8.6.data/scripts/rna_plot_density.py,sha256=j3aNK2qQfgvANOyu9HVH9og9yx_CmIXbZp86Hn9aGOQ,1780
+rna_tools-3.8.6.data/scripts/rna_plot_heatmap.py,sha256=FdK_h6F-7XdysaLyXt8BbffiCVl695Ln_MmS68-Eimg,2024
+rna_tools-3.8.6.data/scripts/rna_plot_hist.py,sha256=ao_22HSl_ssyd9uYKIMOWS0ThkRkTh0laW2gYFx0IVc,1946
+rna_tools-3.8.6.data/scripts/rna_prediction_significance.py,sha256=q-PpYRCHYj85WU_-O8oB-GqROXLMxRYL0pXcJU83jEM,1079
+rna_tools-3.8.6.data/scripts/rna_refinement.py,sha256=yWhOHjD7QwUoPebuOL6vtb_2hKkrJb4duoHwjmKpd9U,7029
+rna_tools-3.8.6.data/scripts/rna_rosetta_check_progress.py,sha256=ZnnScybmyhAE4_GT1j7jmYvIJgb0-yeZbC8vgXvayXU,6070
+rna_tools-3.8.6.data/scripts/rna_rosetta_cluster.py,sha256=-ME8xtu3tZxh76SMQf4JKLAdJKrUoPhiXPR9yQSrf0M,5462
+rna_tools-3.8.6.data/scripts/rna_rosetta_extract_lowscore_decoys.py,sha256=rtOoYdXU1nH5zs-CQpq5jz6kQ2QtTSa_CXtyfG5n43A,1523
+rna_tools-3.8.6.data/scripts/rna_rosetta_head.py,sha256=puKYbGWowXfsWBkmvAKCYeZobTGbRXYWPfDgF_t-xmE,3977
+rna_tools-3.8.6.data/scripts/rna_rosetta_min.py,sha256=-_As6TK-nHddrBWzE_j6_ZESZ0DpL3bM3K3AZv41Vvo,4903
+rna_tools-3.8.6.data/scripts/rna_rosetta_n.py,sha256=UnGxNcLSXwRMmHMt38K9JT0MhnCvFKUpAM_F3ogpP0E,1045
+rna_tools-3.8.6.data/scripts/rna_rosetta_run.py,sha256=-hfi00IlE_1k43xvbdpDrNEFI1aYTtaPCTxB9DbzG6Q,10095
+rna_tools-3.8.6.data/scripts/rna_sali2dotbracket.py,sha256=wNjnSqr4nIEuUN4ZAmzd7hAPxnkWFXx11p9jSPfx-PA,3450
+rna_tools-3.8.6.data/scripts/rna_secondary_structure_prediction.py,sha256=GvfyLqtsyBGxmpv0bLaL7UcT5G95ppnebEVyPK2mjOg,2159
+rna_tools-3.8.6.data/scripts/rna_simrna_cluster.py,sha256=6XQTsrtlNjEdwSDltdpo5kFCrDJe7s_PT0zSJj3KwCo,1221
+rna_tools-3.8.6.data/scripts/rna_simrna_extract.py,sha256=5Dg076wBbu0usBOnTpq9aiDsGIepMjzMWcCHlebi7fw,3636
+rna_tools-3.8.6.data/scripts/rna_simrna_lowest.py,sha256=ogh0uqa7CYnpljbplYAhJJchmyBCSRbBZ7vKP8qIuUA,1265
+rna_tools-3.8.6.data/scripts/rna_simrnaweb_download_job.py,sha256=V7TjQi_MROq0elUKbg0Q-FPl6DP8fbLs6dvFk3HPhzk,12460
+rna_tools-3.8.6.data/scripts/rna_ss_get_bps.py,sha256=TQ6LRHgzoRw1V6HxcmgiFkIU8QQTtDDPDrhxCQihDgk,1645
+rna_tools-3.8.6.data/scripts/rna_tools_which.py,sha256=juO5A1cAFKrHynxkbW08jjv_Kgj0zf3rKlQTYJHKp44,678
+rna_tools-3.8.6.data/scripts/rna_traj_n.py,sha256=t1es4Kizu76mNG5KUBRPwE-YcWbAmQjPSD5Oa4VnIgg,1182
+rna_tools-3.8.6.data/scripts/rna_traj_nohetm.py,sha256=vXUScBzsx49Oghg8Npslx7Tvoy2I2Rw9OL90Rm3cC5A,1151
+rna_tools-3.8.6.data/scripts/rna_traj_range.py,sha256=hSZDz7QwLkxS1s36mCmbuxtxOofnRgpKxsqtKNr8-b4,1798
+rna_tools-3.8.6.data/scripts/rna_workflow.py,sha256=3WR5LVGWlGgCjWR-Z5Al2VW0mx14GUrFC6OqQICqvaM,2625
+rna_tools-3.8.6.data/scripts/rna_x3dna.py,sha256=_U9meKmqN83lIprc_SqZyyCkWEJq-Dmglo8A1RfMlr8,6714
+rna_tools-3.8.6.data/scripts/select_seq_fromBLAStn_6outfm.py,sha256=VQsBrgMsn_DajVmEk_XsPH-nw3cMIumwK4JcRSaiF2A,3945
+rna_tools-3.8.6.data/scripts/translate.py,sha256=4UgXoQMlG083J0IavivD-Czh91jC0aoPXTz0FJ8Dq_4,489
+rna_tools-3.8.6.data/scripts/triplexibility.py,sha256=VEKj79t1N2NZ4ZMIIq1FP3ZTfEDejPnrdrb_sYWvSxA,16446
+rna_tools-3.8.6.data/scripts/triplexibility2.py,sha256=XzA0XR0MhG2degMmDS-MKjHsTJTA7NVtrscBKu9i-fA,13671
+rna_tools-3.8.6.data/scripts/trx_mutagen_allvsall.py,sha256=UuSbz2-HpWo64XgOqtv5APhgvxSKgHjpRDrYliYWgAI,1658
+rna_tools-3.8.6.data/scripts/trx_mutgen.py,sha256=2MAJcHUVuk7K5fkH__Cl9ja4kf5J3hI2dqGkI5-4Hm8,1544
+rna_tools-3.8.6.data/scripts/trx_score.py,sha256=L5K0L-FQM_jnnZeDft53yl8XYA-AhnSMvSUykgGqkZk,3321
+rna_tools-3.8.6.data/scripts/trx_score_allvsall.py,sha256=e6U9rZzyr5YtVld1QZx2mfw_KbwUx3pldwyXx_Ti3T8,3275
 tests/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
-rna_tools-3.8.5.dist-info/LICENSE,sha256=MW_8O2W3ZdoVmMUrXjvd_5MShO_yIaiDfeQfz_qabSc,35142
-rna_tools-3.8.5.dist-info/METADATA,sha256=UaoQUYL70UaknUXk9Jd4Fx2GeaQT4ryPajhT6QPMSos,33691
-rna_tools-3.8.5.dist-info/WHEEL,sha256=OqRkF0eY5GHssMorFjlbTIq072vpHpF60fIQA6lS9xA,92
-rna_tools-3.8.5.dist-info/top_level.txt,sha256=Wp7Wgi1fe4eWF1lvzJ26Y_C-ePOCu9Bgh4OeTO2s6nE,16
-rna_tools-3.8.5.dist-info/RECORD,,
+rna_tools-3.8.6.dist-info/LICENSE,sha256=MW_8O2W3ZdoVmMUrXjvd_5MShO_yIaiDfeQfz_qabSc,35142
+rna_tools-3.8.6.dist-info/METADATA,sha256=e7hLDJHLvaQh6Wg69yNOXPvgrKnKvLYVoTPyuYg-6Wc,33691
+rna_tools-3.8.6.dist-info/WHEEL,sha256=OqRkF0eY5GHssMorFjlbTIq072vpHpF60fIQA6lS9xA,92
+rna_tools-3.8.6.dist-info/top_level.txt,sha256=Wp7Wgi1fe4eWF1lvzJ26Y_C-ePOCu9Bgh4OeTO2s6nE,16
+rna_tools-3.8.6.dist-info/RECORD,,
```

